---
title: "The Role of Skin-Specific TRAs on Skin Cancer"
author: "Binnur Özay, Christofer Richard, Denisa Neacsu, Pinar Cavus"
supervisor: "Dr. Maria Dinkelacker"
tutor: "Nils Mechtel"
date: "Data Analysis Project SS21"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# ABSTRACT

Tissue restricted antigen genes have been researched on because of their ability to open ways to new personalized and specific cancer treatments which is not harmful for the rest of the organism, except for the concerned tissue. In this project, certain tissue restricted antigen genes in melanoma show upregulated expressions after being treated with MAPK pathway and MEK inhibitors. As they are analyzed further, genes that are expressed under the influence of the sun and genes related to melanosome biogenesis are found. The dependency of the upregulated genes on the genes representing the rest of the expressions with highest variance was questioned. For this analysis, methods such as principal component analysis, box plots, log fold change, and linear regression models have been used.


# INTRODUCTION

Skin cancer makes up most of the malignancies all around the world, and consists of two main different types; melanoma and non-melanoma (Linares _et al_., 2015). Both are mainly caused by exposure to UV lights, but non-melanoma cancer develops in the outer layers of the skin (Leiter & Garbe, 2008), whereas melanoma develops from the melanocytes (Smith _et al_., 2003). Melanomas only make up a small amount of reported skin cancer diagnoses, however they still account for the majority of skin cancer deaths, making them extremely dangerous (Smith _et al_., 2003). Melanomas have the highest rate of metastatic capacity of all tumors when they have delayed diagnoses (Potrony _et al_., 2015).

More than half of all melanoma cases contain a mutation of the BRAF gene (Alqathama, 2020). The B-Raf protein is a signaling molecule in the Mitogen-activated protein kinase (MAPK) pathway, activating MEK and leading to the phosphorylation of ERK (Alqathama, 2020). This kinase cascade is known to direct important cell processes such as proliferation, differentiation, and survival (Alqathama, 2020).

In this analysis, data from the melanoma cell line WM266-4 is used. This cell line has a BRAF-V600D mutation, meaning that the valine at codon 600 is changed into aspartic acid. This mutation leads to an overexpression of BRAF, thus an uncontrolled increase in oncogenic signaling (Alqathama, 2020). The cell line is treated with Trametinib, which is an inhibitor of MEK 1 and MEK2 (Zeiser, 2014), ERK1/2-inhibitor, and doxycycline-shERK1 for different periods of time; 3 and 24 hours, 3 and 7 days. Later, the mRNA from these cells are extracted and processed into labeled cRNA fragments. These fragments are hybridized on microarrays and scanned, resulting in expression level quantifications of large amounts of genes. It is to mention that there are no units in this data, because it is simply light intensity scanned.

The focus in this microarray analysis is a mechanism cancer cells have been shown to use; the upregulation of tissue restricted antigen (TRA) genes. TRAs are highly expressed selectively in their tissues, and T-cells reacting to these self-antigens result in autoimmune diseases (Kyewski & Derbinski, 2004). So that immature T-cells can gauge the self-reactivity of their antigen receptors, they go through a selection in the thymus, where T-cells reacting to TRAs undergo apoptosis (Klein _et al_., 2014). TRA genes are mostly upregulated in cancer cells for undiscovered reasons, making them good potential drug targets for therapy (Rosenberg, 1999). Current treatments for melanomas include chemotherapy, radiation, immunotherapy, and surgical excision (Gruber & Zito, 2020). Developing a new therapy, where the upregulated skin specific TRA genes are attacked, therefore other tissues don’t get as damaged, would be a significant discovery in the medical field.

Therefore, the main aim of this project is to find upregulated skin specific genes in skin cancer. By looking into the gene expression data of TRA genes in a melanoma cell line, and comparing the variety of the expression levels between different treatments and treatment times, it is possible to gain insight into how the genes affect the survival of the cancer cells. The genes which present significant changes in their expression would be potential drug targets to further investigate, as they would be helping the cancer cell survive under the treatment.





# METHODS AND RESULTS

*Quality Control and Normalization of the Data*

The programming language R Studio Version 1.4.1106 (R Core Team, 2021) is used throughout this entire project. First, all of the data to be analyzed is downloaded; the numerical microarray data with gene expression intensities of the melanoma cell line and the nominal TRA data (Dinkelacker, 2007;2019) with names of genes originating from different tissues. The TRA data has other information such as chromosome numbers of the given genes which will be put to use later.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide', include=FALSE}

# 1. TRA Analysis
## 1.1 Selection of skin-specific TRAs

# Define work directory
projectPath = dirname(rstudioapi::getSourceEditorContext()$path)
# Import TRA Data
setwd(paste(projectPath, 'TRA Daten', sep = '/'))
TRA_Human_gtex = read.csv(file = "tra.2017.human.gtex.5x.table.tsv", sep = "\t")
TRA_Mouse_4301 = read.csv(file = "tra.2014.mouse.4301.5x.table.tsv", sep = "\t")
TRA_Human_roth = read.csv(file = "tra.2014.human.roth.5x.table.tsv", sep = "\t")
TRA_Human_Novartis = read.csv(file = "tra.2014.human.5x.table.tsv", sep = "\t")
TRA_Mouse_Novartis = read.csv(file = "tra.2014.mouse.5x.table.tsv", sep = "\t")
TRA_Human_protien_atlas = read.csv(file = "Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep = ",")
```

TRA data is made up of genes from different tissues, different cell lines of humans and mice. All skin genes are extracted and combined in a vector, all NA values are erased. As the skin data available on mice is very little, only the expression of human genes is analyzed. One set of TRA data has the information of sun exposure, as this information might be valuable further into the analysis, these genes are separated into two vectors; exposed to the sun and not exposed to the sun. A pie chart is finalized to visualize how much of the TRA data comes from skin. It is found out that skin specific genes make up only 2.98% of all TRA data.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# Skin-specific genes from TRA_Human_Novartis
tissue1 = TRA_Human_Novartis[,11]
ind1 = which(tissue1 == "skin")
TRA.symbol1 = TRA_Human_Novartis$gene.symbol
skin.TRA1 = TRA.symbol1[ind1]

# Skin-specific genes from TRA_Human_protein_atlas
tissue2 = TRA_Human_protien_atlas[,11]
ind2 = which(tissue2 == "skin")
TRA.symbol2 = TRA_Human_protien_atlas$Symbol
skin.TRA2 = TRA.symbol2[ind2]

# Skin-specific genes from TRA_Human_gtex (sun exposed)
tissue3.s = TRA_Human_gtex[,10]
ind3.s = which(tissue3.s == "Skin - Sun Exposed")
TRA.symbol3.s = TRA_Human_gtex$ensembl.symbol
skin.TRA3.s = TRA.symbol3.s[ind3.s]

# Skin-specific genes from TRA_Human_gtex (not sun exposed)
tissue3.ns = TRA_Human_gtex[,10]
ind3.ns = which(tissue3.ns == "Skin - Not Sun Exposed")
TRA.symbol3.ns = TRA_Human_gtex$ensembl.symbol
skin.TRA3.ns = TRA.symbol3.ns[ind3.ns]

# Skin-specific genes from TRA_Human_roth (no skin-specific genes)
tissue4 = TRA_Human_roth[,11]
ind4 = which(tissue4 == "skin")
TRA.symbol4 = TRA_Human_roth$gene.symbol
skin.TRA4 = TRA.symbol4[ind4]

# Skin-specific genes from TRA_Mouse_4301 (epidermis-specific genes)
tissue5 = TRA_Mouse_4301[,11]
ind5 = which(tissue5 == "epidermis")
TRA.symbol5 = TRA_Mouse_4301$gene.symbol
skin.TRA5 = TRA.symbol5[ind5]

# Skin-specific genes from TRA_Mouse_Novartis (epidermis-specific genes)
tissue6 = TRA_Mouse_Novartis[,11]
ind6 = which(tissue6 == "epidermis")
TRA.symbol6 = TRA_Mouse_Novartis$gene.symbol
skin.TRA6 = TRA.symbol6[ind6]
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 1.2 Compilation of all skin-specific TRAs
# Skin-specific TRAs for humans were put into a single vector and data was cleaned up.
# Creating vector with all the human skin genes
skin.genes.human = unique(c(skin.TRA1, skin.TRA2, skin.TRA3.ns, skin.TRA3.s))

# Removing NAs and empty values
skin.genes.human = skin.genes.human[-which(is.na(skin.genes.human))] 
skin.genes.human = skin.genes.human[-which(skin.genes.human == "")] 

# Verification that all NA and empty values were removed
which(is.na(skin.genes.human)) 
which(skin.genes.human == "")

# Saving this vector with all skin-specific human TRAs in a csv file
setwd(paste(projectPath, 'tables', sep = '/'))
write.csv(matrix(skin.genes.human), file = "skin.specific.genes.human.TRA", row.names = F, sep = "\t")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# Piechart
# Creating small vectors for each table containing only NON-skin-specific TRAs

# TRA_Human_Novartis
bla1 = which(tissue1 != "skin")
TRA.symbolA = TRA_Human_Novartis$gene.symbol
non.skin.TRA1 = TRA.symbolA[bla1]

# TRA_Human_protein_atlas
bla2 = which(tissue2 != "skin")
TRA.symbolB = TRA_Human_protien_atlas$Symbol
non.skin.TRA2 = TRA.symbolB[bla2]

# TRA_Human_gtex (sun exposed)
bla3.s = which(tissue3.s != "skin")
TRA.symbolC = TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.s = TRA.symbolC[bla3.s]

# TRA_Human_gtex (not sun exposed)
bla3.ns = which(tissue3.ns != "skin")
TRA.symbolD = TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.ns = TRA.symbol3.ns[bla3.ns]

# Putting all the small non-skin-specific TRA vectors in a general vector containing all of them
NON_skin_specific_TRA = unique(c(non.skin.TRA1, non.skin.TRA2, non.skin.TRA3.s, non.skin.TRA3.ns))

# Calculating the total number of TRA
total = length(skin.genes.human) + length(NON_skin_specific_TRA)
prozent = round(length(skin.genes.human)/total*100, 2)
prozent_2 = round(length(NON_skin_specific_TRA)/total*100, 2)

labels = c("Skin-specific TRAs", "Other TRAs")
labels = paste(labels, c(prozent, prozent_2))
labels = paste(labels,"%", sep = "")

# Make pie chart
library(RColorBrewer)
color = brewer.pal(length(labels), "Spectral") 

pie(c(prozent, prozent_2), labels = labels, col = color, main = "Pie Chart skin specific TRAs")
legend("topright", labels, cex = 0.8, fill = color)

# Saving the pie chart
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file="Piechart_skin_TRA.eps") # eps file
dev.copy2pdf(file="Piechart_skin_TRA.pdf") # pdf file
```

Before the analysis, the data must be investigated to see if there are any problems within, for instance a broken microarray chip, that would alter the results. Single chips are controlled by visualizing raw microarray data, there aren’t any abnormalities visible in all of the fifteen chips.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# 2. Quality control of microarray data of skin cancer
# The Affymetrix Human Genome U133 Plus 2.0 Array was used and the package for annotation of the chips was downloaded from the Brainarray website.

## 2.1 Loading the skin cancer dataset
# 15 chips were chosen to be analysed and were downloaded from the GEO website. The 15 chips that were chosen to be used are: GSM1387513, GSM1387514, GSM1387515, GSM1387525, GSM1387527, GSM1387537, GSM1387538, GSM1387544, GSM1387545, GSM1387552, GSM1387553, GSM1387554, GSM1387558, GSM1387559 and GSM1387560.

# Loading the necessary libraries
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(stringr)

#Read all 15 microarray chips
setwd(paste(projectPath, 'rawdata/GSE57721 melanoma', sep = '/'))
data.skin.cancer = ReadAffy()
data.skin.cancer@cdfName = "HGU133Plus2_Hs_ENST"

orig = colnames(exprs(data.skin.cancer))
new.name = substr(orig, 1, nchar(orig)-4)
colnames(exprs(data.skin.cancer)) = new.name

#Saving raw data as a .rda file
setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "data.skin.cancer.rda")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 2.2. Single chip control
#Visualizing raw microarray data
par(mfrow = c(3,5))
image(data.skin.cancer,col=rainbow(100,start = 0, end = 0.75) [100:1]) #All of the chips seem to have a very good quality, so we can use all of them
```

Gene expression profiling results are highly influenced by RNA quality and degradation (Opitz _et al_., 2010), therefore RNA degradation plots are made to investigate. All of the lines, each representing a chip, showed similar slope values, and there is not any deviation visible. For this reason, all chips are to be included in the analysis as they are all qualitative.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 2.2 RNA degradation
# RNA degradation plot shifted and scaled
rna_degradation = AffyRNAdeg(data.skin.cancer)

plotAffyRNAdeg(rna_degradation, col = rainbow(150))
title(sub="Human skin cancer raw")

# Saving shifted and scaled RNA degradation plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "RNA_degradation_data_skin_cancer_raw.eps")

# RNA degradation plot only shifted
plotAffyRNAdeg(rna_degradation, col = rainbow(150), transform = "shift.only")
title(sub = "Human skin cancer raw")

# Saving only-shifted RNA degradation plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file= "RNA_deg_data_skin_cancer_raw.eps")

# Analysis of values and plots from RNA degradation
summaryAffyRNAdeg(rna_degradation)
```

The microarray data is normalized using the package vsn (v3.58.0; Hüber, NA) to remove any systematic variation that can affect the analysis. The pre-normalization and normalized data, which is glog2 transformed, are compared in the forms of boxplots and histograms, in order to see how much systematic variation existed. The boxplots didn’t show a significant difference, even before normalization the median lines are almost in a linear line, indicating that the chips did not have much variation to begin with. In both pre- and after normalization histograms, the chip lines follow the same path, and none of the chips stand out. After the normalization, a meanSdPlot is made, and from this plot, it is seen that the mean values of the chips are almost linear, once again proving that they are high quality chips.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 2.3 Normalization of microarray data
# Data normalization
data.skin.cancer.norm = vsnrma(data.skin.cancer)

# Saving normalized microarray data as a .rda file
setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "normalized_data.rda")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## Boxplots of pre-normalized and normalized data
# Before normalization
par(las = 2)
mmi = c(1.8, 0.7, 1.05, 0.54)
par(mai = mmi)
boxplot(data.skin.cancer, col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer before normalization")

# Saving boxplot with data before normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_rawdata.eps")

# After normalization
par(las = 2, mai = mmi)
boxplot(exprs(data.skin.cancer.norm), col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer after normalization")

# Saving boxplot with data after normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_normalized.eps")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## Histogram
# Histogram before data normalization
hist(data.skin.cancer, col = rainbow(150), main = "Histogram data pre-normalized", xlab = "Log intensity", ylab = "Density")

# Saving histogram before normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data pre-normalized")

# Histogram after data normalization
for (i in 1:ncol(exprs(data.skin.cancer.norm))) if (i == 1) plot(density(exprs(data.skin.cancer.norm)[, i]), col = "red", xlab = "Log intensity", ylab = "Density") else lines(density(exprs(data.skin.cancer.norm)[, i]), col = "red")

# Saving histogram after normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data normalized.eps")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 2.4 meanSd Plot
# Plotting meanSdPlot
meanSdPlot(data.skin.cancer.norm)

# Saving meanSd plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "meanSdPlot_data_skin_cancer_normalized.eps")
```


*Cleaning Data and Creating a Data Frame*

In the data from microarray chips genes are saved as numbers; to be able to view and analyse the data easier, these numbers are switched with their names using the ensembl IDs that were downloaded from https://www.ensembl.org/biomart . Later, all expression values are extracted from the microarray data, and put into a data frame with chip IDs as columns and gene symbols as rows.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# 3. Analysis
## 3.1 Creating the gene expression data frame

# Extraction of the expression values
Expression = exprs(data.skin.cancer.norm)

# Select only ENST IDs, which represent the transcript IDs
enst_id = str_detect(rownames(Expression),"ENST")
Expression = Expression[enst_id,]

# Remove .x_at suffix from each row name
rownames(Expression) = sub("\\..*", "", rownames(Expression)) 
# Remove .CEL suffix from each column name
colnames(Expression) = sub("\\..*", "", colnames(Expression))

# Load ensemble IDs and gene symbols from annotation data
setwd(paste(projectPath, 'tables', sep = '/'))
Ensembl = read.csv(paste(projectPath, 'tables/ensembl.103.txt', sep = '/'), sep = "\t")

# Extracting the transcript IDs and the HGNC symbols (gene symbols)
transcriptIDs = Ensembl$Transcript.stable.ID
HGNC_symbol = Ensembl$HGNC.symbol

# Assigning every gene symbol to a transcript ID
names(HGNC_symbol) = transcriptIDs

# Replace ensemble IDs with HGNC symbol
chipIDs = rownames(Expression)
newChipIDs = chipIDs[chipIDs %in% transcriptIDs]
Expression = Expression[newChipIDs,]
symbol = HGNC_symbol[newChipIDs]
rownames(Expression) = as.character(symbol)
head(Expression)

# Cleaning the Expression data frame (removing NAs and empty values "")
length(which(is.na(Expression)))
length(which(rownames(Expression) == ""))
all.spaces.exp = sapply(rownames(Expression), function(x){which(x == "")})
Expression = Expression[-which(all.spaces.exp>0),] 
rm(all.spaces.exp)
```

After empty spaces and NA values are removed from the data, skin specific antigens are extracted to create the main data frame of skin TRA expression levels in skin cancer to analyze. The column names are changed so that the treatment and time point of every chip is easily viewed.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 3.2 Extraction of skin-specific TRAs from the gene expression data frame
#Filtering the gene expression data frame to obtain expression values only for skin-specific TRAs
skin.genes.human.expression = which(rownames(Expression)%in%skin.genes.human)
data.skin.cancer.human_skin_TRA = Expression[skin.genes.human.expression,]

y = sort(rownames(data.skin.cancer.human_skin_TRA))
data.skin.cancer.human_skin_TRA = data.skin.cancer.human_skin_TRA[unique(y[y !=""]),]

#Renaming the columns in the data frame with the skin-specific genes
colnames(data.skin.cancer.human_skin_TRA)[1] = "Sherk1_D3a"
colnames(data.skin.cancer.human_skin_TRA)[2] = "Sherk1_D3b"
colnames(data.skin.cancer.human_skin_TRA)[3] = "Sherk1_D3c"
colnames(data.skin.cancer.human_skin_TRA)[4] = "Sherk1_D7a"
colnames(data.skin.cancer.human_skin_TRA)[5] = "Sherk1_D7c"
colnames(data.skin.cancer.human_skin_TRA)[6] = "ERK12_inh_3ha"
colnames(data.skin.cancer.human_skin_TRA)[7] = "ERK12_inh_3hb"
colnames(data.skin.cancer.human_skin_TRA)[8] = "Trametinib_3hb"
colnames(data.skin.cancer.human_skin_TRA)[9] = "Trametinib_3hc"
colnames(data.skin.cancer.human_skin_TRA)[10] = "ERK12_inh_24ha"
colnames(data.skin.cancer.human_skin_TRA)[11] = "ERK12_inh_24hb"
colnames(data.skin.cancer.human_skin_TRA)[12] = "ERK12_inh_24hc"
colnames(data.skin.cancer.human_skin_TRA)[13] = "Trametinib_24ha"
colnames(data.skin.cancer.human_skin_TRA)[14] = "Trametinib_24hb"
colnames(data.skin.cancer.human_skin_TRA)[15] = "Trametinib_24hc"

head(data.skin.cancer.human_skin_TRA)
```

Before getting to the gene expression analysis a bar plot is created to find out which chromosome the skin specific TRAs are mostly located at. Most skin related genes are usually located on chromosome 1 (Volz _et al_., 1993), which also seems to be the case in this data.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 3.3 Bar plot
# Bar plot
# G_et al_l the chromosome numbers for all human skin TRAs
chromosome_1 = TRA_Human_Novartis[,c(3,7)]
chromosome_skin1 = as.matrix(chromosome_1[ind1,])

chromosome_2 = TRA_Human_protien_atlas[,c(6,2)]
chromosome_skin2 = as.matrix(chromosome_2[ind2,])

chromosome_3 = TRA_Human_gtex[,c(3,6)]
chromosome_skin3 = as.matrix(chromosome_3[c(ind3.s, ind3.ns),])

# Combine all 3 tables and remove duplicates and empty values to make a clean list of chromosome numbers
chromosome_gene_table = rbind(chromosome_skin1, chromosome_skin2, chromosome_skin3)
rownames(chromosome_gene_table) = c()
a = sort(chromosome_gene_table[,1])

rownames(chromosome_gene_table) = chromosome_gene_table[,1]
chromosome_gene_table = chromosome_gene_table[,2]
chromosome_list = chromosome_gene_table[unique(a[a !=""])]

# Count the number of genes on each chromosome
localization = table(chromosome_list)
localization = as.matrix(localization)
sorted_chromosomes = str_sort(rownames(localization), numeric = T)
chromosome_counts = localization[sorted_chromosomes,]

# Make bar plot
chromosome = c(1:22, "X")
barplot(chromosome_counts, main = "Distribution of skin-specific TRAs over chromosomes", xlab = "Chromosomes", ylab = "Number of skin-specific TRAs", names.arg = chromosome, cex.names = 0.7, col = "darkred")

# Saving the bar plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Distribution of skin-specific TRAs over chromosomes.eps') # eps file
dev.copy2pdf(file = 'Distribution of skin-specific TRAs over chromosomes.pdf') # pdf file
```


*Gene Expression Analysis*

To begin the analysis a heat map is created, showing the expression levels of skin specific TRA genes in all chips. This heat map is useful in recognizing expression patterns to focus on, and it gives a starting point for the further analysis. It is observed that the expression levels of some genes increase as the treatment lasts longer. These are the genes that the cancer cells upregulate against the different treatments. 

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 3.4 Heatmap
# Heatmap
library(gplots)
heatmap.2(data.skin.cancer.human_skin_TRA, key = T, col = redgreen(100), symkey = F, symm = F,symbreaks = T, scale = "row", density.info = "none", trace = "none", cexRow = 0.6, cexCol = 0.6)

# Saving the heatmap
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Heatmap for skin specific TRAs in skin cancer_1.eps') #eps file
dev.copy2pdf(file = 'Heatmap for skin specific TRAs in skin cancer_1.pdf') #pdf file
```

To begin the analysis the gene expression is visualized with box plots. There are too many genes overall, therefore the boxes look like overlapping lines, which does not give much information. Instead, only the highly expressed genes are chosen to be visualized. A highly expressed gene does not have to be an upregulated gene, and an upregulated gene has the possibility of not being a highly expressed gene. A threshold of twelve is set according to the overall expression levels to define the high expression margin. Aside from the two highest expressed genes GPNMB and ANXA2, the genes NDRG1, RPL18, DCT, CD44, and MLANA show varying expression levels between chips with levels above the threshold in some chips and below in others.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# Expression of all skin-specific TRAs in skin cancer
par(las=2)
boxplot(t(data.skin.cancer.human_skin_TRA), col = rainbow(length(rownames(data.skin.cancer.human_skin_TRA))), main = "Expression of all skin genes in melanoma", cex.axis = 0.8)
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

# Putting the highly expressed genes in a vector
highgenes = c()
for (i in length(colnames(data.skin.cancer.human_skin_TRA))) {
  highgenes = c(highgenes, names(which(data.skin.cancer.human_skin_TRA[,i] > 12)))
}
highgenes = unique(highgenes)
high = data.skin.cancer.human_skin_TRA[highgenes,]

# Expression of the highly expressed genes
par(las=2)
boxplot(t(high),col=rainbow(length(rownames(high))),main="Highly expressed skin genes in melanoma",cex.axis=0.8)
```


*Upregulated Genes*

In order to determine the mostly upregulated genes, the expression data has to be transformed by using a log fold change. There is not any data available from cells without any treatment, therefore for chips that have been treated with ERK1/2-inhibitor or Trametinib, the gene expression between the three hour and twenty four hour time points is compared to see which genes showed a higher expression after twenty one hours of treatment. The zero point for dox-shERK1 chips is taken as three days, as there is not any earlier data available.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 3. Log fold change and significance
### 3.5.1 Upregulated genes

# LogFC: Finding the most upregulated genes from each treatment
# Average the gene expression for each treatment at each time point
sherk_3d = apply(data.skin.cancer.human_skin_TRA[,1:3], 1, mean)
sherk_7d = apply(data.skin.cancer.human_skin_TRA[,4:5], 1, mean)
erk12_3h = apply(data.skin.cancer.human_skin_TRA[,6:7], 1, mean)
erk12_24h = apply(data.skin.cancer.human_skin_TRA[,10:12], 1, mean)
trametinib_3h = apply(data.skin.cancer.human_skin_TRA[,8:9], 1, mean)
trametinib_24h = apply(data.skin.cancer.human_skin_TRA[,13:15], 1, mean)

# Calculate the log fold change for each gene from each treatment
logFC_sherk = sort(sherk_7d - sherk_3d, decreasing = T)
logFC_erk12 = sort(erk12_24h - erk12_3h, decreasing = T)
logFC_trametinib =  sort(trametinib_24h - trametinib_3h, decreasing = T)

# Top logFC from each treatment
logFC_sherk_top = logFC_sherk[1:11]
logFC_erk12_top = logFC_erk12[1:10]
logFC_trametinib_top = logFC_trametinib[1:10]
```

Not all genes encountered at the end of this analysis are significantly upregulated to become drug targets. Therefore, ten genes with the highest log-ratios from different treatments are chosen to be filtered through a Welsh t-test. Here, the null hypothesis states that the genes are not differently expressed between the two time points after treatment. Throughout the chips of dox-shERK1, only MPZL3 and BCL1A genes were detected as significantly upregulated genes. From ERK1/2-inhibitor data, the genes TRPM1, EFS, ARHGEF5, TYRP1, ETV7, AACS, PPFBIBP2, TGFBI and from the Trametinib data, the genes VANGL2, TRPM1, ETV7, DAPL1, EFS, DCT, TYRP1 are identified as significantly upregulated genes.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# T-test: Filtering upregulated genes that are significant
# Checking significance for genes in the top logFC of Dox-shERK
t_sherk = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 1:3], data.skin.cancer.human_skin_TRA[x, 4:5])$p.value
}
t_test_sherk = sort(sapply(names(logFC_sherk), t_sherk), decreasing = F)
t_test_sherk_significant = t_test_sherk[which(t_test_sherk < 0.05)]
upreg_sherk_significant = names(logFC_sherk_top)[names(logFC_sherk_top) %in% names(t_test_sherk_significant)]
upreg_sherk_significant

# Checking significance for genes in the top logFC of ERK1/2 inhibitor
t_erk12 = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 6:7], data.skin.cancer.human_skin_TRA[x, 10:12])$p.value
}
t_test_erk12 = sort(sapply(names(logFC_erk12), t_erk12), decreasing = F)
t_test_erk12_significant = t_test_erk12[which(t_test_erk12 < 0.05)]
upreg_erk12_significant = names(logFC_erk12_top)[names(logFC_erk12_top) %in% names(t_test_erk12_significant)]
upreg_erk12_significant

# Checking significance for genes in the top logFC of trametinib
t_trametinib = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 8:9], data.skin.cancer.human_skin_TRA[x, 13:15])$p.value
}
t_test_trametinib = sort(sapply(names(logFC_trametinib), t_trametinib), decreasing = F)
t_test_trametinib_significant = t_test_trametinib[which(t_test_trametinib < 0.05)]
upreg_trametinib_significant = names(logFC_trametinib_top)[names(logFC_trametinib_top) %in% names(t_test_trametinib_significant)]
upreg_trametinib_significant
```

Another heatmap is used to visualize the upregulation of certain skin TRAs. It is clearly visible that the skin-specific genes have a considerable increase in expression on chips from a later time period after treatment, especially on those treated with ERK1/2 inhibitor or Trametinib. Box plots were built to visualize the expressions of the significantly upregulated genes.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

# Heatmap
library(gplots)
library(RColorBrewer)

exp.all.upregulated.tra = data.skin.cancer.human_skin_TRA[c(upreg_sherk_significant,upreg_erk12_significant,upreg_trametinib_significant),]
exp.all.upregulated.tra=exp.all.upregulated.tra[unique(rownames(exp.all.upregulated.tra)),]

yyyy = sort(rownames(exp.all.upregulated.tra))
exp.all.upregulated.tra = exp.all.upregulated.tra[yyyy,]

my_colors = brewer.pal(100, "RdBu" )
lmat = rbind(c(2,3), c(4,1))
lwid = c(1,4)
lhei = c(0.95,4)
heatmap.2(exp.all.upregulated.tra, key = T, col = my_colors, symkey = F, symm = F,symbreaks = T, scale = "row", density.info = "none", trace = "none", cexRow = 0.6, cexCol = 0.6, Rowv = F, Colv = F, lmat = lmat, lwid=lwid, lhei=lhei, main = "Expression of upregulated genes")

# Saving the heatmap
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Heatmap for skin specific TRAs in skin cancer_1.eps') #eps file
dev.copy2pdf(file = 'Heatmap for skin specific TRAs in skin cancer_1.pdf') #pdf file
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## Boxplots for upregulated genes
# Dox-shERK
data_sherk_upreg = data.skin.cancer.human_skin_TRA[upreg_sherk_significant, c(1:5)]
data.new_sherk = rbind(data_sherk_upreg, data_sherk_upreg)
data.new_sherk[c(1:2), c(4:5)] = NA
data.new_sherk[c(3:4), c(1:3)] = NA
order.vector = sort(colnames(t(data.new_sherk)),index.return=T)$ix

boxplot(t(data.new_sherk)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with Dox-shERK1", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:12), col = "grey", lty = 3)
legend("topleft", c("Dox-shERK1 after 3 days", "Dox-shERK1 after 7 days"), col = c("blue", "pink"), pch = 15, bg = "white")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# ERK1/2 inhibitor
data_erk12_upreg = data.skin.cancer.human_skin_TRA[upreg_erk12_significant, c(6:7, 10:12)]
data.new_erk12 = rbind(data_erk12_upreg, data_erk12_upreg)
data.new_erk12[c(1:8), c(3:5)] = NA
data.new_erk12[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_erk12)),index.return=T)$ix

boxplot(t(data.new_erk12)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with ERK1/2 inhibitor", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("ERK1/2 inhibitor after 3h", "ERK1/2 inhibitor after 24h"), col = c("blue", "pink"), pch = 15, bg = "white")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

# Trametinib

data_trametinib_upreg = data.skin.cancer.human_skin_TRA[upreg_trametinib_significant, c(8:9, 13:15)]
data.new_trametinib = rbind(data_trametinib_upreg, data_trametinib_upreg)
data.new_trametinib[c(1:8), c(3:5)] = NA
data.new_trametinib[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_trametinib)),index.return=T)$ix

boxplot(t(data.new_trametinib)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with Trametinib", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("left",c("Trametinib after 3h", "Trametinib after 24h"),col=c("blue", "pink"), pch = 15, bg = "white")
```

The package limma (v4.1, Ritchie _et al_., 2015) is used as a model of comparison to the upregulated genes that were found manually using the log fold change. Limma takes the normalized data and performs statistical tests to discover any quantitative changes in the expression (Ritchie _et al_., 2015). The results show mostly identical genes to the already found upregulated genes. 

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## Limma Analysis
library(limma)

# First group: We compare the expression values from day 3 to day 7 with the Dox-shERK treatment
# Create design matrix
design = model.matrix(~ 0+factor(c(2,2,2,1,1)))
colnames(design) = c("Sherk7D", "Sherk3D")

# Fit linear model
fit = lmFit(data.skin.cancer.human_skin_TRA[,1:5], design)
contrast.matrix = makeContrasts(Sherk7D-Sherk3D, levels=design)
fit2 = contrasts.fit(fit, contrast.matrix)
fit2 =  eBayes(fit2, 0.01)

# Create top table with the differentially expressed genes (DEGs) sorted by log fold change
T1 = topTable(fit2,adjust.method = "fdr", sort.by = "logFC")

# Select only the first 10 genes with a positive log fold change from the table
lola1=T1[which(T1$logFC>0),]
lola1 = lola1[1:10,]

# Confirmed top upregulated genes
intersect(upreg_sherk_significant, rownames(lola1))

# Second group: We compare 3h ERK1/2 and 24h ERK1/2 treatment
# Create design matrix
design_ERK = model.matrix(~ 0+factor(c(1,1,1,2,2)))
colnames(design_ERK) = c("ERK12_inh_24h", "ERK12_inh_3h")

# Fit linear model
data_ERK = data.skin.cancer.human_skin_TRA[,c(10:12, 6,7)]
fit_ERK = lmFit(data_ERK, design_ERK) 
contrast.matrix_ERK = makeContrasts(ERK12_inh_24h-ERK12_inh_3h, levels=design_ERK)
fit2_ERK = contrasts.fit(fit_ERK, contrast.matrix_ERK)
fit2_ERK =  eBayes(fit2_ERK, 0.01)

# Create top table with the DEGs sorted by log fold change
T2 = topTable(fit2_ERK,adjust.method = "fdr", sort.by = "logFC", number = 130) 

# Select only the first 10 genes with a positive log fold change from the table
lola2 = T2[which(T2$logFC>0),]
lola2 = lola2[1:10,]

# Confirmed top upregulated genes
intersect(upreg_erk12_significant, rownames(lola2))

# Third group: We compare 3h to 24h Trametinib treatment
# Create design matrix
design_Trametinib = model.matrix(~ 0+factor(c(1,1,1,2,2)))
colnames(design_Trametinib) = c("Trametinib_24h", "Trametinib_3h")

# Fit linear model
data_Trametinib = data.skin.cancer.human_skin_TRA[,c(13:15, 8,9)]
fit_Trametinib = lmFit(data_Trametinib, design_Trametinib) 
contrast.matrix_Trametinib = makeContrasts(Trametinib_24h-Trametinib_3h, levels=design_Trametinib)
fit2_Trametinib= contrasts.fit(fit_Trametinib, contrast.matrix_Trametinib)
fit2_Trametinib =  eBayes(fit2_Trametinib, 0.01)

# Create top table with the DEGs sorted by log fold change
T3 = topTable(fit2_Trametinib,adjust.method = "fdr", sort.by = "logFC", number = Inf ) 

# Select only the first 10 genes with a positive log fold change from the table
lola3 = T3[which(T3$logFC>0),]
lola3 = lola3[1:10,]

# Confirmed top upregulated genes
intersect(upreg_trametinib_significant, rownames(lola3))
```

Genes that are upregulated in skin cancer and are at the same time TRAs are especially interesting, as they pose as potential drug targets. To visualize these upregulated TRAs, a venn diagram was made.There are not any non-skin specific TRAs that are upregulated in skin cancer.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 3.5.2 Venn Diagram

# Finding upregulated genes in skin cancer by logFC
# Average the gene expression for each treatment at each time point
sherk_3d_all = apply(Expression[,1:3], 1, mean)
sherk_7d_all = apply(Expression[,4:5], 1, mean)
erk12_3h_all = apply(Expression[,6:7], 1, mean)
erk12_24h_all = apply(Expression[,10:12], 1, mean)
trametinib_3h_all = apply(Expression[,8:9], 1, mean)
trametinib_24h_all = apply(Expression[,13:15], 1, mean)

# Calculate the log fold change for each gene from each treatment
logFC_sherk_all = sherk_7d_all - sherk_3d_all
logFC_erk12_all = erk12_24h_all - erk12_3h_all
logFC_trametinib_all =  trametinib_24h_all - trametinib_3h_all

# Only positive logFC from each treatment (only upregulated genes!)
logFC_sherk_pos = which(logFC_sherk_all > 0)
logFC_erk12_pos = which(logFC_erk12_all > 0)
logFC_trametinib_pos = which(logFC_trametinib_all > 0)
upreg_all = unique(c(names(logFC_sherk_pos), names(logFC_erk12_pos), names(logFC_trametinib_pos)))

# T-test to filter significantly upregulated genes

# Checking significance for Dox-shERK
t_sherk_all = function(x) {
  if (Expression[x,1] == Expression[x,2] | Expression[x,1] == Expression[x,3] | Expression[x,3] == Expression[x,2] | Expression[x,4] == Expression[x,5])
    return(0.5)
  else
  t.test(Expression[x, 1:3], Expression[x, 4:5])$p.value
}
t_test_sherk_all = sapply(names(logFC_sherk_all), t_sherk_all)
t_test_sherk_significant_all = t_test_sherk_all[which(t_test_sherk_all < 0.05)]
upreg_sherk_significant_all = names(logFC_sherk_pos)[names(logFC_sherk_pos) %in% names(t_test_sherk_significant_all)]

# Checking significance for ERK1/2 inhibitor
t_erk12_all = function(x) {
  t.test(Expression[x, 6:7], Expression[x, 10:12])$p.value
}
t_test_erk12_all = sapply(names(logFC_erk12_all), t_erk12_all)
t_test_erk12_significant_all = t_test_erk12_all[which(t_test_erk12_all < 0.05)]
upreg_erk12_significant_all = names(logFC_erk12_pos)[names(logFC_erk12_pos) %in% names(t_test_erk12_significant_all)]

# Checking significance for Trametinib
t_trametinib_all = function(x) {
  t.test(Expression[x, 8:9], Expression[x, 13:15])$p.value
}
t_test_trametinib_all = sapply(names(logFC_trametinib_all), t_trametinib_all)
t_test_trametinib_significant_all = t_test_trametinib_all[which(t_test_trametinib_all < 0.05)]
upreg_trametinib_significant_all = names(logFC_trametinib_pos)[names(logFC_trametinib_pos) %in% names(t_test_trametinib_significant_all)]

# Venn diagram (Upregulated genes that are also TRAs)
library(venn)
list_venn = list(A = unique(c(upreg_sherk_significant_all, upreg_erk12_significant_all, upreg_trametinib_significant_all)), B = skin.genes.human)
names(list_venn) = c("Upregulated genes" , "TRA")

# Upregulated non-skin TRAs in skin cancer
upreg_tra = intersect(unique(c(upreg_sherk_significant_all, upreg_erk12_significant_all, upreg_trametinib_significant_all)), skin.genes.human)
upreg_tra_nonskin = upreg_tra[!upreg_tra%in%rownames(data.skin.cancer.human_skin_TRA)]
length(upreg_tra_nonskin)
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}
venn(list_venn, box = F, ilab = T, zcolor = c("#0073C2FF", "#EFC000FF"), cex = 1000)
text(500, 900, "Potential drug targets", cex = 2)
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}
# Save the venn diagramm in a pdf file
setwd(paste(projectPath, 'plots', sep = '/'))
pdf("Venn_diagramm_trial.pdf")
venn(list_venn, box = F, ilab = T, zcolor = c("#0073C2FF", "#EFC000FF"))
text(500, 900, "Potential drug targets", cex = 2)
dev.off()
```

The existence of significantly downregulated genes indicates that the treatment has been successful, and that these genes were inducing tumors before the treatment. That is why, ten genes from different treatments with the lowest log-ratio are filtered out and evaluated under t-tests, and visualized via box plots. The genes MMP2, TYRP1 from dox-shERK1 chips, CMSS1, KRT27, MPP4, CHTF18, POLA1, WEE1, REEP4, MFSD2A, CA12, C3of52 from ERK1/2 inhibitor chips, and RHNO1, MPP4, IRX3, MFSD2A, WEE1, CA12, C3orf52, POLA1 from Trametinib chips proved to be significantly downregulated.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

### 3.6 Upregulated genes
# Least logFC from each treatment
logFC_sherk_down_regulated = logFC_sherk[503:512]
logFC_erk12_down_regulated = logFC_erk12[503:512]
logFC_trametinib_down_regulated = logFC_trametinib[503:512]

# T-test: Filtering downregulated genes that are significant
# Checking significance for genes in the last 10 logFC of Dox-shERK
t_sherk = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 1:3], data.skin.cancer.human_skin_TRA[x, 4:5])$p.value
}
t_test_sherk = sort(sapply(names(logFC_sherk), t_sherk), decreasing = F)
t_test_sherk_significant = t_test_sherk[which(t_test_sherk < 0.05)]
downreg_sherk_significant = names(logFC_sherk_down_regulated)[names(logFC_sherk_down_regulated) %in% names(t_test_sherk_significant)]
downreg_sherk_significant

#Checking significance for genes in the last 10 logFC of ERK1/2 inhibitor
t_erk12 = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 6:7], data.skin.cancer.human_skin_TRA[x, 10:12])$p.value
}
t_test_erk12 = sort(sapply(names(logFC_erk12), t_erk12), decreasing = F)
t_test_erk12_significant = t_test_erk12[which(t_test_erk12 < 0.05)]
downreg_erk12_significant = names(logFC_erk12_down_regulated)[names(logFC_erk12_down_regulated) %in% names(t_test_erk12_significant)]
downreg_erk12_significant

## Checking significance for genes in the last 10 logFC of trametinib
t_trametinib = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 8:9], data.skin.cancer.human_skin_TRA[x, 13:15])$p.value
}
t_test_trametinib = sort(sapply(names(logFC_trametinib), t_trametinib), decreasing = F)
t_test_trametinib_significant = t_test_trametinib[which(t_test_trametinib < 0.05)]
downreg_trametinib_significant = names(logFC_trametinib_down_regulated)[names(logFC_trametinib_down_regulated) %in% names(t_test_trametinib_significant)]
downreg_trametinib_significant
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## Boxplots for downregulated genes
# Dox-shERK
data_sherk_downreg = data.skin.cancer.human_skin_TRA[downreg_sherk_significant, c(1:5)]
data.new_sherk_downreg = rbind(data_sherk_downreg, data_sherk_downreg)
data.new_sherk_downreg[c(1:2), c(4:5)] = NA
data.new_sherk_downreg[c(3:4), c(1:3)] = NA
order.vector = sort(colnames(t(data.new_sherk_downreg)),index.return=T)$ix
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

boxplot(t(data.new_sherk_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with Dox-shERK", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:12), col = "grey", lty = 3)
legend("topleft", c("Dox-shERK after 3 days", "Dox-shERK after 7 days"), col = c("blue", "pink"), pch = 15, bg = "white")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# ERK1/2 inhibitor
data_erk12_downreg = data.skin.cancer.human_skin_TRA[downreg_erk12_significant, c(6:7, 10:12)]
data.new_erk12_downreg = rbind(data_erk12_downreg, data_erk12_downreg)
data.new_erk12_downreg[c(1:8), c(3:5)] = NA
data.new_erk12_downreg[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_erk12_downreg)),index.return=T)$ix

boxplot(t(data.new_erk12_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with ERK1/2 inhibitor", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("ERK1/2 inhibitor after 3h", "ERK1/2 inhibitor after 24h"), col = c("blue", "pink"), pch = 15, bg = "white")
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# Trametinib
data_trametinib_downreg = data.skin.cancer.human_skin_TRA[downreg_trametinib_significant, c(8:9, 13:15)]
data.new_trametinib_downreg = rbind(data_trametinib_downreg, data_trametinib_downreg)
data.new_trametinib_downreg[c(1:8), c(3:5)] = NA
data.new_trametinib_downreg[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_trametinib_downreg)),index.return=T)$ix

boxplot(t(data.new_trametinib_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with Trametinib", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("Trametinib after 3h", "Trametinib after 24h"),col=c("blue", "pink"), pch = 15, bg = "white")
```

It should be noted that TYRP1 is significantly upregulated under Trametinib treatment, but significantly downregulated under dox-shERK1.

A Wilcoxon signed rank test was run for each chip and a significant difference between the expression levels of the two gene groups was observed. Genes expressed both under the sun and without the sun are removed from the data, and the genes that were only expressed under sun exposure are acquired. Four of these sun exposure related genes are also upregulated genes: ETV7, TGFBI, DAPL1, and ARHGEF5.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 3.7 Sun exposure
# Finding the sun exposed and not sun exposed genes in our skin-specific expression dataset
sun = intersect(rownames(data.skin.cancer.human_skin_TRA), skin.TRA3.s)
no.sun = intersect(rownames(data.skin.cancer.human_skin_TRA), skin.TRA3.ns)

# Removing the gene names to make a box plot of each chip
sun_exp = data.skin.cancer.human_skin_TRA[sun,]
nosun_exp = data.skin.cancer.human_skin_TRA[no.sun,]
sunexpchip = unname(sun_exp[,13])
nosunexpchip = unname(nosun_exp[,13])
lmts = range(sunexpchip,nosunexpchip)

# Comparing their significance with a welsh t-test for each chip
ttest_sun = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[sun, x], data.skin.cancer.human_skin_TRA[no.sun, x])$p.value
}
sapply(1:15, ttest_sun)
## All of them have a p value of smaller than 0.05, proving a significant difference between the expression of sun or no sun exposure genes. Now the expression of genes that are expressed with or without sun exposure are compared.

# Genes that are expressed with or without sun exposure
both.sun = intersect(sun, no.sun)
# Genes that are only sun exposure related
only_sun = sun[!sun%in%both.sun]
#Genes that are only no sun exposure related
only_nosun = no.sun[!no.sun%in%both.sun]

# The visualization of these genes can be seen in the graphs showing the logFC for all significantly upregulated genes.
significant_sun_1 = upreg_sherk_significant[upreg_sherk_significant%in%only_sun]
significant_sun_2 = upreg_erk12_significant[upreg_erk12_significant%in%only_sun]
significant_sun_3 = upreg_trametinib_significant[upreg_trametinib_significant%in%only_sun]

upregulated_sun = unique(c(significant_sun_1, significant_sun_2, significant_sun_3))

significant_nosun_1 = upreg_sherk_significant[upreg_sherk_significant%in%only_nosun]
significant_nosun_2 = upreg_erk12_significant[upreg_erk12_significant%in%only_nosun]
significant_nosun_3 = upreg_trametinib_significant[upreg_trametinib_significant%in%only_nosun]

upregulated_nosun = unique(c(significant_nosun_1, significant_nosun_2, significant_nosun_3))
```

The expression difference of genes exposed to sun and genes that were not exposed to sun is visualized by k-means. After running the elbow, silhouette, and one other method provided by the package NbClust (v3.0; Charrad, 2015), it is decided that two clusters are the optimal amount for this data. After clustering and plotting with the package factoextra (v1.0.3; Kassambara, 2020), two clusters are defined with an average silhouette width of 0.65, indicating that there is indeed a clusterable difference between the expressions of these two gene groups.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 4. Clustering
### 4.1 Clustering of sun exposure data
# Load libraries needed for clustering
library(factoextra)
library(clustertend)
library(fpc)
library(cluster)
library(clValid)
library(NbClust)
library(dplyr)

# Deciding on the optimal number of clusters
set.seed(123)
both = data.skin.cancer.human_skin_TRA[c(only_sun, only_nosun),]

# Elbow Method
fviz_nbclust(both, kmeans, method = "wss")+labs(subtitle = "Elbow method")

# Silhouette Method
fviz_nbclust(both, kmeans, method = "silhouette")+labs(subtitle = "Silhouette method")

# NbClust
nb.both = NbClust(both, distance = "euclidean", min.nc = 2,max.nc = 10, method = "kmeans")
fviz_nbclust(nb.both)

# The optimal number of clusters is 2.

# Is the data really clusterable?
hopkins(both, n = nrow(both)-1) #H value is lower than 0.5 and is relatively close to 1, meaning that there is a high probability that there are clusters in the sun/no sun data.

# Cluster Validation: Silhouette Information
km.both = eclust(both, "kmeans", k = 2, nstart = 30, graph = FALSE)
fviz_silhouette(km.both, palette = "jco",
ggtheme = theme_classic())
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

# Visualize k-means clusters
fviz_cluster(km.both, data = both, main = "Clustering of Gene Expression With and Without Sun", palette = c("orange", "blue"), geom = "point", ellipse.type = "convex", ggtheme = theme_bw())
```

Same process is repeated for upregulated genes to see if there are any certain sub-groups of genes inbetween them.Three distant clusters are viewed in k-means, with an average silhouette of 0.61.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

### 4.2 Clustering of upregulated genes

# Gathering the expression of highest upregulated genes
upregulated_all_exp = data.frame(data.skin.cancer.human_skin_TRA[c(upreg_sherk_significant, upreg_erk12_significant, upreg_trametinib_significant),])
upregulated_all_exp = distinct(upregulated_all_exp)

# Elbow method
fviz_nbclust(upregulated_all_exp, kmeans, method = "wss")+labs(subtitle = "Elbow method")

# Silhouette method
fviz_nbclust(upregulated_all_exp, kmeans, method = "silhouette")+labs(subtitle = "Silhouette method")

# The optimal number of clusters is 3.

# Is the data really clusterable?
hopkins(upregulated_all_exp, n = nrow(upregulated_all_exp)-1)
# The H value is also lower than 0.5 here and is relatively close to 0.
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

# Visualize k-means clusters
fviz_cluster(km.upreg, data = upregulated_all_exp, main = "Clustering of Upregulated TRAs in Skin Cancer") + theme(plot.title = element_text(hjust = 0.5))
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

# Cluster validation: silhouette plot
fviz_silhouette(km.upreg, palette = "jco", ggtheme = theme_classic())
```

Principal component analysis (PCA) is carried out on the 24h and 7d chips to find the genes that explain most of the variance in the data and to reduce the amount of dimensions. PC1 and PC2 represent 99.7% of all the variance of the melanoma dataset. Five genes with the top variance from each of these components are chosen to continue the analysis with: ANXA2, DCT, NDRG1, TYRP1, PTPRZ1 from PC1, and DSP, GJA1, PPP1R14C, TGFBI, UCN2 from PC2.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 5. Principal component analysis (PCA)

# Centering and scaling, pick data only from final time and only genes with highest variance
pca_dataframe = scale(data.skin.cancer.human_skin_TRA)
pca_dataframe = t(pca_dataframe[,c(4:5, 10:15)])

topVar = apply(pca_dataframe, 2, var)
q75 = quantile(topVar, probs = 0.75)
i.topvar = which(topVar >= q75)
pca_dataframe_topVar = pca_dataframe[,i.topvar]

# PCA
skin_pca_topVar = prcomp(pca_dataframe_topVar, scale. = F, center = F)

# Biplot of first 2 principal components
biplot(skin_pca_topVar)

# Proportion of variance of each principal component
variance = (skin_pca_topVar$sdev)^2
prop.variance = variance/sum(variance)
names(prop.variance) = 1:length(prop.variance)
barplot(prop.variance, ylab='Proportion of variance', xlab = "Principal components", main = "Proportion of variance of each principal component")

# Total variance explained by PC1 and PC2
total_var = prop.variance[1] + prop.variance[2]
total_var

# Barplots of principal components
par(las = 2)
barplot(skin_pca_topVar$rotation[,1], main = "PC1", ylab = "Contribution", xlab = "Genes")
barplot(skin_pca_topVar$rotation[,2], main = "PC2", ylab = "Contribution", xlab = "Genes")

# Genes with most contribution to PC1 and PC2
top_genes_pca1 = names(sort(abs(skin_pca_topVar$rotation[,1]), decreasing = T)[1:5])
top_genes_pca2 = names(sort(abs(skin_pca_topVar$rotation[,2]), decreasing = T)[1:5])

# Scatterplot of data with PC1 and PC2 as new coordinates
plot(skin_pca_topVar$x[,1], skin_pca_topVar$x[,2], 
     col= c("red", "red", "green", "green", "green", "blue", "blue", "blue"), pch=19,
     xlab='PC1', ylab='PC2', main = "PC1 and PC2 as new coordinates")
legend("topright",c("Sherk", "ERK1/2 inhibitor", "Trametinib"),col=c("red", "green", "blue"), pch = 15, bg = "yellow")
```

Linear regression models are used to see if there are any significant dependencies between the PC genes and the upregulated genes that would justify making a prediction of the expressions. The ability to modulate the upregulation genes by targeting the genes from PCA would open up good opportunities in developing treatments with new drug targets. Before the modeling, Pearson correlation is used to see if the upregulated genes and the PC genes have any correlations within their group. The package corrplot (v0.90; Wei, Levy & Simko, 2021) is used to visualize and calculate the correlations. All of the upregulated genes turn out to be highly correlated with each other except for MPZL3. Since the rest are highly correlated, a representative gene, ARHGEF5 is chosen to do the modeling with. Even though not as much as the upregulated genes, PC genes are found to be somewhat correlated, representatives are chosen. The upregulated genes MPZL3 and ARGHGEF5 are going to be predicted by DSP, GJA1, ANXA2.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

## 6. Linear Regression
# Loading libraries
library(corrplot)
library(QuantPsyc)
library(ggiraph)
library(ggiraphExtra)
library(ggplot2)
library(plyr)
library(car)

### 6.1 Correlation within genes

# Correlation of upregulated genes
df_upr = data.frame(t(data.skin.cancer.human_skin_TRA[unique(c(upreg_sherk_significant, upreg_erk12_significant, upreg_trametinib_significant)),]))

# Scatterplot PPFIBP2 and ARHGEF5 (example)
plot(df_upr[,"PPFIBP2"], df_upr[,"ARHGEF5"], pch=20, xlab='PPFIBP2', ylab='ARHGEF5')
# A linear relationship can be seen. 

# Creating a correlation matrix with pearson correlation
cor_upr = cor(df_upr, method='pearson')

# Visualization of the Pearson correlation
corrplot(cor_upr, type="upper", order="alphabet",title = "Correlation within the upregulated genes (Pearson)")

# Correlation of genes from PCA
df_pca = data.frame(t(data.skin.cancer.human_skin_TRA[unique(c(top_genes_pca1, top_genes_pca2)),]))

# Creating a correlation matrix with pearson correlation
cor_pca = cor(df_pca, method='pearson')

# Visualization of the Pearson correlation
corrplot(cor_pca, type="upper", order="alphabet", title = "Correlation within the PCA genes (Pearson)")
```

Next, the distribution of the residuals of the upregulated genes are investigated with a histogram, they are almost normally distributed. The residuals have to meet some conditions in order to be able to form a linear relationship with the PC genes and the upregulated genes. Their means have to be zero and the residuals must not correlate with the PC genes. It is confirmed that the residuals have a mean of approximately zero, and that they do not correlate with the PC genes.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

### 6.2.1 Multiple linear regression of MPZL3

# H0: The model doesn't refer to any significance.
# H1: The model refers to a significance.
# For the F-Test:
# If the p-value is lower than the chosen threshold 0.01, it is proven that the independent variables improve the fit in our model.

# MPZL3 expression predicted with PCA genes
mpzl3_lr = lm(MPZL3 ~ DSP+GJA1+ANXA2, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(mpzl3_lr)
# p-value < 0.01 -> model significantly better than null model
# R squared is high -> good model

# Normal distribution of residuals
hist(mpzl3_lr$residuals,breaks=20)
shapiro.test(mpzl3_lr$residuals)

# Correlation within residuals
plot(data.skin.cancer.human_skin_TRA["DSP",],mpzl3_lr$residuals,pch=20, title("Correlation of the residuals"), xlab="MPZL3", ylab="MPZL3_lr-Residuals")
cor(data.skin.cancer.human_skin_TRA["DSP",], mpzl3_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["GJA1",], mpzl3_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["ANXA2",], mpzl3_lr$residuals)
# No correlation between any of the explanatory genes and the residuals.

# Mean of the residuals
mean(mpzl3_lr$residuals) 
# Mean is approximately 0.
```

Multiple linear regression is carried out between MPZL3 and the aforementioned three PC genes, and a highly significant result is observed. With an adjusted R squared value of 0.9557 and a low p-value, it is right to say that MPZL3 expression can definitely be predicted  by these three genes’ expression levels. DSP appears to be able to predict the expression of MPZL3 the best out of all. This has been decided with help from the package QuantPsyc (v1.5; Fletcher, 2012).

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

### 6.2.2 Simple linear regression of MPZL3

# MPZL3 expression predicted with a single gene

# In order to find the variable which has the most impact on the model, the standardized regression coefficients for each predictor gene have to be calculated.
lm.beta(mpzl3_lr)
# This means a change in 1 standard deviation of DSP has more impact than a 1 standard deviation change in other variables. 
```

The linear relationship of the PC genes and ARHGEF5, with an adjusted R squared value of 0.8897, and a low p-value also show a good dependency. The expression of ARHGEF5 is best represented by ANXA2 among all three PC genes.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

### 6.3.1 Multiple linear regression of ARHGEF5

# ARHGEF5 expression predicted with PCA genes
arhgef5_lr = lm(ARHGEF5 ~ DSP+GJA1+ANXA2, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(arhgef5_lr)
# The p-value is very small (<0.01), and R squared is very close to 1, it is a good model significantly better than the null model.

hist(arhgef5_lr$residuals,breaks=20)
shapiro.test(arhgef5_lr$residuals)
#P-value is relatively high -> normally distributed.

# Correlation within residuals
plot(data.skin.cancer.human_skin_TRA["ANXA2",],arhgef5_lr$residuals,pch=20, title("Correlation of the residuals"), xlab="ARHGEF5", ylab="ARHGEF5_lr-Residuals")
cor(data.skin.cancer.human_skin_TRA["DSP",], arhgef5_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["GJA1",], arhgef5_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["ANXA2",], arhgef5_lr$residuals)
# No correlation within any of the explanatory genes and the residuals.

# Mean of the residuals
mean(arhgef5_lr$residuals)
# Mean is approximately 0.
```

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE, fig.show='hide',include=FALSE}

### 6.3.2 Simple linear regression of ARHGEF5

#ARHGEF5 expression predicted with a single gene

#variable with the most impact:
lm.beta(arhgef5_lr)
```

To confirm the accuracy of the predictions made by the PC genes on the expression of the two representative upregulated genes ARHGEF5 and MPZL3, the predicted values are plotted against the real values of expression. They seem to be nearly identical. Since the gene ARHGEF5 is highly correlated with the rest of the upregulated genes, it can be concluded that the expression of almost all of the upregulated genes can be predicted by the expression profiles of the PC genes.

```{r, echo = FALSE, results='hide', message=FALSE, warning=FALSE}

### 6.4 Plots of Linear Regression

# Predicting the MPZL3 expression with a multiple linear regression
plot(data.skin.cancer.human_skin_TRA["MPZL3",], mpzl3_lr$fitted.values,pch=20,col='blue', title("Multiple linear regression of MPZL3"), xlab='Real values',ylab='Predicted values');abline(0,1,col='red')

# Plotting MPZL3 against only the variable that has the most impact: DSP
mpzl3_dsp = lm(MPZL3 ~ DSP, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(mpzl3_dsp)
df_mpzl3_dsp = data.frame(y= data.skin.cancer.human_skin_TRA["MPZL3",], x= data.skin.cancer.human_skin_TRA["DSP",])

plot(df_mpzl3_dsp$x, df_mpzl3_dsp$y, xlab="Predictor(DSP)", ylab="Outcome(MPZL3)", main="Simple linear regression of MPZL3 by DSP", col="darkorange",pch=16)
abline(mpzl3_dsp, col="red")
for (i in 1:length(mpzl3_dsp$fitted.values))
  lines(c(df_mpzl3_dsp$x[i], df_mpzl3_dsp$x[i]), c(df_mpzl3_dsp$y[i], mpzl3_dsp$fitted.values[i]), col="blue")

# Predicting the ARHGEF5 expression with a multiple linear regression
plot(data.skin.cancer.human_skin_TRA["ARHGEF5",], arhgef5_lr$fitted.values,pch=20,col='blue', title("Multiple linear regression of ARHGEF5"), xlab='Real values',ylab='Predicted values');abline(0,1,col='red')

# Plotting ARHGEF5 against only the variable that has the most impact: TGFBI
arhgef5_anxa2 = lm(ARHGEF5 ~ ANXA2, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(arhgef5_anxa2)
df_arhgef5_anxa2 = data.frame(y= data.skin.cancer.human_skin_TRA["ARHGEF5",], x= data.skin.cancer.human_skin_TRA["ANXA2",])

plot(df_arhgef5_anxa2$x, df_arhgef5_anxa2$y, xlab="Predictor(ANXA2)", ylab="Outcome(ARHGEF5)", main="Simple linear regression of ARHGEF5 by ANXA2", col="darkorange",pch=16)
abline(arhgef5_anxa2, col="red")
for (i in 1:length(arhgef5_anxa2$fitted.values))
  lines(c(df_arhgef5_anxa2$x[i], df_arhgef5_anxa2$x[i]), c(df_arhgef5_anxa2$y[i], arhgef5_anxa2$fitted.values[i]), col="blue")

```

In conclusion, in both multiple regression models the p-value is low enough to overthrow the H0 hypothesis. The linear model refers to significance and the dependency between the genes is significant enough to predict upregulated genes.


# DISCUSSION

Regardless of treatment type and time variation between different chips, GPNMB and ANXA2 proved to be highest expressed in all of them. ANXA2 is known as a biomarker in cancers, it is already seen as an ideal target for cancer therapy because of its crucial functions in cancer cell proliferation, migration, invasion, metastasis (Wang _et al_., 2019). GPNMB has been formerly identified as an aggressive pro-metastatic protein that is highly expressed in many tumor types, including melanoma (Taya & Hammes, 2018). The fact that brings these two together is that they are both responsible for the genesis of melanosomes (Setaluri & Jayanthy, 2013; Delevoye _et al_., 2015).

Melanoma develops from melanocytes, which normally assemble melanosomes; organelles in mammals responsible for melanin production (Setaluri & Jayanthy, 2013). It makes sense that these two genes related to the assembly and maturation of melanosomes are found to be highly expressed in melanoma. The two other important genes TYRP1 and DCT found in this analysis, which are also involved in the development of melanosomes (Setaluri & Jayanthy, 2013), are both upregulated and principal component genes at the same time. DCT also known as TYRP2 encodes proteins that play roles in the melanin synthesis, just as TYRP1 (Setaluri & Jayanthy, 2013). The upregulation and high expression of these melanosome related genes indicates that a therapy focusing to direct melanosome functions would be effective against melanoma.

Microphthalmia-associated transcription factor (MITF) is a key signal regulator with the subset target genes GPNMB and TYRP1 (Rose _et al_., 2016). MITF plays important roles in melanosomal function, pigmentation, but it is also required for sustaining the BRAF-mutant melanoma, and supporting the melanoma cell proliferation (Rose _et al_., 2016). It has been observed before that MITF related genes are upregulated under certain BRAF and MEK inhibitors, such as Trametinib (Rose _et al_., 2016). This would explain the expression results of TYRP1 in this project. It is upregulated as it is needed for the maintenance of the melanoma cell, plus the Trametinib treatment, which is a MEK1/2 inhibitor further accelerates its expression. The other two treatments are inhibitors of ERK, which partially successfully seem to have worked, because one of the genes supporting the cancer cell, TYRP1, is downregulated under them.

When the upregulated genes are clustered, three groups are to be seen. The genes in the first cluster seem to be mostly prognostic markers (Zheng _et al_., 2021; Zhou _et al_., 2020; Neumann _et al_.,). MPZL3 is the most important one between them, as it did not correlate with any of the other upregulated genes in this data. An overexpression of this gene is usually associated with increased CD8+ immune infiltration and microsatellite instability, leading to better prognosis (Zheng _et al_., 2021). The second cluster (green) is made up of two genes which are closely related each other; TYRP1 and DCT (TYRP2) code for the same family of proteins and take part in the melanosome biosynthesis as mentioned before (Kobayashi _et al_., 1998). 

In the third cluster (blue) a couple of genes are found to be related to each other. TGFBI is a pro-inflammatory factor induced by TGFß, proven to be playing a role in the epithelial-mesenchymal transition (EMT) of cells (Pérez _et al_., 2017). EMT is how epithelial tumor cells usually obtain malignant properties (Komiya _et al_., 2016). ARHGEF5 is also activated by TGFß and helps execute malignancy through EMT (Komiya _et al_., 2016). However, a melanoma cell does not go through EMT as it is not derived from epithelial cells (Noguchi _et al_., 2017). The proliferative cells go into their invasive stage by a phenotype switching when the MITF is lowly expressed, meaning that an EMT-like process happens when the high levels of MITF in the cell drop (Noguchi _et al_., 2017).

Data in this project on the other hand shows that MITF related genes and EMT related genes have been both upregulated and are even under the third cluster, contradicting the former information. Two other genes, TRPM1 and PPFIBP2, from the same cluster also show a relation to the already known melanoma marker MITF (Mohammadi _et al_., 2018). Former studies have revealed that TRPM1 is an indicator of melanocytic differentiation and is regulated by MITF (Miller _et al_., 2004), and that PPFIBP2 is a part of the MITF sub-network (Mohammadi _et al_., 2018). This makes the reason they are clustered together more a wonder.

They are all related because MITF has been found to specially induce the expression of TGFBI (Lavelle _et al_., 2020). TGFBI protein regulates cyclin-dependent kinases in melanomas, which are known to be in charge of the cell cycle, and allows a high metastatic potential to the melanoma (Lauden _et al_, 2014). The upregulation of ARHGEF5 is also explained this way, as it seems to be closely related to TGFBI from former results. Therefore, it is even further proven that MITF genes have a really important role in melanoma development, and survival. They are highly related to each other over MITF, which makes MITF a great drug target for melanoma.

Another observation to mention here is that all four of these genes are upregulated under the ERK1/2 inhibitor. It is known that high ERK activity lowers the MITF protein levels (Wellbrock & Arozarena, 2015), meaning an inhibitor of ERK has an enhancing effect on the upregulation of the MITF related genes. This gives this cluster of genes another meaning and questions the usage of ERK inhibitors on melanoma.

There is not enough related information found on the other two genes AACS and VANGL2 from the same cluster to build up any theories about their relationship. However, they both seem to be related to breast cancer (Smith _et al_., 2019; Chou _et al_., 2013).

Another significant result is the discovery of the four upregulated genes that are only expressed under the sun. DAPL1, TGFBI are genes upregulated by the aforementioned MITF (Ma _et al_., 2018; Lavelle _et al_., 2020) and ARHGEF5 aids in the invasive potential of cancer cells (Kuroiwa _et al_., 2011). The knowledge that such genes, supporting the development of melanoma, have only been upregulated under sun exposure confirms the already commonly known fact that the exposure to sun is a trigger for skin cancer.

The fourth upregulated gene under sun exposure, ETV7, codes for a transcription factor, which is usually seen to be downregulated in melanoma patients and correlates with a poor prognosis (Qu _et al_., 2020). However, ETV7 is seen to be upregulated in our data under Trametinib, suggesting that it might have a role in attacking the tumor microenvironment, leading to a better prognosis. The fact that this gene is expressed under the sun suggests that it might be a natural response to the tumorigenic genes, upregulated under sun exposure. Trametinib helps upregulate it further to fight tumor proliferation. The relationship between ETV7 and the three genes mentioned above should be looked into with more detail, as a treatment upregulating its expression seems to be possibly effective against melanoma.

It is important to pay attention to the outcome that almost all of the upregulated genes are either regulated by MITF or are from the data that was exposed to the sun. This indicates that the high expression of the genes that are related to MITF are as effective as exposure to sun when it comes to helping create or supporting the melanoma. This means that MITF related genes make extremely good drug targets for melanoma. Since all of the upregulated genes analysed on are human skin specific TRA genes, a new treatment where the expression of the genes mentioned above are targeted, would not harm healthy tissues. This opens countless opportunities for future research and treatment of melanomas, especially because melanoma is highly metastatic.

Through the opposite expression of TYRP1, and varying up- and downregulated genes between chips of different treatments, the different effects of MEK and ERK inhibitors on melanoma have been shown. Even though these treatments in this project have shown to downregulate some known tumor inducing genes such as WEE1, an important regulator in the cell cycle (Tashnizi _et al_., 2016), it should be further researched and discussed on if these inhibitors should be used on melanoma, as they also showed to upregulate many genes inducing melanoma. If yes, a combination of other drugs are needed as treatment to inhibit the accelerating effects of MAPK inhibitors on melanosome related genes.

The strong linear dependency found between the expression of the gene DSP and MPZL3 also raises new questions requiring additional research. To be able to predict MPZL3 is important for characterizing the cancer; identifying it would help to see if the patient has a good prognosis or not. DSP seems to be fit for this prediction. It codes for a desmosomal protein, which helps human melanoma metastases to escape the immune system (Salerno _et al_., 2016). Some desmosomal proteins are used to identify melanomas, even when there are not any immune signatures seen (Salerno _et al_., 2016). 

The significant linear relationship between the genes ARHGEF5 and ANXA2 mentioned above might also be useful for treatment development as ANXA2 is an already widely known cancer biomarker (Wang _et al_., 2019). Considering ARHGEF5 strongly correlates with and represents almost all of the upregulated genes, all of them except for MPZL3 can be predicted by this already popular drug target. Besides the prediction of upregulated genes, PC genes can also be considered as potential drug targets, since they have shown a correlation between each other.

Not all of the upregulated genes found in this dataset were informative, as there has not been much research done on them before. More research is needed for further analysis. However,the findings that melanosome and MITF related genes, and genes highly expressed under sun exposure strongly influence the melanoma cells, and that there is a strong linear relationship between the upregulated and the high variance genes set a significant base for future research in new treatments for melanoma.




# LITERATURE

Alqathama A. (2020). BRAF in malignant melanoma progression and metastasis: potentials and challenges. 

American journal of cancer research, 10(4), 1103–1114.

Chou, HL., Yao, CT., Su, SL. _et al_. (2013).Gene expression profiling of breast cancer survivability by pooled cDNA microarray analysis using logistic regression, artificial neural networks and decision trees. BMC Bioinformatics 14, 100. https://doi.org/10.1186/1471-2105-14-100

Delevoye, C., Heiligenstein, X., Ripoll, L., Gilles-Marsens, F., Dennis, M. K., Linares, R. A., Derman, L., Gokhale, A., Morel, E., Faundez, V., Marks, M.S., Raposo, G. (2015). BLOC-1 Brings Together the Actin and Microtubule Cytoskeletons to Generate Recycling Endosomes, Current Biology, Vol 26(1), 1-13, https://doi.org/10.1016/j.cub.2015.11.020

Dinkelacker, 2007. A database of genes that are expressed in a tissue-restricted manner to analyse promiscuous gene expression in medullary thymic epithelial cells. Diplomarbeit, Albert-Ludwigs-Universitaet, Freiburg, Germany.

Dinkelacker, 2019. Chromosomal clustering of tissue restricted antigens, Dissertation, University Heidelberg, Germany.

Gruber, P., Zito, P. M. (2021). Jan, Skin Cancer, Treasure Island (FL): StatPearls Publishing; Available from: https://www.ncbi.nlm.nih.gov/books/NBK441949/

Klein, L., Kyewski, B., Allen, P. _et al_. (2014). Positive and negative selection of the T cell repertoire: what thymocytes see (and don't see). Nat Rev Immunol 14, 377–391. https://doi.org/10.1038/nri3667

Kobayashi, T., Imokawa, G., Bennett, D.C., and Hearing, V.J. (1998). Tyrosinase Stabilization by Tyrp1 (the brown Locus Protein). Journal of Biological Chemistry 273, 31801-31805. https://doi.org/10.1074/jbc.273.48.31801.

Komiya, Y., Onodera, Y., Kuroiwa, M., Nomimura, S., Kubo, Y., Nam, J. M., Kajiwara, K., Nada, S., Oneyama, C., Sabe, H., & Okada, M. (2016). The Rho guanine nucleotide exchange factor ARHGEF5 promotes tumor malignancy via epithelial-mesenchymal transition. Oncogenesis, 5(9), e258. https://doi.org/10.1038/oncsis.2016.59

Kuroiwa, M., Oneyama, C., Nada, S., & Okada, M. (2011). The guanine nucleotide exchange factor Arhgef5 plays crucial roles in Src-induced podosome formation. Journal of cell science, 124(Pt 10), 1726–1738. https://doi.org/10.1242/jcs.080291

Kyewski, B., Derbinski, J. (2004) Self-representation in the thymus: an extended view. Nat Rev Immunol 4, 688–698. https://doi.org/10.1038/nri1436

Lauden, L., Siewiera, J., Boukouaci, W., Ramgolam, K., Mourah, S., Lebbe, C., Charron, D., Aoudjit, F., Jabrane-Ferrat, N., Al-Daccak, R. (2014). TGF-ß-Induced (TGFBI) Protein in Melanoma: A Signature of High Metastatic Potential, Journal of Investigative Dermatology, Vol:134(6), 1675-1685, ISSN 0022-202X, https://doi.org/10.1038/jid.2014.20.

Lavelle, T. J., Alver, T. N., Heintz, K. M., Wernhoff, P., Nygaard, V., Nakken, S., Øy, G. F., Bøe, S. L., Urbanucci, A., & Hovig, E. (2020). Dysregulation of MITF Leads to Transformation in MC1R-Defective Melanocytes. Cancers, 12(7), 1719. https://doi.org/10.3390/cancers12071719	

Leiter, U., & Garbe, C. (2008). Epidemiology of melanoma and nonmelanoma skin cancer--the role of sunlight. Advances in experimental medicine and biology, 624, 89–103. https://doi.org/10.1007/978-0-387-77574-6_8

Linares, M. A., Zakaria, A., & Nizran, P. (2015). Skin Cancer. Primary care, 42(4), 645–659. https://doi.org/10.1016/j.pop.2015.07.006

Ma, X., Hua, J., Zheng, G., Li, F., Rao, C., Li, H., Wang, J., Pan, L., & Hou, L. (2018). Regulation of cell proliferation in the retinal pigment epithelium: Differential regulation of the death-associated protein like-1 DAPL1 by alternative MITF splice forms. Pigment cell & melanoma research, 31(3), 411–422. https://doi.org/10.1111/pcmr.12676

Miller, A. J., Du, J., Rowan, S., Hershey, C. L., Widlund, H. R., & Fisher, D. E. (2004). Transcriptional regulation of the melanoma prognostic marker melastatin (TRPM1) by MITF in melanocytes and melanoma. Cancer research, 64(2), 509–516. https://doi.org/10.1158/0008-5472.can-03-2440

Mohammadi, S., Ravindra, V., Gleich, D. F., & Grama, A. (2018). A geometric approach to characterize the functional identity of single cells. Nature communications, 9(1), 1516. https://doi.org/10.1038/s41467-018-03933-2

Noguchi, K., Dalton, A. C., Howley, B. V., McCall, B. J., Yoshida, A., Diehl, J. A., & Howe, P. H. (2017). Interleukin-like EMT inducer regulates partial phenotype switching in MITF-low melanoma cell lines. PloS one, 12(5), e0177830. https://doi.org/10.1371/journal.pone.0177830

Opitz, L., Salinas-Riester, G., Grade, M., Jung, K., Jo, P., Emons, G., Ghadimi, B. M., Beissbarth, T., & Gaedcke, J. (2010). Impact of RNA degradation on gene expression profiling. BMC medical genomics, 3, 36. https://doi.org/10.1186/1755-8794-3-36

Pérez, L. _et al_ (2017). Endothelial-to-mesenchymal transition: Cytokine-mediated pathways that determine endothelial fibrosis under inflammatory conditions. Cytokine & Growth Factor Reviews 33, 41-54. https://doi.org/10.1016/j.cytogfr.2016.09.002.

Potrony, M., Badenas, C., Aguilera, P., Puig-Butille, J. A., Carrera, C., Malvehy, J., Puig, S. (2015). Update in genetic susceptibility in melanoma. Annals of translational medicine, 3(15), 210.	 https://doi.org/10.3978/j.issn.2305-5839.2015.08.11

Qu, H., Zhao, H., Zhang, X., Liu, Y., Li, F., Sun, L., & Song, Z. (2020). Integrated Analysis of the ETS Family in Melanoma Reveals a Regulatory Role of ETV7 in the Immune Microenvironment. Frontiers in immunology, 11, 612784. https://doi.org/10.3389/fimmu.2020.612784

Ritchie, M. E., Phipson, B., Wu, D., Hu, Y., Law, C. W., Shi, W., & Smyth, G. K. (2015). limma powers differential expression analyses for RNA-sequencing and microarray studies. Nucleic acids research, 43(7), e47. https://doi.org/10.1093/nar/gkv007

Rose, A. A., Annis, M. G., Frederick, D. T., Biondini, M., Dong, Z., Kwong, L., Chin, L., Keler, T., Hawthorne, T., Watson, I. R., Flaherty, K. T., Siegel, P. M. (2016). MAPK Pathway Inhibitors Sensitize BRAF-Mutant Melanoma to an Antibody-Drug Conjugate Targeting GPNMB, DOI: 10.1158/1078-0432.CCR-16-1192 

Rosenberg S. A. (1999). A new era for cancer immunotherapy based on the genes that encode cancer antigens. Immunity, 10(3), 281–287. https://doi.org/10.1016/s1074-7613(00)80028-x

Salerno, E. P., Bedognetti, D., Mauldin, I. S., Deacon, D. H., Shea, S. M., Pinczewski, J., Obeid, J. M., Coukos, G., Wang, E., Gajewski, T. F., Marincola, F. M., & Slingluff, C. L., Jr (2016). Human melanomas and ovarian cancers overexpressing mechanical barrier molecule genes lack immune signatures and have increased patient mortality risk. Oncoimmunology, 5(12), e1240857. https://doi.org/10.1080/2162402X.2016.1240857

Setaluri, V., Jayanthy, A. (2013). Coat Color Mutations, Animals, Brenner's Encyclopedia of Genetics, Second Edition, Academic Press, 58-60, ISBN 9780080961569, https://doi.org/10.1016/B978-0-12-374984-0.00275-8.

Smith, P., Godde, N., Rubio, S. _et al_. (2019). VANGL2 regulates luminal epithelial organization and cell turnover in the mammary gland. Sci Rep 9, 7079.  https://doi.org/10.1038/s41598-019-43444-8

Smith, R. A., Mettlin, C. J., Eyre, H. (2003). Melanoma and Nonmelanoma Skin Cancer, Holland-Frei Cancer Medicine. 6th edition. Hamilton (ON): BC Decker;  https://www.ncbi.nlm.nih.gov/books/NBK13764/

Tashnizi, A. H., Jaberipour, M., Razmkhah, M., Rahnama, S., & Habibagahi, M. (2016). Tumour suppressive effects of WEE1 gene silencing in neuroblastomas. Journal of cancer research and therapeutics, 12(1), 221–227. https://doi.org/10.4103/0973-1482.165861

Taya, M., Hammes, S. R. (2018). Glycoprotein Non-Metastatic Melanoma Protein B (GPNMB) and Cancer: A Novel 
Potential Therapeutic Target. Steroids, 133, 102–107. https://doi.org/10.1016/j.steroids.2017.10.013

Volz, A., Korge, B. P., Compton, J. G., Ziegler, A., Steinert, P. M., & Mischke, D. (1993). Physical mapping of a functional cluster of epidermal differentiation genes on chromosome 1q21. Genomics, 18(1), 92–99. https://doi.org/10.1006/geno.1993.1430

Wang, T., Wang, Z., Niu, R., & Wang, L. (2019). Crucial role of Anxa2 in cancer progression: highlights on its novel regulatory mechanism. Cancer biology & medicine, 16(4), 671–687. https://doi.org/10.20892/j.issn.2095-3941.2019.0228

Wellbrock, C., & Arozarena, I. (2015). Microphthalmia-associated transcription factor in melanoma development and MAP-kinase pathway targeted therapy. Pigment cell & melanoma research, 28(4), 390–406. https://doi.org/10.1111/pcmr.12370

Zeiser, R. (2014). Trametinib. Recent results in cancer research. Fortschritte der Krebsforschung. Progrès dans les recherches sur le cancer, 201, 241–248. https://doi.org/10.1007/978-3-642-54490-3_15
