---
title: "The role of skin-specific genes in skin cancer"
author: "Denisa Neacsu, Binnur Ã–zay, Christofer Richard, Pinar Cavus"
date: "4/28/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 0. Introduction

The goal of this project is to find the expression of skin specific genes in skin cancer datasets. As datasets, we used microarray data (Affymetrix chips).


## 1. TRA Analysis

The first line below is from my computer personally, your home user name and file names are probably different in your computer!

```{r}
projectPath=dirname(rstudioapi::getSourceEditorContext()$path)
projectPath

#Import TRA Data
setwd(paste(projectPath, 'TRA Daten', sep = '/'))
TRA_Human_gtex=read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")
TRA_Mouse_4301=read.csv(file="tra.2014.mouse.4301.5x.table.tsv", sep="\t")
TRA_Human_roth=read.csv(file="tra.2014.human.roth.5x.table.tsv", sep="\t")
TRA_Human_Novartis=read.csv(file="tra.2014.human.5x.table.tsv", sep="\t")
TRA_Mouse_Novartis=read.csv(file="tra.2014.mouse.5x.table.tsv", sep="\t")
TRA_Human_protien_atlas=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep=",")
```

```{r}
#Skin-specific genes from TRA_Human_Novartis
tissue1=TRA_Human_Novartis[,11]
ind1=which(tissue1=="skin")
TRA.symbol1=TRA_Human_Novartis$gene.symbol

skin.TRA1=TRA.symbol1[ind1]
skin.TRA1
```

```{r}
#Skin-specific genes from TRA_Human_protein_atlas
tissue2=TRA_Human_protien_atlas[,11]
ind2=which(tissue2=="skin")
TRA.symbol2=TRA_Human_protien_atlas$Symbol

skin.TRA2=TRA.symbol2[ind2]
skin.TRA2
```

#TRA_Human_gtex for skin sun exposed

```{r}
#Skin-specific genes from TRA_Human_gtex (sun exposed)
tissue3.s=TRA_Human_gtex[,10]
ind3.s=which(tissue3.s=="Skin - Sun Exposed")
TRA.symbol3.s=TRA_Human_gtex$ensembl.symbol

skin.TRA3.s=TRA.symbol3.s[ind3.s]
skin.TRA3.s
```

#TRA_Human_gtex for skin not sun exposed

```{r}
#Skin-specific genes from TRA_Human_gtex (not sun exposed)
tissue3.ns=TRA_Human_gtex[,10]
ind3.ns=which(tissue3.ns=="Skin - Not Sun Exposed")
TRA.symbol3.ns=TRA_Human_gtex$ensembl.symbol

skin.TRA3.ns=TRA.symbol3.ns[ind3.ns]
skin.TRA3.ns
```

#We got the output "character(0)" here, but that is because this table actually doesn't have any data for skin tissue in it.

```{r}
#Skin-specific genes from TRA_Human_roth (no skin-specific genes)
tissue4=TRA_Human_roth[,11]
ind4=which(tissue4=="skin")
TRA.symbol4=TRA_Human_roth$gene.symbol

skin.TRA4=TRA.symbol4[ind4]
skin.TRA4
```

#For the two mouse datasets we couldn't find any tissue named "skin", but we did find tissue named "epidermis". I'm guessing that is because the tables show results of mice and not humans, so the researchers decided to name the sublayer of the skin itself.

```{r}
#Skin-specific genes from TRA_Mouse_4301 (epidermis-specific genes)
tissue5=TRA_Mouse_4301[,11]
ind5=which(tissue5=="epidermis")
TRA.symbol5=TRA_Mouse_4301$gene.symbol

skin.TRA5=TRA.symbol5[ind5]
skin.TRA5
```

```{r}
#Skin-specific genes from TRA_Mouse_Novartis (epidermis-specific genes)
tissue6=TRA_Mouse_Novartis[,11]
ind6=which(tissue6=="epidermis")
TRA.symbol6=TRA_Mouse_Novartis$gene.symbol

skin.TRA6=TRA.symbol6[ind6]
skin.TRA6
```

#Vectors for all human skin- and mouse epidermis-specific genes

```{r}
#Creating vector with all the human skin genes
skin.genes.human = unique(c(skin.TRA1, skin.TRA2, skin.TRA3.ns, skin.TRA3.s))
skin.genes.human
length(skin.genes.human)

#Removing NAs
vector = lapply(skin.genes.human,function(x){sum(is.na(x))}) #sum up all missing values
vector
which(vector>0) # display the missing value
# no.239, that means 1 NA
skin.genes.human = skin.genes.human[-which(vector>0)] 
rm(vector)

length(skin.genes.human)
#646

which(is.na(skin.genes.human)) # Verification that the NA was removed


#Creating vector with all the mouse skin (epidermis) genes
skin.genes.mouse = c(skin.TRA5, skin.TRA6)
skin.genes.mouse
length(skin.genes.mouse)
```

#Creating .csv files for both human skin genes and mouse skin genes

```{r}
#The data is unfiltered, meaning no imputation of missing values was applied. We observed numerous NA in both human and mouse skin genes

#Putting the tables in a tables folder
setwd(paste(projectPath, 'tables', sep = '/')) #
#Using write.csv to create the files
write.csv(matrix(skin.genes.human), file = "skin.specific.genes.human.TRA", row.names = F, sep = "\t")
write.csv(matrix(skin.genes.mouse), file = "skin.specific.genes.mouse.TRA", row.names = F, sep = "\t")

#If write.cvs doesn't work for you you can try to use write.table. 
```

##2. Quality control of microarray data

#2.1 Analysis of skin cancer data set

```{r}
#Loading the necessary libraries
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(stringr)

standardPath <- .libPaths()
standardPath
```

```{r}
#Setting directory for microarray data
setwd(paste(projectPath, 'rawdata/GSE57721 melanoma', sep = '/'))

#Read all 15 microarray chips
data.skin.cancer = ReadAffy()
data.skin.cancer@cdfName <- "HGU133Plus2_Hs_ENST"
data.skin.cancer

orig = colnames(exprs(data.skin.cancer))
new.name = substr(orig, 1, nchar(orig)-4)
colnames(exprs(data.skin.cancer)) = new.name

setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "data.skin.cancer.rda")

```

#2.1.1. Single chip control

```{r}
#Visualizing raw microarray data
image(data.skin.cancer,col=rainbow(100,start = 0, end = 0.75) [100:1])

##All of the chips seem to have a very good quality, so we can use all of them
```


#3. Normalization of microarray data

```{r}
#Normalization is used to remove systematic variation that can affect the analysis of the chips
#Still in work

data.skin.cancer.norm = vsnrma(data.skin.cancer)
data.skin.cancer.norm

setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "normalized_data.rda")
```

#4.1. meanSdPlot

```{r}
#Plotting meanSdPlot
meanSdPlot(data.skin.cancer.norm)

setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "meanSdPlot_data_skin_cancer_normalized.eps")
```



#4.3. Boxplots of pre-normalized and normalized data

```{r}
#Before normalization
  ## Install the newest version of XQuartz from https://www.xquartz.org/ in your computer
    ### We would recommend to have the XQuartz application open while running the code
x11(w=10, h=7)

par(las=2)
mmi = c(1.8, 0.7, 1.0477939, 0.5366749)
par(mai=mmi)

boxplot(data.skin.cancer, col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer before normalization")

setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_rawdata.eps")


#After normalization
x11(w=10, h=7)

par(las=2)
mmi = c(1.8, 0.7, 1.0477939, 0.5366749)
par(mai=mmi)

boxplot(exprs(data.skin.cancer.norm), col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer after normalization") #Here use exprs(), otherwise there will be an error "x must be atomic"

setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_normalized.eps")
```



#4.5. RNA degradation 
```{r}
rna_degradation = AffyRNAdeg(data.skin.cancer)

# RNA degradation plot shifted and scaled
plotAffyRNAdeg(rna_degradation, col = rainbow(150))
title(sub="Human skin cancer raw")
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "RNA_degradation_data_skin_cancer_raw.eps")

#RNA degradation plot only shifted
plotAffyRNAdeg(rna_degradation, col = rainbow(150), transform = "shift.only")
title(sub = "Human skin cancer raw")

setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file= "RNA_deg_data_skin_cancer_raw.eps")

summaryAffyRNAdeg(rna_degradation) # According to the values of the slopes, which have similar values, it seems that all the chips should be considered in our analysis

## According to the plots, there is no obvious deviation, no line stands out; all of the chips have high expression intensity
## The chips are qualitative
```

#4.6. Histogram
```{r}
## Histogram before data normalization
x11(w=10, h=7)
hist(data.skin.cancer, col = rainbow(150), main = "Histogram data pre-normalized")
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data pre-normalized")

## Histogram after data normalization
for (i in 1:ncol(exprs(data.skin.cancer.norm))) if (i == 1) plot(density(log2(exprs(data.skin.cancer.norm)[, i])), col = "red") else lines(density(log2(exprs(data.skin.cancer.norm)[, i])), col = "red")

     
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data normalized.eps")
```

#4.7 Heatmap for the purpose of correlation visualization
```{r}
## Correlation matrix of the normalized data
cor.mat = cor(exprs(data.skin.cancer.norm), method="spearman")

## Heatmap 
library(pheatmap)
pheatmap(cor.mat)

setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file =  "Heatmap correlation chips.eps")
```

#4.2 Creating the gene expression data frame
```{r}
## extract expression values
Expression=exprs(data.skin.cancer.norm)
Expression
dim(Expression)
# 95721    15

## Select only ENST IDs, which represent the transcript IDs
enst_id = str_detect(rownames(Expression),"ENST")
Expression=Expression[enst_id,]
Expression
dim(Expression)
# 95659    15


## remove .x_at suffix by using \\..*, which basically means removing every character after the dot "."
rownames(Expression) = sub("\\..*","",rownames(Expression)) 
dim(Expression)
Expression

## Remove .CEL suffix from each column name
colnames(Expression) = sub("\\..*","", colnames(Expression))
Expression

## load ensemble IDs and gene symbols from https://www.ensembl.org/biomart/martview/74741c6cba3dea58c83e6520f12fa790
## Create the ensembl.103.txt file using BioMart
## Here beware of Transcript.stable.ID, Chromosome, AFFY HG U133 Plus 2 probe and HGNC symbol, as we need them to visualize the gene data frame

setwd(paste(projectPath, 'tables', sep = '/'))
Ensembl = read.csv(paste(projectPath, 'tables/ensembl.103.txt', sep = '/'), sep = "\t")
head(Ensembl)

## Extracting the transcript IDs and the HGNC symbols (or gene symbols) from the ensembl data frame
transcriptIDs = Ensembl$Transcript.stable.ID
HGNC_symbol = Ensembl$HGNC.symbol

## Assigning every gene symbol to a transcript ID
names(HGNC_symbol) = transcriptIDs
HGNC_symbol

## replace ensemble IDs with HGNC symbol
chipIDs = rownames(Expression)
noMatchIDs = chipIDs[!chipIDs %in% transcriptIDs] # transcript IDs that are not in ensemble.103.txt
length(noMatchIDs)
# 112 
newChipIDs = chipIDs[chipIDs %in% transcriptIDs] # only transcript IDs that are in ensemble.103.txt (95659 - 112 = 95547)
Expression = Expression[newChipIDs, ]
dim(Expression)
# 95547    15

symbol = HGNC_symbol[newChipIDs]
rownames(Expression) = as.character(symbol)
head(Expression)



### Cleaning the Expression data frame

## Checking for NAs
length(which(is.na(Expression)))
# 0 

## Checking for ""
length(which(rownames(Expression)==""))
#1096

## Removing "" (spaces)
all.spaces.exp = sapply(rownames(Expression),function(x){which(x=="")}) #sum up all spaces ""

which(all.spaces.exp>0) # display the spaces ""

Expression = Expression[-which(all.spaces.exp>0),] 
rm(all.spaces.exp)
dim(Expression)
#94451 15
```

#4.4. Skin-specific antigens extraction from gene expression df
##Checking to see how much skin specific TRAs are expressed in the melanoma data set
```{r}
skin.genes.human.expression = which(rownames(Expression)%in%skin.genes.human)
data.skin.cancer.human_skin_TRA = Expression[skin.genes.human.expression,]
y = sort(rownames(data.skin.cancer.human_skin_TRA))
unique(y[y !=""])
data.skin.cancer.human_skin_TRA = data.skin.cancer.human_skin_TRA[unique(y[y !=""]),]
dim(data.skin.cancer.human_skin_TRA)
#512  15
head(data.skin.cancer.human_skin_TRA)
```

#5. Analysis

```{r}
#Heatmap (Visualization)
library(RColorBrewer)
coul = colorRampPalette(brewer.pal(8, "PiYG"))(25)
heatmap(data.skin.cancer.human_skin_TRA, Rowv = NULL, Colv = NULL, col = coul)
```

```{r}
#Barplot (Localisation in chromosomes)
##Get all the chromosome numbers for all human skin TRAs
chromosome_1 = TRA_Human_Novartis[,c(3,7)]
chromosome_skin1 = as.matrix(chromosome_1[ind1,])

chromosome_2 = TRA_Human_protien_atlas[,c(6,2)]
chromosome_skin2 = as.matrix(chromosome_2[ind2,])

chromosome_3 = TRA_Human_gtex[,c(3,6)]
chromosome_skin3 = as.matrix(chromosome_3[c(ind3.s, ind3.ns),])

##Combine all 3 tables and remove duplicates and empty values to make a clean list of chromosome numbers
chromosome_gene_table = rbind(chromosome_skin1, chromosome_skin2, chromosome_skin3)
rownames(chromosome_gene_table) = c()
head(chromosome_gene_table)

a = sort(chromosome_gene_table[,1])

rownames(chromosome_gene_table) = chromosome_gene_table[,1]
chromosome_gene_table = chromosome_gene_table[,2]

chromosome_list = chromosome_gene_table[unique(a[a !=""])]

##Count the number of genes on each chromosome
localization = table(chromosome_list)
localization = as.matrix(localization)

sorted_chromosomes = str_sort(rownames(localization), numeric = T)
chromosome_counts = localization[sorted_chromosomes,]

##Make barplot
chromosome = c(1:22, "X")

barplot(chromosome_counts, main = "Distribution of skin-specific TRAs over chromosomes", xlab = "Chromosomes", ylab = "Number of skin-specific TRAs", names.arg = chromosome, cex.names = 0.7, col = "darkred")
```

```{r}
#Piechart (skin TRAs)

#Creating small vectors for each table containing only NON-skin-specific TRAs
#TRA_Human_Novartis
bla1 = which(tissue1!="skin")
TRA.symbolA=TRA_Human_Novartis$gene.symbol
non.skin.TRA1=TRA.symbolA[bla1]
non.skin.TRA1

#TRA_Human_protein_atlas
bla2 = which(tissue2!="skin")
TRA.symbolB=TRA_Human_protien_atlas$Symbol
non.skin.TRA2=TRA.symbolB[bla2]
non.skin.TRA2

#TRA_Human_gtex (sun exposed)
bla3.s = which(tissue3.s!="skin")
TRA.symbolC=TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.s=TRA.symbolC[bla3.s]
non.skin.TRA3.s

#TRA_Human_gtex (not sun exposed)
bla3.ns = which(tissue3.ns!="skin")
TRA.symbolD=TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.ns=TRA.symbol3.ns[bla3.ns]
non.skin.TRA3.ns


# Putting all the small non-skin-specific TRA vectors in a general vector containing all of them
NON_skin_specific_TRA = unique(c(non.skin.TRA1, non.skin.TRA2, non.skin.TRA3.s, non.skin.TRA3.ns))
length(NON_skin_specific_TRA)

# Calculating the total number of TRA
total = length(skin.genes.human) + length(NON_skin_specific_TRA)
total
prozent = round(length(skin.genes.human)/total*100,2)
prozent 
prozent_2 = round(length(NON_skin_specific_TRA)/total*100,2)
prozent_2
labels = c("Skin-specific TRAs", "Other TRAs")
labels = paste(labels, c(prozent, prozent_2)) # add percents to labels
labels = paste(labels,"%",sep="") # ad % to labels

# install.packages("RColorBrewer")
library(RColorBrewer)

color <- brewer.pal(length(labels), "Set3") 

pie( c(prozent, prozent_2),labels = labels, col = color,
   main="Pie Chart skin specific TRAs")


dev.copy2eps(file="Piechart_skin_TRA.eps") 
```

```{r}
#Venn diagram (Upregulated genes that are also TRAs)

```

```{r}
#Boxplots(Gene expression after different treatments after 3 hours)

#Boxplots(Gene expression after different treatments after 24 hours)

#Boxplots(Gene expression after with or without sun exposure)

```


```{r}
#Clustering
##k-means
km = kmeans(x = t(data_topVar), centers = 4, nstart = 15)
km$cluster

##Elbow method
###Picking the most variable genes
topVar = apply(data.skin.cancer.human_skin_TRA, 1, var)
data_topVar = data.skin.cancer.human_skin_TRA[topVar > quantile(topVar, probs = 0.8),]

###Plot graph for WSS
wss = sapply(2:10, function(k){kmeans(x = t(data_topVar), centers = k)$tot.withinss})
plot(2:10, wss, type = 'b', pch = 19, xlab = "Number of clusters", ylab ="WSS")
##Optimal number of clusters = 3 or 4 (more to 3(~70%), but sometimes also 4)

##Silhouette method
###Computing the pairwise distances of all patients
paired_d = dist(t(data_topVar))

###Plot graph
library(cluster)
plot(silhouette(km$cluster, paired_d))


##3 Clusters: 1) sherk D3 & D7    2)erk & trametinib 3h   3)erk & trametinib 24h
##4 Clusters: 1) sherk D3     2)sherk D7    3)erk & trametinib 3h   4)erk & trametinib 24h
```


```{r}
#Principal component analysis

```


```{r}
#Linear regression model (Predict drug target genes with identified genes)

#F-test

```

