---
title: "The role of skin-specific genes in skin cancer"
author: "Denisa Neacsu, Binnur Ã–zay, Christofer Richard, Pinar Cavus"
date: "4/28/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 0. Introduction

The goal of this project is to find the expression of skin specific genes in skin cancer datasets. As datasets, we used microarray data (Affymetrix chips).


## 1. TRA Analysis

The first line below is from my computer personally, your home user name and file names are probably different in your computer!

```{r}
setwd("/Users/christofer.richard/Desktop/DA 2.3/TRA Daten/")

#Import TRA Data
TRA_Human_gtex=read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")
TRA_Mouse_4301=read.csv(file="tra.2014.mouse.4301.5x.table.tsv", sep="\t")
TRA_Human_roth=read.csv(file="tra.2014.human.roth.5x.table.tsv", sep="\t")
TRA_Human_Novartis=read.csv(file="tra.2014.human.5x.table.tsv", sep="\t")
TRA_Mouse_Novartis=read.csv(file="tra.2014.mouse.5x.table.tsv", sep="\t")
TRA_Human_protien_atlas=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep=",")
```

```{r}
#Skin-specific genes from TRA_Human_Novartis
tissue1=TRA_Human_Novartis[,11]
ind1=which(tissue1=="skin")
TRA.symbol1=TRA_Human_Novartis$gene.symbol

skin.TRA1=TRA.symbol1[ind1]
skin.TRA1
```

```{r}
#Skin-specific genes from TRA_Human_protein_atlas
tissue2=TRA_Human_protien_atlas[,11]
ind2=which(tissue2=="skin")
TRA.symbol2=TRA_Human_protien_atlas$Symbol

skin.TRA2=TRA.symbol2[ind2]
skin.TRA2
```

#TRA_Human_gtex for skin sun exposed

```{r}
#Skin-specific genes from TRA_Human_gtex (sun exposed)
tissue3.s=TRA_Human_gtex[,10]
ind3.s=which(tissue3.s=="Skin - Sun Exposed")
TRA.symbol3.s=TRA_Human_gtex$ensembl.symbol

skin.TRA3.s=TRA.symbol3.s[ind3.s]
skin.TRA3.s
```

#TRA_Human_gtex for skin not sun exposed

```{r}
#Skin-specific genes from TRA_Human_gtex (not sun exposed)
tissue3.ns=TRA_Human_gtex[,10]
ind3.ns=which(tissue3.ns=="Skin - Not Sun Exposed")
TRA.symbol3.ns=TRA_Human_gtex$ensembl.symbol

skin.TRA3.ns=TRA.symbol3.ns[ind3.ns]
skin.TRA3.ns
```

#We got the output "character(0)" here, but that is because this table actually doesn't have any data for skin tissue in it.

```{r}
#Skin-specific genes from TRA_Human_roth (no skin-specific genes)
tissue4=TRA_Human_roth[,11]
ind4=which(tissue4=="skin")
TRA.symbol4=TRA_Human_roth$gene.symbol

skin.TRA4=TRA.symbol4[ind4]
skin.TRA4
```

#For the two mouse datasets we couldn't find any tissue named "skin", but we did find tissue named "epidermis". I'm guessing that is because the tables show results of mice and not humans, so the researchers decided to name the sublayer of the skin itself.

```{r}
#Skin-specific genes from TRA_Mouse_4301 (epidermis-specific genes)
tissue5=TRA_Mouse_4301[,11]
ind5=which(tissue5=="epidermis")
TRA.symbol5=TRA_Mouse_4301$gene.symbol

skin.TRA5=TRA.symbol5[ind5]
skin.TRA5
```

```{r}
#Skin-specific genes from TRA_Mouse_Novartis (epidermis-specific genes)
tissue6=TRA_Mouse_Novartis[,11]
ind6=which(tissue6=="epidermis")
TRA.symbol6=TRA_Mouse_Novartis$gene.symbol

skin.TRA6=TRA.symbol6[ind6]
skin.TRA6
```

#Vectors for all human skin- and mouse epidermis-specific genes

```{r}
#Creating vector with all the human skin genes
skin.genes.human = c(skin.TRA1, skin.TRA2, skin.TRA3.ns, skin.TRA3.s)
skin.genes.human
length(skin.genes.human)

#Creating vector with all the mouse skin (epidermis) genes
skin.genes.mouse = c(skin.TRA5, skin.TRA6)
skin.genes.mouse
length(skin.genes.mouse)
```

#Creating .csv files for both human skin genes and mouse skin genes

```{r}
#The data is unfiltered, meaning no imputation of missing values was applied. We observed numerous NA in both human and mouse skin genes

#Putting the tables in a tables folder
setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/tables/") #Please change the directory according to your PC
#Using write.csv to create the files
write.csv(matrix(skin.genes.human), file = "skin.specific.genes.human.TRA", row.names = F, sep = "\t")
write.csv(matrix(skin.genes.mouse), file = "skin.specific.genes.mouse.TRA", row.names = F, sep = "\t")

#If write.cvs doesn't work for you you can try to use write.table. 
```

##2. Quality control of microarray data

#2.1 Analysis of skin cancer data set

```{r}
#Loading the necessary libraries
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(stringr)

standardPath <- .libPaths()
standardPath
```
```{r}
#Setting directory for microarray data
setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/rawdata/GSE57721 melanoma/")

#Read all 15 microarray chips
data.skin.cancer = ReadAffy()
data.skin.cancer@cdfName <- "HGU133Plus2_Hs_ENST"
data.skin.cancer

orig = colnames(exprs(data.skin.cancer))
new.name = substr(orig, 1, nchar(orig)-4)
colnames(exprs(data.skin.cancer)) = new.name

setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/sessions/rda")
save.image(file = "data.skin.cancer.rda")

```

#2.1.1. Single chip control

```{r}
#Visualizing raw microarray data
image(data.skin.cancer,col=rainbow(100,start = 0, end = 0.75) [100:1])

##All of the chips seem to have a very good quality, so we can use all of them
```


#3. Normalization of microarray data

```{r}
#Normalization is used to remove systematic variation that can affect the analysis of the chips
#Still in work

data.skin.cancer.norm = vsnrma(data.skin.cancer)
data.skin.cancer.norm

setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/sessions/rda")
save.image(file = "normalized_data.rda")
```

#4.1. meanSdPlot

```{r}
#Plotting meanSdPlot
meanSdPlot(data.skin.cancer.norm)

setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file = "meanSdPlot_data_skin_cancer_normalized.eps")
```

#4.2 Creating the gene expression data frame
```{r}
## extract expression values
Expression=exprs(data.skin.cancer.norm)
Expression
dim(Expression)
# 95721    15

## Select only ENST IDs, which represent the transcript IDs
enst_id = str_detect(rownames(Expression),"ENST")
Expression=Expression[enst_id,]
Expression
dim(Expression)
# 95659    15


## remove .x_at suffix by using \\..*, which basically means removing every character after the dot "."
rownames(Expression) = sub("\\..*","",rownames(Expression)) 
dim(Expression)
Expression

## Remove .CEL suffix from each column name
colnames(Expression) = sub("\\..*","", colnames(Expression))
Expression

## load ensemble IDs and gene symbols from https://www.ensembl.org/biomart/martview/74741c6cba3dea58c83e6520f12fa790
## Create the ensembl.103.txt file using BioMart
## Here beware of Transcript.stable.ID, Chromosome, AFFY HG U133 Plus 2 probe and HGNC symbol, as we need them to visualize the gene data frame

setwd("/Users/denisaneacsu/Documents/Bioinfo SS21/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/tables") #Directory please
Ensembl = read.csv("ensembl.103.txt", sep = ",") #Separation by comma! otherwise error
head(Ensembl)

## Extracting the transcript IDs and the HGNC symbols (or gene symbols) from the ensembl data frame
transcriptIDs = Ensembl$Transcript.stable.ID
HGNC_symbol = Ensembl$HGNC.symbol

## Assigning every gene symbol to a transcript ID
names(HGNC_symbol) = transcriptIDs
HGNC_symbol

## replace ensemble IDs with HGNC symbol
chipIDs = rownames(Expression)
noMatchIDs = chipIDs[!chipIDs %in% transcriptIDs] # transcript IDs that are not in ensemble.103.txt
length(noMatchIDs)
# 112 
newChipIDs = chipIDs[chipIDs %in% transcriptIDs] # only transcript IDs that are in ensemble.103.txt (95659 - 112 = 95547)
Expression = Expression[newChipIDs, ]
dim(Expression)
# 95547    15

symbol = HGNC_symbol[newChipIDs]
rownames(Expression) = as.character(symbol)
head(Expression)


```

#4.3. Boxplots of pre-normalized and normalized data

```{r}
#Before normalization
  ## Install the newest version of XQuartz from https://www.xquartz.org/ in your computer
    ### We would recommend to have the XQuartz application open while running the code
x11(w=10, h=7)

par(las=2)

mmi = c(1.8, 0.7, 1.0477939, 0.5366749)
par(mai=mmi)

boxplot(data.skin.cancer, col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer before normalization")

setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file = "boxplot_skin_cancer_rawdata.eps")


#After normalization
x11(w=10, h=7)

par(las=2)

mmi = c(1.8, 0.7, 1.0477939, 0.5366749)
par(mai=mmi)

boxplot(exprs(data.skin.cancer.norm), col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer after normalization") #Here use exprs(), otherwise there will be an error "x must be atomic"

setwd("/Users/christofer.richard/Desktop/DA 2.3/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file = "boxplot_skin_cancer_normalized.eps")
```

#4.4. Skin-specific antigens extraction from gene expression df
##Checking to see how much skin specific TRAs are expressed in the melanoma data set
```{r}
skin.genes.human.expression = which(rownames(Expression)%in%skin.genes.human)
data_human_skin_specific_TRA_affy = Expression[skin.genes.human.expression,]
y = sort(rownames(data_human_skin_specific_TRA_affy))
unique(y[y !=""])
data_human_skin_specific_TRA_affy = data_human_skin_specific_TRA_affy[unique(y[y !=""]),]
dim(data_human_skin_specific_TRA_affy)
#512  15
head(data_human_skin_specific_TRA_affy)

```


#4.5. RNA degradation 
```{r}
rna_degradation = AffyRNAdeg(data.skin.cancer)

# RNA degradation plot shifted and scaled
plotAffyRNAdeg(rna_degradation, col = rainbow(150))
title(sub="Human skin cancer raw")
setwd("/Users/denisaneacsu/Documents/Bioinfo SS21/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file = "RNA_degradation_data_skin_cancer_raw.eps")

#RNA degradation plot only shifted
plotAffyRNAdeg(rna_degradation, col = rainbow(150), transform = "shift.only")
title(sub = "Human skin cancer raw")

setwd("/Users/denisaneacsu/Documents/Bioinfo SS21/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file= "RNA_deg_data_skin_cancer_raw.eps")

summaryAffyRNAdeg(rna_degradation) # According to the values of the slopes, which have similar values, it seems that all the chips should be considered in our analysis

## According to the plots, there is no obvious deviation, no line stands out; all of the chips have high expression intensity
## The chips are qualitative
```

#4.6. Histogram
```{r}
## Histogram before data normalization
x11(w=10, h=7)
hist(data.skin.cancer, col = rainbow(150), main = "Histogram data pre-normalized")
setwd("/Users/denisaneacsu/Documents/Bioinfo SS21/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file = "Histogram data pre-normalized")

## Histogram after data normalization
plot(density(exprs(data.skin.cancer.norm)[,1]), type = "n", xlab = "log intensity", ylim = c(0,1), main = "Histogram data normalized")
for (i in 1:ncol(exprs(data.skin.cancer.norm))) if (i == 1) plot(density(log2(exprs(data.skin.cancer.norm)[, i])), col = "blue") else lines(density(log2(exprs(data.skin.cancer.norm)[, i])), col = "red")



     
setwd("/Users/denisaneacsu/Documents/Bioinfo SS21/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file = "Histogram data normalized.eps")
```

#4.7 Heatmap for the purpose of correlation visualization
```{r}
## Correlation matrix of the normalized data
cor.mat = cor(exprs(data.skin.cancer.norm), method="spearman")

## Heatmap 
library(pheatmap)
pheatmap(cor.mat)

setwd("/Users/denisaneacsu/Documents/Bioinfo SS21/Bioinformatik TRA Projekt/The role of skin specific genes in skin cancer/plots")
dev.copy2eps(file =  "Heatmap correlation chips.eps")
```


