---
title: "The Role of Skin-Specific TRAs on Skin Cancer"
author: "Binnur Ozay, Christofer Richard, Denisa Neacsu, Pinar Cavus"
date: "7/15/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

The goal of this project is to find the expression of skin specific genes in skin cancer datasets. As datasets, we used microarray data (Affymetrix chips).

# 1. TRA Analysis
## 1.1. Selection of skin-specific TRAs

Six TRA datasets were used to find skin-specific genes for humans and mice.

```{r}
#Define work directory
projectPath = dirname(rstudioapi::getSourceEditorContext()$path)

#Import TRA Data
setwd(paste(projectPath, 'TRA Daten', sep = '/'))
TRA_Human_gtex = read.csv(file = "tra.2017.human.gtex.5x.table.tsv", sep = "\t")
TRA_Mouse_4301 = read.csv(file = "tra.2014.mouse.4301.5x.table.tsv", sep = "\t")
TRA_Human_roth = read.csv(file = "tra.2014.human.roth.5x.table.tsv", sep = "\t")
TRA_Human_Novartis = read.csv(file = "tra.2014.human.5x.table.tsv", sep = "\t")
TRA_Mouse_Novartis = read.csv(file = "tra.2014.mouse.5x.table.tsv", sep = "\t")
TRA_Human_protien_atlas = read.csv(file = "Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep = ",")

#Skin-specific genes from TRA_Human_Novartis
tissue1 = TRA_Human_Novartis[,11]
ind1 = which(tissue1 == "skin")
TRA.symbol1 = TRA_Human_Novartis$gene.symbol
skin.TRA1 = TRA.symbol1[ind1]

#Skin-specific genes from TRA_Human_protein_atlas
tissue2 = TRA_Human_protien_atlas[,11]
ind2 = which(tissue2 == "skin")
TRA.symbol2 = TRA_Human_protien_atlas$Symbol
skin.TRA2 = TRA.symbol2[ind2]

#Skin-specific genes from TRA_Human_gtex (sun exposed)
tissue3.s = TRA_Human_gtex[,10]
ind3.s = which(tissue3.s == "Skin - Sun Exposed")
TRA.symbol3.s = TRA_Human_gtex$ensembl.symbol
skin.TRA3.s = TRA.symbol3.s[ind3.s]

#Skin-specific genes from TRA_Human_gtex (not sun exposed)
tissue3.ns = TRA_Human_gtex[,10]
ind3.ns = which(tissue3.ns == "Skin - Not Sun Exposed")
TRA.symbol3.ns = TRA_Human_gtex$ensembl.symbol
skin.TRA3.ns = TRA.symbol3.ns[ind3.ns]

#Skin-specific genes from TRA_Human_roth (no skin-specific genes)
tissue4 = TRA_Human_roth[,11]
ind4 = which(tissue4 == "skin")
TRA.symbol4 = TRA_Human_roth$gene.symbol
skin.TRA4 = TRA.symbol4[ind4]

#Skin-specific genes from TRA_Mouse_4301 (epidermis-specific genes)
tissue5 = TRA_Mouse_4301[,11]
ind5 = which(tissue5 == "epidermis")
TRA.symbol5 = TRA_Mouse_4301$gene.symbol
skin.TRA5 = TRA.symbol5[ind5]

#Skin-specific genes from TRA_Mouse_Novartis (epidermis-specific genes)
tissue6 = TRA_Mouse_Novartis[,11]
ind6 = which(tissue6 == "epidermis")
TRA.symbol6 = TRA_Mouse_Novartis$gene.symbol
skin.TRA6 = TRA.symbol6[ind6]
```

## 1.2. Compilation of all skin-specific TRAs

Skin-specific TRAs for humans were put into a single vector and data was cleaned up.

```{r}
#Creating vector with all the human skin genes
skin.genes.human = unique(c(skin.TRA1, skin.TRA2, skin.TRA3.ns, skin.TRA3.s))

#Removing NAs and empty values
skin.genes.human = skin.genes.human[-which(is.na(skin.genes.human))] 
skin.genes.human = skin.genes.human[-which(skin.genes.human == "")] 

#Verification that all NA and empty values were removed
which(is.na(skin.genes.human)) 
which(skin.genes.human == "")

#Saving this vector with all skin-specific human TRAs in a csv file
setwd(paste(projectPath, 'tables', sep = '/'))
write.csv(matrix(skin.genes.human), file = "skin.specific.genes.human.TRA", row.names = F, sep = "\t")
```

#2. Quality control of microarray data of skin cancer

Microarray data was taken from the GEO website, series number GSE57721. Here, gene expression of the WM266-4 BRAF V600D mutant melanoma cell line was measured on different time points after treatment with several inhibitors, such as Doxycycline-shERK1, ERK1/2 inhibitor and trametinib. The Affymetrix Human Genome U133 Plus 2.0 Array was used and the package for annotation of the chips was downloaded from the Brainarray website.

##2.1. Loading the skin cancer dataset

15 chips were chosen to be analysed and were downloaded from the GEO website. The 15 chips that were chosen to be used are: GSM1387513, GSM1387514, GSM1387515, GSM1387525, GSM1387527, GSM1387537, GSM1387538, GSM1387544, GSM1387545, GSM1387552, GSM1387553, GSM1387554, GSM1387558, GSM1387559 and GSM1387560.

```{r}
#Loading the necessary libraries
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(stringr)

#Read all 15 microarray chips
setwd(paste(projectPath, 'rawdata/GSE57721 melanoma', sep = '/'))
data.skin.cancer = ReadAffy()
data.skin.cancer@cdfName = "HGU133Plus2_Hs_ENST"

orig = colnames(exprs(data.skin.cancer))
new.name = substr(orig, 1, nchar(orig)-4)
colnames(exprs(data.skin.cancer)) = new.name

#Saving raw data as a .rda file
setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "data.skin.cancer.rda")
```

## 2.2. Single chip control

Qualitative quality control by looking at the chips was done by visualizing every single chip.

```{r}
#Visualizing raw microarray data
par(mfrow = c(3,5))
image(data.skin.cancer,col=rainbow(100,start = 0, end = 0.75) [100:1]) #All of the chips seem to have a very good quality, so we can use all of them
```

## 2.3. Normalization of microarray data

Normalization is used to remove systematic variation that can affect the analysis of the chips. The microarray data for skin cancer was normalized with the vsn package. The values after normalization are glog2-transformed.

```{r}
#Data normalization
data.skin.cancer.norm = vsnrma(data.skin.cancer)

#Saving normalized microarray data as a .rda file
setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "normalized_data.rda")
```

## 2.4. meanSd Plot

A meanSd plot was made to show the relation between the mean and variance of the data. For good quality chips, we need a relatively horizontal line (no relation between mean and variance).

```{r}
#Plotting meanSdPlot
meanSdPlot(data.skin.cancer.norm)

#Saving meanSd plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "meanSdPlot_data_skin_cancer_normalized.eps")
```

## 2.5. Boxplots of pre-normalized and normalized data

Boxplots before and after normalization had to be compared to show that the normalization worked and that the distribution of all chips are similar.

```{r}
#Before normalization
par(las = 2)
mmi = c(1.8, 0.7, 1.05, 0.54)
par(mai = mmi)
boxplot(data.skin.cancer, col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer before normalization")

#Saving boxplot with data before normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_rawdata.eps")

#After normalization
par(las = 2, mai = mmi)
boxplot(exprs(data.skin.cancer.norm), col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer after normalization")

#Saving boxplot with data after normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_normalized.eps")
```

## 2.6. RNA degradation

RNA degradation plots were also made to ensure that the RNAs used on the chips are of good quality. Good quality chips show a similar trend to each other and are relatively flat on the only-shifted plot.

```{r}
# RNA degradation plot shifted and scaled
rna_degradation = AffyRNAdeg(data.skin.cancer)

plotAffyRNAdeg(rna_degradation, col = rainbow(150))
title(sub="Human skin cancer raw")

#Saving shifted and scaled RNA degradation plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "RNA_degradation_data_skin_cancer_raw.eps")

#RNA degradation plot only shifted
plotAffyRNAdeg(rna_degradation, col = rainbow(150), transform = "shift.only")
title(sub = "Human skin cancer raw")

#Saving only-shifted RNA degradation plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file= "RNA_deg_data_skin_cancer_raw.eps")

#Analysis of values and plots from RNA degradation
summaryAffyRNAdeg(rna_degradation) # The values of the slopes are similar, so all of the chips are usable for further analysis

## According to the plots, there is no obvious deviation, no line stands out; all of the chips have high expression intensity. In otherwords, all chips are of high quality.
```

## 2.7. Histogram

Histograms before and after normalization were made to visualize the change in expression after normalization. Each line of the histogram presents the data from each microarray chip.

```{r}
#Histogram before data normalization
hist(data.skin.cancer, col = rainbow(150), main = "Histogram data pre-normalized", xlab = "Log intensity", ylab = "Density")

#Saving histogram before normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data pre-normalized")

#Histogram after data normalization
for (i in 1:ncol(exprs(data.skin.cancer.norm))) if (i == 1) plot(density(exprs(data.skin.cancer.norm)[, i]), col = "red", xlab = "Log intensity", ylab = "Density") else lines(density(exprs(data.skin.cancer.norm)[, i]), col = "red")

#Saving histogram after normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data normalized.eps")
```

# 3. Analysis
## 3.1. Creating the gene expression data frame

First of all, a clean gene expression data frame has to be made before the data can be analyzed. The annotation data (ensembl.103.txt, containing Transcript.stable.ID, Chromosome, AFFY HG U133 Plus 2 probe and HGNC symbol) was downloaded from https://www.ensembl.org/biomart and used to replace the transcript IDs with gene names.

```{r}
#Extraction of the expression values
Expression = exprs(data.skin.cancer.norm)

#Select only ENST IDs, which represent the transcript IDs
enst_id = str_detect(rownames(Expression),"ENST")
Expression = Expression[enst_id,]

#Remove .x_at suffix from each row name
rownames(Expression) = sub("\\..*", "", rownames(Expression)) 

#Remove .CEL suffix from each column name
colnames(Expression) = sub("\\..*", "", colnames(Expression))

#Load ensemble IDs and gene symbols from annotation data
setwd(paste(projectPath, 'tables', sep = '/'))
Ensembl = read.csv(paste(projectPath, 'tables/ensembl.103.txt', sep = '/'), sep = "\t")

#Extracting the transcript IDs and the HGNC symbols (gene symbols)
transcriptIDs = Ensembl$Transcript.stable.ID
HGNC_symbol = Ensembl$HGNC.symbol

#Assigning every gene symbol to a transcript ID
names(HGNC_symbol) = transcriptIDs

#Replace ensemble IDs with HGNC symbol
chipIDs = rownames(Expression)
newChipIDs = chipIDs[chipIDs %in% transcriptIDs]
Expression = Expression[newChipIDs,]

symbol = HGNC_symbol[newChipIDs]
rownames(Expression) = as.character(symbol)
head(Expression)


#Cleaning the Expression data frame (removing NAs and empty values "")
length(which(is.na(Expression)))
length(which(rownames(Expression) == ""))

all.spaces.exp = sapply(rownames(Expression), function(x){which(x == "")})
Expression = Expression[-which(all.spaces.exp>0),] 
rm(all.spaces.exp)

```

## 3.2. Extraction of skin-specific TRAs from the gene expression data frame

A new data frame has to be made with only skin-specific genes for a more specific analysis, as its these genes that are interesting.

```{r}
#Filtering the gene expression data frame to obtain expression values only for skin-specific TRAs
skin.genes.human.expression = which(rownames(Expression)%in%skin.genes.human)
data.skin.cancer.human_skin_TRA = Expression[skin.genes.human.expression,]

y = sort(rownames(data.skin.cancer.human_skin_TRA))
data.skin.cancer.human_skin_TRA = data.skin.cancer.human_skin_TRA[unique(y[y !=""]),]

#Renaming the columns in the data frame with the skin-specific genes
colnames(data.skin.cancer.human_skin_TRA)[1] = "Sherk1_D3a"
colnames(data.skin.cancer.human_skin_TRA)[2] = "Sherk1_D3b"
colnames(data.skin.cancer.human_skin_TRA)[3] = "Sherk1_D3c"
colnames(data.skin.cancer.human_skin_TRA)[4] = "Sherk1_D7a"
colnames(data.skin.cancer.human_skin_TRA)[5] = "Sherk1_D7c"
colnames(data.skin.cancer.human_skin_TRA)[6] = "ERK12_inh_3ha"
colnames(data.skin.cancer.human_skin_TRA)[7] = "ERK12_inh_3hb"
colnames(data.skin.cancer.human_skin_TRA)[8] = "Trametinib_3hb"
colnames(data.skin.cancer.human_skin_TRA)[9] = "Trametinib_3hc"
colnames(data.skin.cancer.human_skin_TRA)[10] = "ERK12_inh_24ha"
colnames(data.skin.cancer.human_skin_TRA)[11] = "ERK12_inh_24hb"
colnames(data.skin.cancer.human_skin_TRA)[12] = "ERK12_inh_24hc"
colnames(data.skin.cancer.human_skin_TRA)[13] = "Trametinib_24ha"
colnames(data.skin.cancer.human_skin_TRA)[14] = "Trametinib_24hb"
colnames(data.skin.cancer.human_skin_TRA)[15] = "Trametinib_24hc"

head(data.skin.cancer.human_skin_TRA)
```

## 3.3. Heatmap

A heatmap was made to visualize the expression data of all skin-specific TRAs over all chips.

```{r}
#Heatmap
library(gplots)
library(RColorBrewer)

exp.all.upregulated.tra = data.skin.cancer.human_skin_TRA[c(upreg_sherk_significant,upreg_erk12_significant,upreg_trametinib_significant),]
exp.all.upregulated.tra=exp.all.upregulated.tra[unique(rownames(exp.all.upregulated.tra)),]

yyyy = sort(rownames(exp.all.upregulated.tra))
exp.all.upregulated.tra = exp.all.upregulated.tra[yyyy,]

my_colors = brewer.pal(100, "RdBu" )
lmat = rbind(c(2,3), c(4,1))
lwid = c(1,4)
lhei = c(0.95,4)
heatmap.2(exp.all.upregulated.tra, key = T, col = my_colors, symkey = F, symm = F,symbreaks = T, scale = "row", density.info = "none", trace = "none", cexRow = 0.6, cexCol = 0.6, Rowv = F, Colv = F, lmat = lmat, lwid=lwid, lhei=lhei, main = "Expression of upregulated genes")

#Saving the heatmap
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Heatmap for skin specific TRAs in skin cancer_1.eps') #eps file
dev.copy2pdf(file = 'Heatmap for skin specific TRAs in skin cancer_1.pdf') #pdf file
```

## 3.4. Barplot

To show the localization of the skin-specific TRAs in the human genome, a barplot was made to easily visualize this distribution.

```{r}
#Barplot
#Get all the chromosome numbers for all human skin TRAs
chromosome_1 = TRA_Human_Novartis[,c(3,7)]
chromosome_skin1 = as.matrix(chromosome_1[ind1,])

chromosome_2 = TRA_Human_protien_atlas[,c(6,2)]
chromosome_skin2 = as.matrix(chromosome_2[ind2,])

chromosome_3 = TRA_Human_gtex[,c(3,6)]
chromosome_skin3 = as.matrix(chromosome_3[c(ind3.s, ind3.ns),])

#Combine all 3 tables and remove duplicates and empty values to make a clean list of chromosome numbers
chromosome_gene_table = rbind(chromosome_skin1, chromosome_skin2, chromosome_skin3)
rownames(chromosome_gene_table) = c()
a = sort(chromosome_gene_table[,1])

rownames(chromosome_gene_table) = chromosome_gene_table[,1]
chromosome_gene_table = chromosome_gene_table[,2]
chromosome_list = chromosome_gene_table[unique(a[a !=""])]

#Count the number of genes on each chromosome
localization = table(chromosome_list)
localization = as.matrix(localization)
sorted_chromosomes = str_sort(rownames(localization), numeric = T)
chromosome_counts = localization[sorted_chromosomes,]

#Make barplot
chromosome = c(1:22, "X")

barplot(chromosome_counts, main = "Distribution of skin-specific TRAs over chromosomes", xlab = "Chromosomes", ylab = "Number of skin-specific TRAs", names.arg = chromosome, cex.names = 0.7, col = "darkred")

#Saving the barplot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Distribution of skin-specific TRAs over chromosomes.eps') # eps file
dev.copy2pdf(file = 'Distribution of skin-specific TRAs over chromosomes.pdf') # pdf file
```

## 3.5. Pie chart

To visualize the portion of the skin-specific genes over all TRAs, a pie chart was made.

```{r}
#Piechart
#Creating small vectors for each table containing only NON-skin-specific TRAs

#TRA_Human_Novartis
bla1 = which(tissue1 != "skin")
TRA.symbolA = TRA_Human_Novartis$gene.symbol
non.skin.TRA1 = TRA.symbolA[bla1]

#TRA_Human_protein_atlas
bla2 = which(tissue2 != "skin")
TRA.symbolB = TRA_Human_protien_atlas$Symbol
non.skin.TRA2 = TRA.symbolB[bla2]

#TRA_Human_gtex (sun exposed)
bla3.s = which(tissue3.s != "skin")
TRA.symbolC = TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.s = TRA.symbolC[bla3.s]

#TRA_Human_gtex (not sun exposed)
bla3.ns = which(tissue3.ns != "skin")
TRA.symbolD = TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.ns = TRA.symbol3.ns[bla3.ns]

#Putting all the small non-skin-specific TRA vectors in a general vector containing all of them
NON_skin_specific_TRA = unique(c(non.skin.TRA1, non.skin.TRA2, non.skin.TRA3.s, non.skin.TRA3.ns))

#Calculating the total number of TRA
total = length(skin.genes.human) + length(NON_skin_specific_TRA)
prozent = round(length(skin.genes.human)/total*100, 2)
prozent_2 = round(length(NON_skin_specific_TRA)/total*100, 2)

labels = c("Skin-specific TRAs", "Other TRAs")
labels = paste(labels, c(prozent, prozent_2))
labels = paste(labels,"%", sep = "")

#Make pie chart
library(RColorBrewer)
color = brewer.pal(length(labels), "Spectral") 

pie(c(prozent, prozent_2), labels = labels, col = color, main = "Pie Chart skin specific TRAs")
legend("topright", labels, cex = 0.8, fill = color)

#Saving the pie chart
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file="Piechart_skin_TRA.eps") # eps file
dev.copy2pdf(file="Piechart_skin_TRA.pdf") # pdf file
```

## 3.6. Highly upregulated genes

Highly regulated genes in skin cancer may be interesting to look into.

```{r}
#Expression of all skin-specific TRAs in skin cancer
par(las=2)
boxplot(t(data.skin.cancer.human_skin_TRA), col = rainbow(length(rownames(data.skin.cancer.human_skin_TRA))), main = "Expression of all skin genes in melanoma", cex.axis = 0.8)

#Putting the highly expressed genes in a vector
highgenes = c()
for (i in length(colnames(data.skin.cancer.human_skin_TRA))) {
  highgenes = c(highgenes, names(which(data.skin.cancer.human_skin_TRA[,i] > 12)))
}
highgenes = unique(highgenes)

high = data.skin.cancer.human_skin_TRA[highgenes,]

# Expression of the highly expressed genes
par(las=2)
boxplot(t(high),col=rainbow(length(rownames(high))),main="Highly expressed skin genes in melanoma",cex.axis=0.8) #It can be clearly observed that there are two genes with distinctively highest expressions: GPNMB and ANXA2.
```

## 3.7. Log fold change and significance

In order to determine the mostly upregulated genes, the expression data has to be transformed by using a log fold change. Downregulated genes was also determined, as interesting results may arise after comparison. To further confirm whether the genes are significantly up- or downregulated, Welsh t-test was applied as a statistical test. Here, the null hypothesis states that the genes are not differently expressed between the two time points after treatment.

### 3.7.1. Upregulated genes

Upregulated genes are important to be analysed, as these are genes that are most essential for the survival of cancer cells. Since the cancer cells are attacked by the treatment (inhibitors), they have to upregulate certain genes in order to keep on living. By identifying these upregulated genes, survival pathways can be determined, analyzed and used as future drug targets.

```{r}
#LogFC: Finding the most upregulated genes from each treatment
#Average the gene expression for each treatment at each time point
sherk_3d = apply(data.skin.cancer.human_skin_TRA[,1:3], 1, mean)
sherk_7d = apply(data.skin.cancer.human_skin_TRA[,4:5], 1, mean)
erk12_3h = apply(data.skin.cancer.human_skin_TRA[,6:7], 1, mean)
erk12_24h = apply(data.skin.cancer.human_skin_TRA[,10:12], 1, mean)
trametinib_3h = apply(data.skin.cancer.human_skin_TRA[,8:9], 1, mean)
trametinib_24h = apply(data.skin.cancer.human_skin_TRA[,13:15], 1, mean)

#Calculate the log fold change for each gene from each treatment
logFC_sherk = sort(sherk_7d - sherk_3d, decreasing = T)
logFC_erk12 = sort(erk12_24h - erk12_3h, decreasing = T)
logFC_trametinib =  sort(trametinib_24h - trametinib_3h, decreasing = T)

#Top logFC from each treatment
logFC_sherk_top = logFC_sherk[1:11]
logFC_erk12_top = logFC_erk12[1:10]
logFC_trametinib_top = logFC_trametinib[1:10]

#T-test: Filtering upregulated genes that are significant
#Checking significance for genes in the top logFC of Dox-shERK
t_sherk = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 1:3], data.skin.cancer.human_skin_TRA[x, 4:5])$p.value
}
t_test_sherk = sort(sapply(names(logFC_sherk), t_sherk), decreasing = F)
t_test_sherk_significant = t_test_sherk[which(t_test_sherk < 0.05)]

upreg_sherk_significant = names(logFC_sherk_top)[names(logFC_sherk_top) %in% names(t_test_sherk_significant)]
upreg_sherk_significant

#Checking significance for genes in the top logFC of ERK1/2 inhibitor
t_erk12 = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 6:7], data.skin.cancer.human_skin_TRA[x, 10:12])$p.value
}
t_test_erk12 = sort(sapply(names(logFC_erk12), t_erk12), decreasing = F)
t_test_erk12_significant = t_test_erk12[which(t_test_erk12 < 0.05)]

upreg_erk12_significant = names(logFC_erk12_top)[names(logFC_erk12_top) %in% names(t_test_erk12_significant)]
upreg_erk12_significant

#Checking significance for genes in the top logFC of trametinib
t_trametinib = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 8:9], data.skin.cancer.human_skin_TRA[x, 13:15])$p.value
}
t_test_trametinib = sort(sapply(names(logFC_trametinib), t_trametinib), decreasing = F)
t_test_trametinib_significant = t_test_trametinib[which(t_test_trametinib < 0.05)]

upreg_trametinib_significant = names(logFC_trametinib_top)[names(logFC_trametinib_top) %in% names(t_test_trametinib_significant)]
upreg_trametinib_significant
```

### 3.7.2. Downregulated genes

Downregulated genes show that the drug has worked on genes that were tumour inducing at the beginning. Analysis of these genes was made to see if there are any interesting genes that may have a signficant meaning to the treatment.

```{r}
#Least logFC from each treatment
logFC_sherk_down_regulated = logFC_sherk[503:512]
logFC_erk12_down_regulated = logFC_erk12[503:512]
logFC_trametinib_down_regulated = logFC_trametinib[503:512]

#T-test: Filtering downregulated genes that are significant
#Checking significance for genes in the last 10 logFC of Dox-shERK
t_sherk = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 1:3], data.skin.cancer.human_skin_TRA[x, 4:5])$p.value
}
t_test_sherk = sort(sapply(names(logFC_sherk), t_sherk), decreasing = F)
t_test_sherk_significant = t_test_sherk[which(t_test_sherk < 0.05)]

downreg_sherk_significant = names(logFC_sherk_down_regulated)[names(logFC_sherk_down_regulated) %in% names(t_test_sherk_significant)]
downreg_sherk_significant

#Checking significance for genes in the last 10 logFC of ERK1/2 inhibitor
t_erk12 = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 6:7], data.skin.cancer.human_skin_TRA[x, 10:12])$p.value
}
t_test_erk12 = sort(sapply(names(logFC_erk12), t_erk12), decreasing = F)
t_test_erk12_significant = t_test_erk12[which(t_test_erk12 < 0.05)]

downreg_erk12_significant = names(logFC_erk12_down_regulated)[names(logFC_erk12_down_regulated) %in% names(t_test_erk12_significant)]
downreg_erk12_significant

## Checking significance for genes in the last 10 logFC of trametinib
t_trametinib = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 8:9], data.skin.cancer.human_skin_TRA[x, 13:15])$p.value
}
t_test_trametinib = sort(sapply(names(logFC_trametinib), t_trametinib), decreasing = F)
t_test_trametinib_significant = t_test_trametinib[which(t_test_trametinib < 0.05)]

downreg_trametinib_significant = names(logFC_trametinib_down_regulated)[names(logFC_trametinib_down_regulated) %in% names(t_test_trametinib_significant)]
downreg_trametinib_significant
```

## 3.8. Boxplots for different treatments on different times for up- and downregulated genes

To better visualize the up- and downregulation of these genes, boxplots were made.

### 3.8.1. Upregulated genes

```{r}
#Dox-shERK
data_sherk_upreg = data.skin.cancer.human_skin_TRA[upreg_sherk_significant, c(1:5)]
data.new_sherk = rbind(data_sherk_upreg, data_sherk_upreg)
data.new_sherk[c(1:2), c(4:5)] = NA
data.new_sherk[c(3:4), c(1:3)] = NA
order.vector = sort(colnames(t(data.new_sherk)),index.return=T)$ix

boxplot(t(data.new_sherk)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with Dox-shERK1", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:12), col = "grey", lty = 3)
legend("topleft", c("Dox-shERK1 after 3 days", "Dox-shERK1 after 7 days"), col = c("blue", "pink"), pch = 15, bg = "white")

#ERK1/2 inhibitor
data_erk12_upreg = data.skin.cancer.human_skin_TRA[upreg_erk12_significant, c(6:7, 10:12)]
data.new_erk12 = rbind(data_erk12_upreg, data_erk12_upreg)
data.new_erk12[c(1:8), c(3:5)] = NA
data.new_erk12[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_erk12)),index.return=T)$ix

boxplot(t(data.new_erk12)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with ERK1/2 inhibitor", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("ERK1/2 inhibitor after 3h", "ERK1/2 inhibitor after 24h"), col = c("blue", "pink"), pch = 15, bg = "white")

#Trametinib
data_trametinib_upreg = data.skin.cancer.human_skin_TRA[upreg_trametinib_significant, c(8:9, 13:15)]
data.new_trametinib = rbind(data_trametinib_upreg, data_trametinib_upreg)
data.new_trametinib[c(1:8), c(3:5)] = NA
data.new_trametinib[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_trametinib)),index.return=T)$ix

boxplot(t(data.new_trametinib)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with Trametinib", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("left",c("Trametinib after 3h", "Trametinib after 24h"),col=c("blue", "pink"), pch = 15, bg = "white")
```

### 3.8.2. Downregulated genes

```{r}
#Dox-shERK
data_sherk_downreg = data.skin.cancer.human_skin_TRA[downreg_sherk_significant, c(1:5)]
data.new_sherk_downreg = rbind(data_sherk_downreg, data_sherk_downreg)
data.new_sherk_downreg[c(1:2), c(4:5)] = NA
data.new_sherk_downreg[c(3:4), c(1:3)] = NA
order.vector = sort(colnames(t(data.new_sherk_downreg)),index.return=T)$ix

boxplot(t(data.new_sherk_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with Dox-shERK", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:12), col = "grey", lty = 3)
legend("topleft", c("Dox-shERK after 3 days", "Dox-shERK after 7 days"), col = c("blue", "pink"), pch = 15, bg = "white")

#ERK1/2 inhibitor
data_erk12_downreg = data.skin.cancer.human_skin_TRA[downreg_erk12_significant, c(6:7, 10:12)]
data.new_erk12_downreg = rbind(data_erk12_downreg, data_erk12_downreg)
data.new_erk12_downreg[c(1:8), c(3:5)] = NA
data.new_erk12_downreg[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_erk12_downreg)),index.return=T)$ix

boxplot(t(data.new_erk12_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with ERK1/2 inhibitor", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("ERK1/2 inhibitor after 3h", "ERK1/2 inhibitor after 24h"), col = c("blue", "pink"), pch = 15, bg = "white")

#Trametinib
data_trametinib_downreg = data.skin.cancer.human_skin_TRA[downreg_trametinib_significant, c(8:9, 13:15)]
data.new_trametinib_downreg = rbind(data_trametinib_downreg, data_trametinib_downreg)
data.new_trametinib_downreg[c(1:8), c(3:5)] = NA
data.new_trametinib_downreg[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_trametinib_downreg)),index.return=T)$ix

boxplot(t(data.new_trametinib_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with Trametinib", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("Trametinib after 3h", "Trametinib after 24h"),col=c("blue", "pink"), pch = 15, bg = "white")
```

## 3.9. Limma Analysis

The aim for using limma analysis is to find the differential expressed genes (or DEGs). The idea behind limma is that it takes the normalized data and performs statistical tests to discover any quantitative changes in the gene expression over time. Here, limma was used as a comparison to the upregulated genes found manually. 

```{r}
library(limma)

#First group: We compare the expression values from day 3 to day 7 with the Dox-shERK treatment
# Create design matrix
design = model.matrix(~ 0+factor(c(2,2,2,1,1)))
colnames(design) = c("Sherk7D", "Sherk3D")

#Fit linear model
fit = lmFit(data.skin.cancer.human_skin_TRA[,1:5], design)
contrast.matrix = makeContrasts(Sherk7D-Sherk3D, levels=design)
fit2 = contrasts.fit(fit, contrast.matrix)
fit2 =  eBayes(fit2, 0.01)

#Create top table with the DEGs sorted by log fold change
T1 = topTable(fit2,adjust.method = "fdr", sort.by = "logFC")

#Select only the first 10 genes with a positive log fold change from the table
lola1=T1[which(T1$logFC>0),]
lola1 = lola1[1:10,]

#Confirmed top upregulated genes
intersect(upreg_sherk_significant, rownames(lola1))


#Second group: We compare 3h ERK1/2 and 24h ERK1/2 treatment
#Create design matrix
design_ERK = model.matrix(~ 0+factor(c(1,1,1,2,2)))
colnames(design_ERK) = c("ERK12_inh_24h", "ERK12_inh_3h")

#Fit linear model
data_ERK = data.skin.cancer.human_skin_TRA[,c(10:12, 6,7)]
fit_ERK = lmFit(data_ERK, design_ERK) 
contrast.matrix_ERK = makeContrasts(ERK12_inh_24h-ERK12_inh_3h, levels=design_ERK)
fit2_ERK = contrasts.fit(fit_ERK, contrast.matrix_ERK)
fit2_ERK =  eBayes(fit2_ERK, 0.01)

#Create top table with the DEGs sorted by log fold change
T2 = topTable(fit2_ERK,adjust.method = "fdr", sort.by = "logFC", number = 130) 

#Select only the first 10 genes with a positive log fold change from the table
lola2 = T2[which(T2$logFC>0),]
lola2 = lola2[1:10,]

#Confirmed top upregulated genes
intersect(upreg_erk12_significant, rownames(lola2))


#Third group: We compare 3h to 24h Trametinib treatment
#Create design matrix
design_Trametinib = model.matrix(~ 0+factor(c(1,1,1,2,2)))
colnames(design_Trametinib) = c("Trametinib_24h", "Trametinib_3h")

#Fit linear model
data_Trametinib = data.skin.cancer.human_skin_TRA[,c(13:15, 8,9)]
fit_Trametinib = lmFit(data_Trametinib, design_Trametinib) 
contrast.matrix_Trametinib = makeContrasts(Trametinib_24h-Trametinib_3h, levels=design_Trametinib)
fit2_Trametinib= contrasts.fit(fit_Trametinib, contrast.matrix_Trametinib)
fit2_Trametinib =  eBayes(fit2_Trametinib, 0.01)

#Create top table with the DEGs sorted by log fold change
T3 = topTable(fit2_Trametinib,adjust.method = "fdr", sort.by = "logFC", number = Inf ) 

#Select only the first 10 genes with a positive log fold change from the table
lola3 = T3[which(T3$logFC>0),]
lola3 = lola3[1:10,]

#Confirmed top upregulated genes
intersect(upreg_trametinib_significant, rownames(lola3))


#The genes from limma analysis were very similar to the ones found manually, which means that the manual way succeeded. This was checked by doing the intersection of the vectors for the significant upregulated genes of each treatment and the top 10 genes found with limma.
```

## 3.10. Venn diagram

Genes that are upregullated in skin cancer and are at the same time TRAs are especially interesting, as they pose as potential drug targets. To visualize these upregulated TRAs, a venn diagram was made.

```{r}
#Finding upregulated genes in skin cancer by logFC
#Average the gene expression for each treatment at each time point
sherk_3d_all = apply(Expression[,1:3], 1, mean)
sherk_7d_all = apply(Expression[,4:5], 1, mean)
erk12_3h_all = apply(Expression[,6:7], 1, mean)
erk12_24h_all = apply(Expression[,10:12], 1, mean)
trametinib_3h_all = apply(Expression[,8:9], 1, mean)
trametinib_24h_all = apply(Expression[,13:15], 1, mean)

## Calculate the log fold change for each gene from each treatment
logFC_sherk_all = sherk_7d_all - sherk_3d_all
logFC_erk12_all = erk12_24h_all - erk12_3h_all
logFC_trametinib_all =  trametinib_24h_all - trametinib_3h_all

## Only positive logFC from each treatment (only upregulated genes!)
logFC_sherk_pos = which(logFC_sherk_all > 0)
logFC_erk12_pos = which(logFC_erk12_all > 0)
logFC_trametinib_pos = which(logFC_trametinib_all > 0)

upreg_all = unique(c(names(logFC_sherk_pos), names(logFC_erk12_pos), names(logFC_trametinib_pos)))

# T-test to filter significantly upregulated genes
## Checking significance for Dox-shERK
t_sherk_all = function(x) {
  if (Expression[x,1] == Expression[x,2] | Expression[x,1] == Expression[x,3] | Expression[x,3] == Expression[x,2] | Expression[x,4] == Expression[x,5])
    return(0.5)
  else
  t.test(Expression[x, 1:3], Expression[x, 4:5])$p.value
}
t_test_sherk_all = sapply(names(logFC_sherk_all), t_sherk_all)
t_test_sherk_significant_all = t_test_sherk_all[which(t_test_sherk_all < 0.05)]

upreg_sherk_significant_all = names(logFC_sherk_pos)[names(logFC_sherk_pos) %in% names(t_test_sherk_significant_all)]

## Checking significance for ERK1/2 inhibitor
t_erk12_all = function(x) {
  t.test(Expression[x, 6:7], Expression[x, 10:12])$p.value
}
t_test_erk12_all = sapply(names(logFC_erk12_all), t_erk12_all)
t_test_erk12_significant_all = t_test_erk12_all[which(t_test_erk12_all < 0.05)]

upreg_erk12_significant_all = names(logFC_erk12_pos)[names(logFC_erk12_pos) %in% names(t_test_erk12_significant_all)]

## Checking significance for Trametinib
t_trametinib_all = function(x) {
  t.test(Expression[x, 8:9], Expression[x, 13:15])$p.value
}
t_test_trametinib_all = sapply(names(logFC_trametinib_all), t_trametinib_all)
t_test_trametinib_significant_all = t_test_trametinib_all[which(t_test_trametinib_all < 0.05)]

upreg_trametinib_significant_all = names(logFC_trametinib_pos)[names(logFC_trametinib_pos) %in% names(t_test_trametinib_significant_all)]

#Venn diagram (Upregulated genes that are also TRAs)
library(venn)
list_venn = list(A = unique(c(upreg_sherk_significant_all, upreg_erk12_significant_all, upreg_trametinib_significant_all)), B = skin.genes.human)
names(list_venn) = c("Upregulated genes" , "TRA")

venn(list_venn, box = F, ilab = T, zcolor = c("#0073C2FF", "#EFC000FF"), cex = 1000)
text(500, 900, "Potential drug targets", cex = 2)

#Save the venn diagramm in a pdf file
setwd(paste(projectPath, 'plots', sep = '/'))
pdf("Venn_diagramm_trial.pdf")
venn(list_venn, box = F, ilab = T, zcolor = c("#0073C2FF", "#EFC000FF"))
text(500, 900, "Potential drug targets", cex = 2)
dev.off()

#Upregulated non-skin TRAs in skin cancer
upreg_tra = intersect(unique(c(upreg_sherk_significant_all, upreg_erk12_significant_all, upreg_trametinib_significant_all)), skin.genes.human)
upreg_tra_nonskin = upreg_tra[!upreg_tra%in%rownames(data.skin.cancer.human_skin_TRA)]
length(upreg_tra_nonskin) #There are no non-skin specific TRAs that are upregulated in skin cancer
```

## 3.11. Sun exposure

to show melanoma is caused by uv/sun --> which upregulated genes are sun exposed genes (related to uv exposure)
Boxplots of sun exposure vs no sun exposure in all data

```{r}
#Finding the sun exposed and not sun exposed genes in our skin-specific expression dataset
sun = intersect(rownames(data.skin.cancer.human_skin_TRA), skin.TRA3.s)
no.sun = intersect(rownames(data.skin.cancer.human_skin_TRA), skin.TRA3.ns)

#Removing the gene names to make a box plot of each chip
sun_exp = data.skin.cancer.human_skin_TRA[sun,]
nosun_exp = data.skin.cancer.human_skin_TRA[no.sun,]
sunexpchip = unname(sun_exp[,13])
nosunexpchip = unname(nosun_exp[,13])
lmts = range(sunexpchip,nosunexpchip)

#Comparing their significance with a welsh t-test for each chip
ttest_sun = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[sun, x], data.skin.cancer.human_skin_TRA[no.sun, x])$p.value
}
sapply(1:15, ttest_sun)
## All of them have a p value of smaller than 0.05, proving a significant difference between the expression of sun or no sun exposure genes. So now we will compare the expression of genes that are expressed with or without sun exposure.

# Genes that are expressed with or without sun exposure
both.sun = intersect(sun, no.sun)
both.sun
# Genes that are only sun exposure related
only_sun = sun[!sun%in%both.sun]
only_sun
#Genes that are only no sun exposure related
only_nosun = no.sun[!no.sun%in%both.sun]
only_nosun

#The upregulated sun exposed genes ETV7 and TGFBI are especially interesting, as ETV7 downregulation is related to poor prognosis (so working against the cancer) and TGFBI is strongly associated with tumor proliferation.

# The visualization of these genes can be seen in the graphs showing the logFC for all significantly upregulated genes.
significant_sun_1 = upreg_sherk_significant[upreg_sherk_significant%in%only_sun]
significant_sun_2 = upreg_erk12_significant[upreg_erk12_significant%in%only_sun]
significant_sun_3 = upreg_trametinib_significant[upreg_trametinib_significant%in%only_sun]

upregulated_sun = unique(c(significant_sun_1, significant_sun_2, significant_sun_3))
upregulated_sun

significant_nosun_1 = upreg_sherk_significant[upreg_sherk_significant%in%only_nosun]
significant_nosun_2 = upreg_erk12_significant[upreg_erk12_significant%in%only_nosun]
significant_nosun_3 = upreg_trametinib_significant[upreg_trametinib_significant%in%only_nosun]

upregulated_nosun = unique(c(significant_nosun_1, significant_nosun_2, significant_nosun_3))
upregulated_nosun
```

## 4. Clustering

Clustering was done to observe if there are any distinctive groups in our dataset.

```{r}
#Load libraries needed for clustering
library(factoextra)
library(clustertend)
library(fpc)
library(cluster)
library(clValid)
library(NbClust)
library(dplyr)
```

## 4.1. Clustering of sun / no sun exposure data

Sun and no sun exposure data were clustered to see if there are any obvious differences between the two gene groups.

```{r}
#Deciding on the optimal number of clusters
set.seed(123)
both = data.skin.cancer.human_skin_TRA[c(only_sun, only_nosun),]

#Elbow Method
fviz_nbclust(both, kmeans, method = "wss")+labs(subtitle = "Elbow method")

#Silhouette Method
fviz_nbclust(both, kmeans, method = "silhouette")+labs(subtitle = "Silhouette method")

#NbClust
nb.both = NbClust(both, distance = "euclidean", min.nc = 2,max.nc = 10, method = "kmeans")
fviz_nbclust(nb.both)
#The optimal number of clusters is 2.

#Is the data really clusterable?
hopkins(both, n = nrow(both)-1) #H value is lower than 0.5 and is relatively close to 1, meaning that there is a high probability that there are clusters in the sun/no sun data.

#Visualize k-means clusters
km.both = eclust(both, "kmeans", k = 2, nstart = 30, graph = FALSE)
fviz_cluster(km.both, data = both, palette = c("orange", "blue"), geom = "point", main = "Clustering of genes with or without sun exposure") + theme(plot.title = element_text(hjust = 0.5))

#Cluster Validation: Silhouette Information
fviz_silhouette(km.both, palette = "jco",
ggtheme = theme_classic())
#Average silhouette width of both clusters are higher than 0.5, meaning they are significantly clustered.
```

## 4.2. Clustering of the upregulated skin specific TRAs

```{r}
#Gathering the expression of highest upregulated genes
upregulated_all_exp = data.frame(data.skin.cancer.human_skin_TRA[c(upreg_sherk_significant, upreg_erk12_significant, upreg_trametinib_significant),])
upregulated_all_exp = distinct(upregulated_all_exp)

#Elbow method
fviz_nbclust(upregulated_all_exp, kmeans, method = "wss")+labs(subtitle = "Elbow method")

#Silhouette method
fviz_nbclust(upregulated_all_exp, kmeans, method = "silhouette")+labs(subtitle = "Silhouette method")
#The optimal number of clusters is 3.

#Is the data really clusterable?
hopkins(upregulated_all_exp, n = nrow(upregulated_all_exp)-1)#The H value is also lower than 0.5 and is relatively close to 0, so data probably has clusters.

#Visualize k-means clusters
km.upreg = eclust(upregulated_all_exp, "kmeans", k = 3, nstart = 25, graph = FALSE)
fviz_cluster(km.upreg, data = upregulated_all_exp, main = "Clustering of upregulated TRAs in skin cancer") + theme(plot.title = element_text(hjust = 0.5))

#Cluster validation: silhouette plot
fviz_silhouette(km.upreg, palette = "jco", ggtheme = theme_classic())
```

## 5. Principal component analysis

PCA was done to find the genes that explain most of the variance in the data. These genes were picked from principal components with the highest percentage of variance explained. Later on, the expression of upregulated genes will be predicted by using one or more of these genes through regression models.

```{r}
#Centering and scaling, pick data only from final time and only genes with highest variance
pca_dataframe = scale(data.skin.cancer.human_skin_TRA)
pca_dataframe = t(pca_dataframe[,c(4:5, 10:15)])

topVar = apply(pca_dataframe, 2, var)
q75 = quantile(topVar, probs = 0.75)
i.topvar = which(topVar >= q75)
pca_dataframe_topVar = pca_dataframe[,i.topvar]

#PCA
skin_pca_topVar = prcomp(pca_dataframe_topVar, scale. = F, center = F)

#Biplot of first 2 principal components
biplot(skin_pca_topVar)

#Proportion of variance of each principal component
variance = (skin_pca_topVar$sdev)^2
prop.variance = variance/sum(variance)
names(prop.variance) = 1:length(prop.variance)
barplot(prop.variance, ylab='Proportion of variance', xlab = "Principal components", main = "Proportion of variance of each principal component")

#Total variance explained by PC1 and PC2
total_var = prop.variance[1] + prop.variance[2]
total_var

#Barplots of principal components
par(las = 2)
barplot(skin_pca_topVar$rotation[,1], main = "PC1", ylab = "Contribution", xlab = "Genes")
barplot(skin_pca_topVar$rotation[,2], main = "PC2", ylab = "Contribution", xlab = "Genes")

#Genes with most contribution to PC1 and PC2
top_genes_pca1 = names(sort(abs(skin_pca_topVar$rotation[,1]), decreasing = T)[1:5])
top_genes_pca2 = names(sort(abs(skin_pca_topVar$rotation[,2]), decreasing = T)[1:6])

#Scatterplot of data with PC1 and PC2 as new coordinates
plot(skin_pca_topVar$x[,1], skin_pca_topVar$x[,2], 
     col= c("red", "red", "green", "green", "green", "blue", "blue", "blue"), pch=19,
     xlab='PC1', ylab='PC2', main = "PC1 and PC2 as new coordinates")
legend("topright",c("Sherk", "ERK1/2 inhibitor", "Trametinib"),col=c("red", "green", "blue"), pch = 15, bg = "yellow")
```

## 6. Regression models

Finally, regression models were made to predict the expression of the upregulated TRAs by using the genes previously selected by PCA. It is hoped that the expression of these essential upregulated genes in cancer can be modulated by targeting the genes from PCA. This would then be a good opportunity in developing new drugs with new drug targets.

## 6.1. Correlation between genes

Correlation between upregulated genes and genes obtained from PCA must first be done

```{r}
#Load libraries
library(corrplot)
library(QuantPsyc)
library(ggiraph)
library(ggiraphExtra)
library(ggplot2)
library(plyr)
library(car)

#Correlation of upregulated genes
df_upr = data.frame(t(data.skin.cancer.human_skin_TRA[unique(c(upreg_sherk_significant, upreg_erk12_significant, upreg_trametinib_significant)),]))

#Scatterplot PPFIBP2 and ARHGEF5 (example)
plot(df_upr[,"PPFIBP2"], df_upr[,"ARHGEF5"], pch=20, xlab='PPFIBP2', ylab='ARHGEF5')
#A linear relationship can clearly be seen. 

#Creating a correlation matrix with pearson correlation
cor_upr = cor(df_upr, method='pearson')

#Visualization of the Pearson correlation
corrplot(cor_upr, type="upper", order="alphabet",title = "Correlation within the upregulated genes (Pearson)")
#From the plot it can be seen that there is a high correlation between all upregulated genes except MPZL3. For the linear regression only MPZL3 and one other gene representing all others will be used (2 upregulated genes used in total).

#Correlation of genes from PCA
df_pca = data.frame(t(data.skin.cancer.human_skin_TRA[unique(c(top_genes_pca1, top_genes_pca2)),]))

#Creating a correlation matrix with pearson correlation
cor_pca = cor(df_pca, method='pearson')

#Visualization of the Pearson correlation
corrplot(cor_pca, type="upper", order="alphabet", title = "Correlation within the PCA genes (Pearson)")
#Genes that have a high correlation will be considered as they have the same impact on the upregulated genes. DCT, DSP and TGFBI will be used for the linear regression (5 genes from PCA used in total).
```

## 6.2. Linear regression and F-Tes

F-Test:
H0: The model doesn't refer to any significance.
H1: The model refers to a significance.
If the p-value is lower than the chosen threshold (in our case we chose 0.01), it is proven that the independent variables improve the fit in our model

The following have to be checked:
* the residuals are **normally distributed**
* there is **no correlation** between the residuals and the explanatory variable
* mean values of the residuals are close to 0

## 6.2.1. Multiple linear regression of MPZL3

```{r}
#MPZL3 expression predicted with PCA genes
mpzl3_lr = lm(MPZL3 ~ GJA1+DSP+ANXA2, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(mpzl3_lr)
#We can see that there is a significance on DSP. The p-value is also very small (<0.01), meaning that this model is significantly better than the null model. R-squared value is also high, meaning it was a good model (proportion of variance explained is high).

#Normal distribution of residuals
hist(mpzl3_lr$residuals,breaks=20)
shapiro.test(mpzl3_lr$residuals) #P-value is very high, meaning its almost definitely normally distributed.

#Correlation within residuals
plot(data.skin.cancer.human_skin_TRA["DSP",],mpzl3_lr$residuals,pch=20, title("Correlation of the residuals"), xlab="MPZL3", ylab="MPZL3_lr-Residuals")
cor(data.skin.cancer.human_skin_TRA["GJA1",], mpzl3_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["DSP",], mpzl3_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["ANXA2",], mpzl3_lr$residuals)
#No correlation between any of the explanatory genes and the residuals.

#Mean of the residuals
mean(mpzl3_lr$residuals) #Mean is almost 0.

#Predicting the MPZL3 expression with a multiple linear regression
plot(data.skin.cancer.human_skin_TRA["MPZL3",], mpzl3_lr$fitted.values,pch=20,col='blue', title("Multiple linear regression of MPZL3"), xlab='Real values',ylab='Predicted values');abline(0,1,col='red')
#We can see that the predicted values and the real values are quite similar, indicating that these genes have a strong relationship. If one is being highly expressed the other one would also be so.
```

## 6.2.2. Simple linear regression of MPZL3

```{r}
#MPZL3 expression predicted with a single gene
#In order to find the variable which has the most impact on the model, the standardized regression coefficients for each predictor gene have to be calculated.
lm.beta(mpzl3_lr)
#This means a change in 1 standard deviation of DSP has more impact than a 1 standard deviation change in other variables. 

#Plotting MPZL3 against only the variable that has the most impact (DSP)
mpzl3_dsp = lm(MPZL3 ~ DSP, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(mpzl3_dsp)

df_mpzl3_dsp = data.frame(y= data.skin.cancer.human_skin_TRA["MPZL3",], x= data.skin.cancer.human_skin_TRA["DSP",])

plot(df_mpzl3_dsp$x, df_mpzl3_dsp$y, xlab="Predictor(DSP)", ylab="Outcome(MPZL3)", main="Simple linear regression of MPZL3 by DSP", col="darkorange",pch=16)
abline(mpzl3_dsp, col="red")
for (i in 1:length(mpzl3_dsp$fitted.values))
  lines(c(df_mpzl3_dsp$x[i], df_mpzl3_dsp$x[i]), c(df_mpzl3_dsp$y[i], mpzl3_dsp$fitted.values[i]), col="blue")
```

## 6.2.3. Multiple linear regression of ARHGEF5

```{r}
#ARHGEF5 expression predicted with PCA genes
arhgef5_lr = lm(ARHGEF5 ~ DSP+GJA1+ANXA2, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(arhgef5_lr)
#We can see that there is a significance on DCT, but especially on TGFBI. The p-value is also very small (<0.01), meaning that this model is significantly better than the null model. R-squared value is also high, meaning it was a good model (proportion of variance explained is high).

#Normal distribution of residuals
hist(arhgef5_lr$residuals,breaks=20)
shapiro.test(arhgef5_lr$residuals) #P-value is relatively high, meaning its probably normally distributed.

#Correlation within residuals
plot(data.skin.cancer.human_skin_TRA["ANXA2",],arhgef5_lr$residuals,pch=20, title("Correlation of the residuals"), xlab="ARHGEF5", ylab="ARHGEF5_lr-Residuals")
cor(data.skin.cancer.human_skin_TRA["DSP",], arhgef5_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["GJA1",], arhgef5_lr$residuals)
cor(data.skin.cancer.human_skin_TRA["ANXA2",], arhgef5_lr$residuals)
#No correlation between any of the explanatory genes and the residuals.

#Mean of the residuals
mean(arhgef5_lr$residuals) #Mean is almost 0.

#Predicting the ARHGEF5 expression with a multiple linear regression
plot(data.skin.cancer.human_skin_TRA["ARHGEF5",], arhgef5_lr$fitted.values,pch=20,col='blue', title("Multiple linear regression of ARHGEF5"), xlab='Real values',ylab='Predicted values');abline(0,1,col='red')
#Here the predicted values and the real values are very similar everywhere along the regression line, meaning these genes have a very strong relationship. The expressions of ARHGEF5 and ANXA2 are directly proportional to each other.
```

## 6.2.2. Simple linear regression of ARHGEF5

```{r}
#ARHGEF5 expression predicted with a single gene
#In order to find the variable which has the most impact on the model, the standardized regression coefficients for each predictor gene have to be calculated.
lm.beta(arhgef5_lr)

#Plotting ARHGEF5 against only the variable that has the most impact (TGFBI)
arhgef5_anxa2 = lm(ARHGEF5 ~ ANXA2, data = data.frame(t(data.skin.cancer.human_skin_TRA)))
summary(arhgef5_anxa2)

df_arhgef5_anxa2 = data.frame(y= data.skin.cancer.human_skin_TRA["ARHGEF5",], x= data.skin.cancer.human_skin_TRA["ANXA2",])

plot(df_arhgef5_anxa2$x, df_arhgef5_anxa2$y, xlab="Predictor(ANXA2)", ylab="Outcome(ARHGEF5)", main="Simple linear regression of ARHGEF5 by ANXA2", col="darkorange",pch=16)
abline(arhgef5_anxa2, col="red")
for (i in 1:length(arhgef5_anxa2$fitted.values))
  lines(c(df_arhgef5_anxa2$x[i], df_arhgef5_anxa2$x[i]), c(df_arhgef5_anxa2$y[i], arhgef5_anxa2$fitted.values[i]), col="blue")

```

