---
title: "The Role of Skin-Specific TRAs on Skin Cancer"
author: "Binnur Özay, Christofer Richard, Denisa Neacsu, Pinar Cavus"
date: "Data Analysis Project SS21"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# Introduction

Skin cancer makes up most of the malignancies all around the world, and consists of two main different types; melanoma and non-melanoma (Linares et al., 2015). Both are mainly caused by exposure to UV lights, but non-melanoma cancer develops in the outer layers of the skin (Leiter & Garbe, 2008), whereas melanoma develops from the melanocytes (Smith et al., 2003). Melanomas only make up a small amount of reported skin cancer diagnoses, however they still account for the majority of skin cancer deaths, making them extremely dangerous (Smith et al., 2003). Melanomas have the highest rate of metastatic capacity of all tumors when they have delayed diagnoses (Potrony et al., 2015). 
More than half of all melanoma cases contain a mutation of the BRAF gene (Alqathama, 2020). The B-Raf protein is a signaling molecule in the Mitogen-activated protein kinase (MAPK) pathway, activating MEK and leading to the phosphorylation of ERK (Alqathama, 2020). This kinase cascade is known to direct important cell processes such as proliferation, differentiation, and survival (Alqathama, 2020).
In this analysis, data from the melanoma cell line WM266-4 is used. This cell line has a BRAF-V600D mutation, meaning that the valine at codon 600 is changed into aspartic acid. This mutation leads to an overexpression of BRAF, therefore an uncontrolled increase in oncogenic signaling (Alqathama, 2020). The cell line is treated with Trametinib, which is an inhibitor of MEK 1 and MEK2 (Zeiser, 2014), ERK1/2-inhibitor, and doxycycline-shERK1 for different periods of time; 3 and 24 hours, 3 and 7 days. Later, the mRNA from these cells are extracted and processed into labeled cRNA fragments. These fragments are hybridized on microarrays and scanned, resulting in expression level quantifications of large amounts of genes. It is to mention that there are no units in this data, because it is simply light intensity scanned. 
The focus in this microarray analysis is a mechanism cancer cells have been shown to use; the upregulation of tissue restricted antigen (TRA) genes. TRAs are highly expressed selectively in their tissues, and T-cells reacting to these self-antigens result in autoimmune diseases (Kyewski & Derbinski, 2004). So that immature T-cells can gauge the self-reactivity of their antigen receptors, they go through a selection in the thymus, where T-cells reacting to TRAs undergo apoptosis (Klein et al., 2014). TRA genes are mostly upregulated in cancer cells for undiscovered reasons, making them good potential drug targets for therapy (Rosenberg, 1999). Current treatments for melanomas include chemotherapy, radiation, immunotherapy, and surgical excision (Gruber & Zito, 2020). Developing a new therapy, where the upregulated skin specific TRA genes are attacked, therefore other tissues don’t get as damaged, would be a significant discovery in the medical field.
Therefore, the main aim of this project is to find upregulated skin specific genes in skin cancer. By looking into the gene expression data of TRA genes in a melanoma cell line, and comparing the variety of the expression levels between different treatments and treatment times, it is possible to gain insight into how the genes affect the survival of the cancer cells. The genes which present significant changes in their expression would be potential drug targets to further investigate, as they would be helping the cancer cell survive under the treatment.


# METHODS AND RESULTS

Quality Control and Normalization of the Data

The programming language R Studio Version 1.4.1106 (R Core Team, 2021) is used throughout this entire project. First, all of the data to be analyzed is downloaded; the numerical microarray data with gene expression intensities of the melanoma cell line and the nominal TRA data (Dinkelacker, 2007;2019) with names of genes originating from different tissues. The TRA data has other information such as chromosome numbers of the given genes which will be put to use later.

```{r}
# 1. TRA Analysis
## 1.1. Selection of skin-specific TRAs

#Define work directory
projectPath = dirname(rstudioapi::getSourceEditorContext()$path)

#Import TRA Data
setwd(paste(projectPath, 'TRA Daten', sep = '/'))
TRA_Human_gtex = read.csv(file = "tra.2017.human.gtex.5x.table.tsv", sep = "\t")
TRA_Mouse_4301 = read.csv(file = "tra.2014.mouse.4301.5x.table.tsv", sep = "\t")
TRA_Human_roth = read.csv(file = "tra.2014.human.roth.5x.table.tsv", sep = "\t")
TRA_Human_Novartis = read.csv(file = "tra.2014.human.5x.table.tsv", sep = "\t")
TRA_Mouse_Novartis = read.csv(file = "tra.2014.mouse.5x.table.tsv", sep = "\t")
TRA_Human_protien_atlas = read.csv(file = "Human_protein_atlas_TRA_5median_genes_annotated.tsv", sep = ",")
```
TRA data is made up of genes from different tissues, different cell lines of humans and mice. All skin genes are extracted and combined in a vector, all NA values are erased. As the skin data available on mice is very little, only the expression of human genes is analyzed. One set of TRA data has the information of sun exposure, as this information might be valuable further into the analysis, these genes are separated into two vectors; exposed to the sun and not exposed to the sun.

```{r}
#Skin-specific genes from TRA_Human_Novartis
tissue1 = TRA_Human_Novartis[,11]
ind1 = which(tissue1 == "skin")
TRA.symbol1 = TRA_Human_Novartis$gene.symbol
skin.TRA1 = TRA.symbol1[ind1]

#Skin-specific genes from TRA_Human_protein_atlas
tissue2 = TRA_Human_protien_atlas[,11]
ind2 = which(tissue2 == "skin")
TRA.symbol2 = TRA_Human_protien_atlas$Symbol
skin.TRA2 = TRA.symbol2[ind2]

#Skin-specific genes from TRA_Human_gtex (sun exposed)
tissue3.s = TRA_Human_gtex[,10]
ind3.s = which(tissue3.s == "Skin - Sun Exposed")
TRA.symbol3.s = TRA_Human_gtex$ensembl.symbol
skin.TRA3.s = TRA.symbol3.s[ind3.s]

#Skin-specific genes from TRA_Human_gtex (not sun exposed)
tissue3.ns = TRA_Human_gtex[,10]
ind3.ns = which(tissue3.ns == "Skin - Not Sun Exposed")
TRA.symbol3.ns = TRA_Human_gtex$ensembl.symbol
skin.TRA3.ns = TRA.symbol3.ns[ind3.ns]

#Skin-specific genes from TRA_Human_roth (no skin-specific genes)
tissue4 = TRA_Human_roth[,11]
ind4 = which(tissue4 == "skin")
TRA.symbol4 = TRA_Human_roth$gene.symbol
skin.TRA4 = TRA.symbol4[ind4]

#Skin-specific genes from TRA_Mouse_4301 (epidermis-specific genes)
tissue5 = TRA_Mouse_4301[,11]
ind5 = which(tissue5 == "epidermis")
TRA.symbol5 = TRA_Mouse_4301$gene.symbol
skin.TRA5 = TRA.symbol5[ind5]

#Skin-specific genes from TRA_Mouse_Novartis (epidermis-specific genes)
tissue6 = TRA_Mouse_Novartis[,11]
ind6 = which(tissue6 == "epidermis")
TRA.symbol6 = TRA_Mouse_Novartis$gene.symbol
skin.TRA6 = TRA.symbol6[ind6]

## 1.2. Compilation of all skin-specific TRAs
#Skin-specific TRAs for humans were put into a single vector and data was cleaned up.
#Creating vector with all the human skin genes

skin.genes.human = unique(c(skin.TRA1, skin.TRA2, skin.TRA3.ns, skin.TRA3.s))

#Removing NAs and empty values
skin.genes.human = skin.genes.human[-which(is.na(skin.genes.human))] 
skin.genes.human = skin.genes.human[-which(skin.genes.human == "")] 

#Verification that all NA and empty values were removed
which(is.na(skin.genes.human)) 
which(skin.genes.human == "")

#Saving this vector with all skin-specific human TRAs in a csv file
setwd(paste(projectPath, 'tables', sep = '/'))
write.csv(matrix(skin.genes.human), file = "skin.specific.genes.human.TRA", row.names = F, sep = "\t")
```
A pie chart is finalized to visualize how much of the TRA data comes from skin. It is found out that skin specific genes make up only 4.84% of all TRA data.
```{r}
#Piechart
#Creating small vectors for each table containing only NON-skin-specific TRAs

#TRA_Human_Novartis
bla1 = which(tissue1 != "skin")
TRA.symbolA = TRA_Human_Novartis$gene.symbol
non.skin.TRA1 = TRA.symbolA[bla1]

#TRA_Human_protein_atlas
bla2 = which(tissue2 != "skin")
TRA.symbolB = TRA_Human_protien_atlas$Symbol
non.skin.TRA2 = TRA.symbolB[bla2]

#TRA_Human_gtex (sun exposed)
bla3.s = which(tissue3.s != "skin")
TRA.symbolC = TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.s = TRA.symbolC[bla3.s]

#TRA_Human_gtex (not sun exposed)
bla3.ns = which(tissue3.ns != "skin")
TRA.symbolD = TRA_Human_gtex$ensembl.symbol
non.skin.TRA3.ns = TRA.symbol3.ns[bla3.ns]

#Putting all the small non-skin-specific TRA vectors in a general vector containing all of them
NON_skin_specific_TRA = unique(c(non.skin.TRA1, non.skin.TRA2, non.skin.TRA3.s, non.skin.TRA3.ns))

#Calculating the total number of TRA
total = length(skin.genes.human) + length(NON_skin_specific_TRA)
prozent = round(length(skin.genes.human)/total*100, 2)
prozent_2 = round(length(NON_skin_specific_TRA)/total*100, 2)

labels = c("Skin-specific TRAs", "Other TRAs")
labels = paste(labels, c(prozent, prozent_2))
labels = paste(labels,"%", sep = "")

#Make pie chart
library(RColorBrewer)
color = brewer.pal(length(labels), "Spectral") 

pie(c(prozent, prozent_2), labels = labels, col = color, main = "Pie Chart skin specific TRAs")
legend("topright", labels, cex = 0.8, fill = color)

#Saving the pie chart
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file="Piechart_skin_TRA.eps") # eps file
dev.copy2pdf(file="Piechart_skin_TRA.pdf") # pdf file
```
Before the analysis, the data must be investigated to see if there are any problems within, for instance a  broken microarray chip, that would alter the results.
```{r, fig.show = 'hide'}
#2. Quality control of microarray data of skin cancer
The Affymetrix Human Genome U133 Plus 2.0 Array was used and the package for annotation of the chips was downloaded from the Brainarray website.

##2.1. Loading the skin cancer dataset
#15 chips were chosen to be analysed and were downloaded from the GEO website. The 15 chips that were chosen to be used are: GSM1387513, GSM1387514, GSM1387515, GSM1387525, GSM1387527, GSM1387537, GSM1387538, GSM1387544, GSM1387545, GSM1387552, GSM1387553, GSM1387554, GSM1387558, GSM1387559 and GSM1387560.

#Loading the necessary libraries
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(hexbin)
library(stringr)

#Read all 15 microarray chips
setwd(paste(projectPath, 'rawdata/GSE57721 melanoma', sep = '/'))
data.skin.cancer = ReadAffy()
data.skin.cancer@cdfName = "HGU133Plus2_Hs_ENST"

orig = colnames(exprs(data.skin.cancer))
new.name = substr(orig, 1, nchar(orig)-4)
colnames(exprs(data.skin.cancer)) = new.name

#Saving raw data as a .rda file
setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "data.skin.cancer.rda")
```
Single chips are controlled by visualizing raw microarray data, there aren’t any abnormalities visible in all of the fifteen chips.
```{r,fig.show = 'hide'}
## 2.2. Single chip control
#Visualizing raw microarray data
par(mfrow = c(3,5))
image(data.skin.cancer,col=rainbow(100,start = 0, end = 0.75) [100:1]) #All of the chips seem to have a very good quality, so we can use all of them
```
Gene expression profiling results are highly influenced by RNA quality and degradation (Opitz et al., 2010), therefore RNA degradation plots are made to investigate. All of the lines, each representing a chip, showed similar slope values, and there isn’t any deviation visible. For this reason, all chips are to be included in the analysis as they are all qualitative.
```{r,fig.show = 'hide'}
## 2.2. RNA degradation
# RNA degradation plot shifted and scaled
rna_degradation = AffyRNAdeg(data.skin.cancer)

plotAffyRNAdeg(rna_degradation, col = rainbow(150))
title(sub="Human skin cancer raw")

#Saving shifted and scaled RNA degradation plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "RNA_degradation_data_skin_cancer_raw.eps")

#RNA degradation plot only shifted
plotAffyRNAdeg(rna_degradation, col = rainbow(150), transform = "shift.only")
title(sub = "Human skin cancer raw")

#Saving only-shifted RNA degradation plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file= "RNA_deg_data_skin_cancer_raw.eps")

#Analysis of values and plots from RNA degradation
summaryAffyRNAdeg(rna_degradation) # The values of the slopes are similar, so all of the chips are usable for further analysis

## According to the plots, there is no obvious deviation, no line stands out; all of the chips have high expression intensity. In otherwords, all chips are of high quality.
```
The microarray data is normalized using the package vsn (v3.58.0; Hüber, NA) to remove any systematic variation that can affect the analysis. The pre-normalization and normalized data, which is glog2 transformed, are compared in the forms of boxplots and histograms, in order to see how much systematic variation existed. The boxplots didn’t show a significant difference, even before normalization the median lines are almost in a linear line, indicating that the chips didn’t have much variation to begin with. In both pre- and after normalization histograms, the chip lines follow the same path, and none of the chips stand out.
```{r,fig.show = 'hide'}
## 2.3. Normalization of microarray data
#Data normalization
data.skin.cancer.norm = vsnrma(data.skin.cancer)

#Saving normalized microarray data as a .rda file
setwd(paste(projectPath, 'sessions/rda', sep = '/'))
save.image(file = "normalized_data.rda")
```
```{r,fig.show = 'hide'}
##Boxplots of pre-normalized and normalized data
#Before normalization
par(las = 2)
mmi = c(1.8, 0.7, 1.05, 0.54)
par(mai = mmi)
boxplot(data.skin.cancer, col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer before normalization")

#Saving boxplot with data before normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_rawdata.eps")

#After normalization
par(las = 2, mai = mmi)
boxplot(exprs(data.skin.cancer.norm), col=rainbow(150), cex.axis = 0.5, main = "Gene expression in skin cancer after normalization")

#Saving boxplot with data after normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "boxplot_skin_cancer_normalized.eps")
```
```{r,fig.show = 'hide'}
##Histogram
#Histogram before data normalization
hist(data.skin.cancer, col = rainbow(150), main = "Histogram data pre-normalized", xlab = "Log intensity", ylab = "Density")

#Saving histogram before normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data pre-normalized")

#Histogram after data normalization
for (i in 1:ncol(exprs(data.skin.cancer.norm))) if (i == 1) plot(density(exprs(data.skin.cancer.norm)[, i]), col = "red", xlab = "Log intensity", ylab = "Density") else lines(density(exprs(data.skin.cancer.norm)[, i]), col = "red")

#Saving histogram after normalization
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "Histogram data normalized.eps")
```
After the normalization, a meanSdPlot is made, and from this plot, it is seen that the mean values of the chips are almost linear, once again proving that they are high quality chips.
```{r,fig.show = 'hide'}
## 2.4. meanSd Plot
#Plotting meanSdPlot
meanSdPlot(data.skin.cancer.norm)

#Saving meanSd plot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = "meanSdPlot_data_skin_cancer_normalized.eps")
```
Cleaning Data and Creating a Dataframe

In the data from microarray chips genes are saved as numbers, to be able to view and analyse the data easier, these numbers are switched with their names using the ensembl IDs that were downloaded from https://www.ensembl.org/biomart . Later, all expression values are extracted from the microarray data, and put into a data frame with chip IDs as columns and gene symbols as rows.
```{r,fig.show = 'hide'}
# 3. Analysis
## 3.1. Creating the gene expression data frame

#Extraction of the expression values
Expression = exprs(data.skin.cancer.norm)

#Select only ENST IDs, which represent the transcript IDs
enst_id = str_detect(rownames(Expression),"ENST")
Expression = Expression[enst_id,]

#Remove .x_at suffix from each row name
rownames(Expression) = sub("\\..*", "", rownames(Expression)) 

#Remove .CEL suffix from each column name
colnames(Expression) = sub("\\..*", "", colnames(Expression))

#Load ensemble IDs and gene symbols from annotation data
setwd(paste(projectPath, 'tables', sep = '/'))
Ensembl = read.csv(paste(projectPath, 'tables/ensembl.103.txt', sep = '/'), sep = "\t")

#Extracting the transcript IDs and the HGNC symbols (gene symbols)
transcriptIDs = Ensembl$Transcript.stable.ID
HGNC_symbol = Ensembl$HGNC.symbol

#Assigning every gene symbol to a transcript ID
names(HGNC_symbol) = transcriptIDs

#Replace ensemble IDs with HGNC symbol
chipIDs = rownames(Expression)
newChipIDs = chipIDs[chipIDs %in% transcriptIDs]
Expression = Expression[newChipIDs,]

symbol = HGNC_symbol[newChipIDs]
rownames(Expression) = as.character(symbol)
head(Expression)

#Cleaning the Expression data frame (removing NAs and empty values "")
length(which(is.na(Expression)))
length(which(rownames(Expression) == ""))

all.spaces.exp = sapply(rownames(Expression), function(x){which(x == "")})
Expression = Expression[-which(all.spaces.exp>0),] 
rm(all.spaces.exp)

```
After empty spaces and NA values are removed from the data, skin specific antigens are extracted to create the main data frame of skin TRA expression levels in skin cancer to analyze. The column names are changed so that the treatment and time point of every chip is easily viewed.
```{r}
## 3.2. Extraction of skin-specific TRAs from the gene expression data frame
#Filtering the gene expression data frame to obtain expression values only for skin-specific TRAs
skin.genes.human.expression = which(rownames(Expression)%in%skin.genes.human)
data.skin.cancer.human_skin_TRA = Expression[skin.genes.human.expression,]

y = sort(rownames(data.skin.cancer.human_skin_TRA))
data.skin.cancer.human_skin_TRA = data.skin.cancer.human_skin_TRA[unique(y[y !=""]),]

#Renaming the columns in the data frame with the skin-specific genes
colnames(data.skin.cancer.human_skin_TRA)[1] = "Sherk1_D3a"
colnames(data.skin.cancer.human_skin_TRA)[2] = "Sherk1_D3b"
colnames(data.skin.cancer.human_skin_TRA)[3] = "Sherk1_D3c"
colnames(data.skin.cancer.human_skin_TRA)[4] = "Sherk1_D7a"
colnames(data.skin.cancer.human_skin_TRA)[5] = "Sherk1_D7c"
colnames(data.skin.cancer.human_skin_TRA)[6] = "ERK12_inh_3ha"
colnames(data.skin.cancer.human_skin_TRA)[7] = "ERK12_inh_3hb"
colnames(data.skin.cancer.human_skin_TRA)[8] = "Trametinib_3hb"
colnames(data.skin.cancer.human_skin_TRA)[9] = "Trametinib_3hc"
colnames(data.skin.cancer.human_skin_TRA)[10] = "ERK12_inh_24ha"
colnames(data.skin.cancer.human_skin_TRA)[11] = "ERK12_inh_24hb"
colnames(data.skin.cancer.human_skin_TRA)[12] = "ERK12_inh_24hc"
colnames(data.skin.cancer.human_skin_TRA)[13] = "Trametinib_24ha"
colnames(data.skin.cancer.human_skin_TRA)[14] = "Trametinib_24hb"
colnames(data.skin.cancer.human_skin_TRA)[15] = "Trametinib_24hc"

head(data.skin.cancer.human_skin_TRA)
```
Before getting to the gene expression analysis a bar plot is created to find out which chromosome the skin specific TRAs are mostly located at. Most skin related genes are usually located on chromosome 1 (Volz et al., 1993), which also seems to be the case in this data.
```{r,fig.show = 'hide'}
## 3.4. Barplot
#Barplot
#Get all the chromosome numbers for all human skin TRAs
chromosome_1 = TRA_Human_Novartis[,c(3,7)]
chromosome_skin1 = as.matrix(chromosome_1[ind1,])

chromosome_2 = TRA_Human_protien_atlas[,c(6,2)]
chromosome_skin2 = as.matrix(chromosome_2[ind2,])

chromosome_3 = TRA_Human_gtex[,c(3,6)]
chromosome_skin3 = as.matrix(chromosome_3[c(ind3.s, ind3.ns),])

#Combine all 3 tables and remove duplicates and empty values to make a clean list of chromosome numbers
chromosome_gene_table = rbind(chromosome_skin1, chromosome_skin2, chromosome_skin3)
rownames(chromosome_gene_table) = c()
a = sort(chromosome_gene_table[,1])

rownames(chromosome_gene_table) = chromosome_gene_table[,1]
chromosome_gene_table = chromosome_gene_table[,2]
chromosome_list = chromosome_gene_table[unique(a[a !=""])]

#Count the number of genes on each chromosome
localization = table(chromosome_list)
localization = as.matrix(localization)
sorted_chromosomes = str_sort(rownames(localization), numeric = T)
chromosome_counts = localization[sorted_chromosomes,]

#Make barplot
chromosome = c(1:22, "X")

barplot(chromosome_counts, main = "Distribution of skin-specific TRAs over chromosomes", xlab = "Chromosomes", ylab = "Number of skin-specific TRAs", names.arg = chromosome, cex.names = 0.7, col = "darkred")

#Saving the barplot
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Distribution of skin-specific TRAs over chromosomes.eps') # eps file
dev.copy2pdf(file = 'Distribution of skin-specific TRAs over chromosomes.pdf') # pdf file
```
A heatmap was made to visualize the expression data of all skin-specific TRAs over all chips.
```{r,fig.show = 'hide'}
## 3.3. Heatmap
#Heatmap
library(gplots)

heatmap.2(data.skin.cancer.human_skin_TRA, key = T, col = redgreen(100), symkey = F, symm = F,symbreaks = T, scale = "row", density.info = "none", trace = "none", cexRow = 0.6, cexCol = 0.6)

#Saving the heatmap
setwd(paste(projectPath, 'plots', sep = '/'))
dev.copy2eps(file = 'Heatmap for skin specific TRAs in skin cancer_1.eps') #eps file
dev.copy2pdf(file = 'Heatmap for skin specific TRAs in skin cancer_1.pdf') #pdf file
```
To begin the analysis the gene expression is visualized with box plots. There are too many genes overall, therefore the boxes look like overlapping lines, which does not give much information. Instead, only the highly expressed genes are chosen to be visualized. A highly expressed gene doesn’t have to be an upregulated gene, and an upregulated gene has the possibility of not being a highly expressed gene. A threshold of twelve is set according to the overall expression levels to define the high expression margin. Aside from the two highest expressed genes GPNMB and ANXA2, the genes NDRG1, RPL18, DCT, CD44, and MLANA show varying expression levels between chips with levels above the threshold in some chips and below in others.
```{r,fig.show = 'hide'}
#Expression of all skin-specific TRAs in skin cancer
par(las=2)
boxplot(t(data.skin.cancer.human_skin_TRA), col = rainbow(length(rownames(data.skin.cancer.human_skin_TRA))), main = "Expression of all skin genes in melanoma", cex.axis = 0.8)
```
```{r}
#Putting the highly expressed genes in a vector
highgenes = c()
for (i in length(colnames(data.skin.cancer.human_skin_TRA))) {
  highgenes = c(highgenes, names(which(data.skin.cancer.human_skin_TRA[,i] > 12)))
}
highgenes = unique(highgenes)

high = data.skin.cancer.human_skin_TRA[highgenes,]

# Expression of the highly expressed genes
par(las=2)
boxplot(t(high),col=rainbow(length(rownames(high))),main="Highly expressed skin genes in melanoma",cex.axis=0.8) #It can be clearly observed that there are two genes with distinctively highest expressions: GPNMB and ANXA2.
```
Upregulated Genes
In order to determine the mostly upregulated genes, the expression data has to be transformed by using a log fold change. Downregulated genes was also determined, as interesting results may arise after comparison. To further confirm whether the genes are significantly up- or downregulated, Welsh t-test was applied as a statistical test. Here, the null hypothesis states that the genes are not differently expressed between the two time points after treatment.There isn’t any data available from cells without any treatment, therefore for chips that have been treated with ERK1/2-inhibitor or Trametinib, the gene expression between the three hour and twenty four hour time points is compared to see which genes showed a higher expression after twenty one hours of treatment. The zero point for dox-shERK1 chips is taken as three days, as there isn’t any earlier data available.
```{r}
## 3.7. Log fold change and significance
### 3.7.1. Upregulated genes
#LogFC: Finding the most upregulated genes from each treatment
#Average the gene expression for each treatment at each time point
sherk_3d = apply(data.skin.cancer.human_skin_TRA[,1:3], 1, mean)
sherk_7d = apply(data.skin.cancer.human_skin_TRA[,4:5], 1, mean)
erk12_3h = apply(data.skin.cancer.human_skin_TRA[,6:7], 1, mean)
erk12_24h = apply(data.skin.cancer.human_skin_TRA[,10:12], 1, mean)
trametinib_3h = apply(data.skin.cancer.human_skin_TRA[,8:9], 1, mean)
trametinib_24h = apply(data.skin.cancer.human_skin_TRA[,13:15], 1, mean)

#Calculate the log fold change for each gene from each treatment
logFC_sherk = sort(sherk_7d - sherk_3d, decreasing = T)
logFC_erk12 = sort(erk12_24h - erk12_3h, decreasing = T)
logFC_trametinib =  sort(trametinib_24h - trametinib_3h, decreasing = T)

#Top logFC from each treatment
logFC_sherk_top = logFC_sherk[1:11]
logFC_erk12_top = logFC_erk12[1:10]
logFC_trametinib_top = logFC_trametinib[1:10]
```
Not all genes encountered at the end of this analysis are significantly upregulated to become drug targets. Therefore, ten genes with the highest log-ratios from different treatments are chosen to be filtered through a Welsh t-test. The following upregulated genes are respectively from dox-shERK1, ERK1/2 inhibitor, and Trametinib chips. 
```{r}
#T-test: Filtering upregulated genes that are significant
#Checking significance for genes in the top logFC of Dox-shERK
t_sherk = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 1:3], data.skin.cancer.human_skin_TRA[x, 4:5])$p.value
}
t_test_sherk = sort(sapply(names(logFC_sherk), t_sherk), decreasing = F)
t_test_sherk_significant = t_test_sherk[which(t_test_sherk < 0.05)]

upreg_sherk_significant = names(logFC_sherk_top)[names(logFC_sherk_top) %in% names(t_test_sherk_significant)]
upreg_sherk_significant

#Checking significance for genes in the top logFC of ERK1/2 inhibitor
t_erk12 = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 6:7], data.skin.cancer.human_skin_TRA[x, 10:12])$p.value
}
t_test_erk12 = sort(sapply(names(logFC_erk12), t_erk12), decreasing = F)
t_test_erk12_significant = t_test_erk12[which(t_test_erk12 < 0.05)]

upreg_erk12_significant = names(logFC_erk12_top)[names(logFC_erk12_top) %in% names(t_test_erk12_significant)]
upreg_erk12_significant

#Checking significance for genes in the top logFC of trametinib
t_trametinib = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 8:9], data.skin.cancer.human_skin_TRA[x, 13:15])$p.value
}
t_test_trametinib = sort(sapply(names(logFC_trametinib), t_trametinib), decreasing = F)
t_test_trametinib_significant = t_test_trametinib[which(t_test_trametinib < 0.05)]

upreg_trametinib_significant = names(logFC_trametinib_top)[names(logFC_trametinib_top) %in% names(t_test_trametinib_significant)]
upreg_trametinib_significant
```
Box plots were built to visualize the expressions of the significantly upregulated genes.
```{r,fig.show = 'hide'}
##Boxplots for upregulated genes
#Dox-shERK
data_sherk_upreg = data.skin.cancer.human_skin_TRA[upreg_sherk_significant, c(1:5)]
data.new_sherk = rbind(data_sherk_upreg, data_sherk_upreg)
data.new_sherk[c(1:2), c(4:5)] = NA
data.new_sherk[c(3:4), c(1:3)] = NA
order.vector = sort(colnames(t(data.new_sherk)),index.return=T)$ix

boxplot(t(data.new_sherk)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with Dox-shERK1", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:12), col = "grey", lty = 3)
legend("topleft", c("Dox-shERK1 after 3 days", "Dox-shERK1 after 7 days"), col = c("blue", "pink"), pch = 15, bg = "white")
```
```{r,fig.show = 'hide'}
#ERK1/2 inhibitor
data_erk12_upreg = data.skin.cancer.human_skin_TRA[upreg_erk12_significant, c(6:7, 10:12)]
data.new_erk12 = rbind(data_erk12_upreg, data_erk12_upreg)
data.new_erk12[c(1:8), c(3:5)] = NA
data.new_erk12[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_erk12)),index.return=T)$ix

boxplot(t(data.new_erk12)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with ERK1/2 inhibitor", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("ERK1/2 inhibitor after 3h", "ERK1/2 inhibitor after 24h"), col = c("blue", "pink"), pch = 15, bg = "white")
```
```{r}
#Trametinib
data_trametinib_upreg = data.skin.cancer.human_skin_TRA[upreg_trametinib_significant, c(8:9, 13:15)]
data.new_trametinib = rbind(data_trametinib_upreg, data_trametinib_upreg)
data.new_trametinib[c(1:8), c(3:5)] = NA
data.new_trametinib[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_trametinib)),index.return=T)$ix

boxplot(t(data.new_trametinib)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Upregulated genes in treatment with Trametinib", ylab = "log2(gene expression)", xlab = "Upregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("left",c("Trametinib after 3h", "Trametinib after 24h"),col=c("blue", "pink"), pch = 15, bg = "white")
```
The package “Limma” (v4.1, ???) is used as a model of comparison to the upregulated genes that were found manually using the log fold change. Limma takes the normalized data and performs statistical tests to discover any quantitative changes in the expression (Ritchie et al., 2015). The results show mostly identical genes to the already found upregulated genes. 
```{r,fig.show = 'hide'}
## Limma Analysis
library(limma)

#First group: We compare the expression values from day 3 to day 7 with the Dox-shERK treatment
# Create design matrix
design = model.matrix(~ 0+factor(c(2,2,2,1,1)))
colnames(design) = c("Sherk7D", "Sherk3D")

#Fit linear model
fit = lmFit(data.skin.cancer.human_skin_TRA[,1:5], design)
contrast.matrix = makeContrasts(Sherk7D-Sherk3D, levels=design)
fit2 = contrasts.fit(fit, contrast.matrix)
fit2 =  eBayes(fit2, 0.01)

#Create top table with the DEGs sorted by log fold change
T1 = topTable(fit2,adjust.method = "fdr", sort.by = "logFC")

#Select only the first 10 genes with a positive log fold change from the table
lola1=T1[which(T1$logFC>0),]
lola1 = lola1[1:10,]

#Confirmed top upregulated genes
intersect(upreg_sherk_significant, rownames(lola1))

#Second group: We compare 3h ERK1/2 and 24h ERK1/2 treatment
#Create design matrix
design_ERK = model.matrix(~ 0+factor(c(1,1,1,2,2)))
colnames(design_ERK) = c("ERK12_inh_24h", "ERK12_inh_3h")

#Fit linear model
data_ERK = data.skin.cancer.human_skin_TRA[,c(10:12, 6,7)]
fit_ERK = lmFit(data_ERK, design_ERK) 
contrast.matrix_ERK = makeContrasts(ERK12_inh_24h-ERK12_inh_3h, levels=design_ERK)
fit2_ERK = contrasts.fit(fit_ERK, contrast.matrix_ERK)
fit2_ERK =  eBayes(fit2_ERK, 0.01)

#Create top table with the DEGs sorted by log fold change
T2 = topTable(fit2_ERK,adjust.method = "fdr", sort.by = "logFC", number = 130) 

#Select only the first 10 genes with a positive log fold change from the table
lola2 = T2[which(T2$logFC>0),]
lola2 = lola2[1:10,]

#Confirmed top upregulated genes
intersect(upreg_erk12_significant, rownames(lola2))

#Third group: We compare 3h to 24h Trametinib treatment
#Create design matrix
design_Trametinib = model.matrix(~ 0+factor(c(1,1,1,2,2)))
colnames(design_Trametinib) = c("Trametinib_24h", "Trametinib_3h")

#Fit linear model
data_Trametinib = data.skin.cancer.human_skin_TRA[,c(13:15, 8,9)]
fit_Trametinib = lmFit(data_Trametinib, design_Trametinib) 
contrast.matrix_Trametinib = makeContrasts(Trametinib_24h-Trametinib_3h, levels=design_Trametinib)
fit2_Trametinib= contrasts.fit(fit_Trametinib, contrast.matrix_Trametinib)
fit2_Trametinib =  eBayes(fit2_Trametinib, 0.01)

#Create top table with the DEGs sorted by log fold change
T3 = topTable(fit2_Trametinib,adjust.method = "fdr", sort.by = "logFC", number = Inf ) 

#Select only the first 10 genes with a positive log fold change from the table
lola3 = T3[which(T3$logFC>0),]
lola3 = lola3[1:10,]

#Confirmed top upregulated genes
intersect(upreg_trametinib_significant, rownames(lola3))


#The genes from limma analysis were very similar to the ones found manually, which means that the manual way succeeded. This was checked by doing the intersection of the vectors for the significant upregulated genes of each treatment and the top 10 genes found with limma.
```
The existence of significantly downregulated genes indicates that the treatment has been successful, and that these genes were inducing tumors before the treatment. That is why, ten genes from different treatments with the lowest log-ratio are filtered out and evaluated under t-tests, and visualized via box plots. The following downregulated genes are respectively from dox-shERK1, ERK1/2 inhibitor, and Trametinib chips. 
```{r}
### 3.7.2. Downregulated genes
#Least logFC from each treatment
logFC_sherk_down_regulated = logFC_sherk[503:512]
logFC_erk12_down_regulated = logFC_erk12[503:512]
logFC_trametinib_down_regulated = logFC_trametinib[503:512]

#T-test: Filtering downregulated genes that are significant
#Checking significance for genes in the last 10 logFC of Dox-shERK
t_sherk = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 1:3], data.skin.cancer.human_skin_TRA[x, 4:5])$p.value
}
t_test_sherk = sort(sapply(names(logFC_sherk), t_sherk), decreasing = F)
t_test_sherk_significant = t_test_sherk[which(t_test_sherk < 0.05)]

downreg_sherk_significant = names(logFC_sherk_down_regulated)[names(logFC_sherk_down_regulated) %in% names(t_test_sherk_significant)]
downreg_sherk_significant

#Checking significance for genes in the last 10 logFC of ERK1/2 inhibitor
t_erk12 = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 6:7], data.skin.cancer.human_skin_TRA[x, 10:12])$p.value
}
t_test_erk12 = sort(sapply(names(logFC_erk12), t_erk12), decreasing = F)
t_test_erk12_significant = t_test_erk12[which(t_test_erk12 < 0.05)]

downreg_erk12_significant = names(logFC_erk12_down_regulated)[names(logFC_erk12_down_regulated) %in% names(t_test_erk12_significant)]
downreg_erk12_significant

## Checking significance for genes in the last 10 logFC of trametinib
t_trametinib = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[x, 8:9], data.skin.cancer.human_skin_TRA[x, 13:15])$p.value
}
t_test_trametinib = sort(sapply(names(logFC_trametinib), t_trametinib), decreasing = F)
t_test_trametinib_significant = t_test_trametinib[which(t_test_trametinib < 0.05)]

downreg_trametinib_significant = names(logFC_trametinib_down_regulated)[names(logFC_trametinib_down_regulated) %in% names(t_test_trametinib_significant)]
downreg_trametinib_significant
```
```{r}
##Boxplots for downregulated genes
#Dox-shERK
data_sherk_downreg = data.skin.cancer.human_skin_TRA[downreg_sherk_significant, c(1:5)]
data.new_sherk_downreg = rbind(data_sherk_downreg, data_sherk_downreg)
data.new_sherk_downreg[c(1:2), c(4:5)] = NA
data.new_sherk_downreg[c(3:4), c(1:3)] = NA
order.vector = sort(colnames(t(data.new_sherk_downreg)),index.return=T)$ix

boxplot(t(data.new_sherk_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with Dox-shERK", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:12), col = "grey", lty = 3)
legend("topleft", c("Dox-shERK after 3 days", "Dox-shERK after 7 days"), col = c("blue", "pink"), pch = 15, bg = "white")
```
```{r,fig.show = 'hide'}
#ERK1/2 inhibitor
data_erk12_downreg = data.skin.cancer.human_skin_TRA[downreg_erk12_significant, c(6:7, 10:12)]
data.new_erk12_downreg = rbind(data_erk12_downreg, data_erk12_downreg)
data.new_erk12_downreg[c(1:8), c(3:5)] = NA
data.new_erk12_downreg[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_erk12_downreg)),index.return=T)$ix

boxplot(t(data.new_erk12_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with ERK1/2 inhibitor", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("ERK1/2 inhibitor after 3h", "ERK1/2 inhibitor after 24h"), col = c("blue", "pink"), pch = 15, bg = "white")
```
```{r,fig.show = 'hide'}
#Trametinib
data_trametinib_downreg = data.skin.cancer.human_skin_TRA[downreg_trametinib_significant, c(8:9, 13:15)]
data.new_trametinib_downreg = rbind(data_trametinib_downreg, data_trametinib_downreg)
data.new_trametinib_downreg[c(1:8), c(3:5)] = NA
data.new_trametinib_downreg[c(9:16), c(1:2)] = NA
order.vector = sort(colnames(t(data.new_trametinib_downreg)),index.return=T)$ix

boxplot(t(data.new_trametinib_downreg)[,order.vector], cex.axis = 0.6, col = c("blue", "pink"), main = "Downregulated genes in treatment with Trametinib", ylab = "log2(gene expression)", xlab = "Downregulated genes", las = 2)
abline(h = c(6:14), col = "grey", lty = 3)
legend("topleft",c("Trametinib after 3h", "Trametinib after 24h"),col=c("blue", "pink"), pch = 15, bg = "white")
```
It should be noted that TYRP1 is significantly upregulated under Trametinib treatment, but significantly downregulated under dox-shERK1.

Genes that are upregullated in skin cancer and are at the same time TRAs are especially interesting, as they pose as potential drug targets. To visualize these upregulated TRAs, a venn diagram was made.
```{r}
## 3.9. Venn diagram
#Finding upregulated genes in skin cancer by logFC
#Average the gene expression for each treatment at each time point
sherk_3d_all = apply(Expression[,1:3], 1, mean)
sherk_7d_all = apply(Expression[,4:5], 1, mean)
erk12_3h_all = apply(Expression[,6:7], 1, mean)
erk12_24h_all = apply(Expression[,10:12], 1, mean)
trametinib_3h_all = apply(Expression[,8:9], 1, mean)
trametinib_24h_all = apply(Expression[,13:15], 1, mean)

## Calculate the log fold change for each gene from each treatment
logFC_sherk_all = sherk_7d_all - sherk_3d_all
logFC_erk12_all = erk12_24h_all - erk12_3h_all
logFC_trametinib_all =  trametinib_24h_all - trametinib_3h_all

## Only positive logFC from each treatment (only upregulated genes!)
logFC_sherk_pos = which(logFC_sherk_all > 0)
logFC_erk12_pos = which(logFC_erk12_all > 0)
logFC_trametinib_pos = which(logFC_trametinib_all > 0)

upreg_all = unique(c(names(logFC_sherk_pos), names(logFC_erk12_pos), names(logFC_trametinib_pos)))

# T-test to filter significantly upregulated genes
## Checking significance for Dox-shERK
t_sherk_all = function(x) {
  if (Expression[x,1] == Expression[x,2] | Expression[x,1] == Expression[x,3] | Expression[x,3] == Expression[x,2] | Expression[x,4] == Expression[x,5])
    return(0.5)
  else
  t.test(Expression[x, 1:3], Expression[x, 4:5])$p.value
}
t_test_sherk_all = sapply(names(logFC_sherk_all), t_sherk_all)
t_test_sherk_significant_all = t_test_sherk_all[which(t_test_sherk_all < 0.05)]

upreg_sherk_significant_all = names(logFC_sherk_pos)[names(logFC_sherk_pos) %in% names(t_test_sherk_significant_all)]

## Checking significance for ERK1/2 inhibitor
t_erk12_all = function(x) {
  t.test(Expression[x, 6:7], Expression[x, 10:12])$p.value
}
t_test_erk12_all = sapply(names(logFC_erk12_all), t_erk12_all)
t_test_erk12_significant_all = t_test_erk12_all[which(t_test_erk12_all < 0.05)]

upreg_erk12_significant_all = names(logFC_erk12_pos)[names(logFC_erk12_pos) %in% names(t_test_erk12_significant_all)]

## Checking significance for Trametinib
t_trametinib_all = function(x) {
  t.test(Expression[x, 8:9], Expression[x, 13:15])$p.value
}
t_test_trametinib_all = sapply(names(logFC_trametinib_all), t_trametinib_all)
t_test_trametinib_significant_all = t_test_trametinib_all[which(t_test_trametinib_all < 0.05)]

upreg_trametinib_significant_all = names(logFC_trametinib_pos)[names(logFC_trametinib_pos) %in% names(t_test_trametinib_significant_all)]

#Venn diagram (Upregulated genes that are also TRAs)
library(venn)
list_venn = list(A = unique(c(upreg_sherk_significant_all, upreg_erk12_significant_all, upreg_trametinib_significant_all)), B = skin.genes.human)
names(list_venn) = c("Upregulated genes" , "TRA")

venn(list_venn, box = F, ilab = T, zcolor = c("#0073C2FF", "#EFC000FF"), cex = 1000)
text(500, 900, "Potential drug targets", cex = 2)

#Save the venn diagramm in a pdf file
setwd(paste(projectPath, 'plots', sep = '/'))
pdf("Venn_diagramm_trial.pdf")
venn(list_venn, box = F, ilab = T, zcolor = c("#0073C2FF", "#EFC000FF"))
text(500, 900, "Potential drug targets", cex = 2)
dev.off()

#Upregulated non-skin TRAs in skin cancer
upreg_tra = intersect(unique(c(upreg_sherk_significant_all, upreg_erk12_significant_all, upreg_trametinib_significant_all)), skin.genes.human)
upreg_tra_nonskin = upreg_tra[!upreg_tra%in%rownames(data.skin.cancer.human_skin_TRA)]
length(upreg_tra_nonskin) #There are no non-skin specific TRAs that are upregulated in skin cancer
```

Expression boxplots of chips exposed to sun and not exposed to sun are made, there is not any significant difference seen between them. However, a significant difference between the expression levels of the two gene groups was observed after a Wilcoxon signed rank test was run for each chip. Genes expressed both under the sun and without the sun are removed from the data, and the genes that were only expressed under sun exposure are acquired. Four of these sun exposure related genes are also upregulated genes: ETV7, TGFBI, DAPL1, and ARHGEF5.

```{r}
## 3.10. Sun exposure
#Finding the sun exposed and not sun exposed genes in our skin-specific expression dataset
sun = intersect(rownames(data.skin.cancer.human_skin_TRA), skin.TRA3.s)
no.sun = intersect(rownames(data.skin.cancer.human_skin_TRA), skin.TRA3.ns)

#Removing the gene names to make a box plot of each chip
sun_exp = data.skin.cancer.human_skin_TRA[sun,]
nosun_exp = data.skin.cancer.human_skin_TRA[no.sun,]
sunexpchip = unname(sun_exp[,13])
nosunexpchip = unname(nosun_exp[,13])
lmts = range(sunexpchip,nosunexpchip)

#Comparing their significance with a welsh t-test for each chip
ttest_sun = function(x) {
  t.test(data.skin.cancer.human_skin_TRA[sun, x], data.skin.cancer.human_skin_TRA[no.sun, x])$p.value
}
sapply(1:15, ttest_sun)
## All of them have a p value of smaller than 0.05, proving a significant difference between the expression of sun or no sun exposure genes. So now we will compare the expression of genes that are expressed with or without sun exposure.

# Genes that are expressed with or without sun exposure
both.sun = intersect(sun, no.sun)
# Genes that are only sun exposure related
only_sun = sun[!sun%in%both.sun]
#Genes that are only no sun exposure related
only_nosun = no.sun[!no.sun%in%both.sun]

#The upregulated sun exposed genes ETV7 and TGFBI are especially interesting, as ETV7 downregulation is related to poor prognosis (so working against the cancer) and TGFBI is strongly associated with tumor proliferation.

# The visualization of these genes can be seen in the graphs showing the logFC for all significantly upregulated genes.
significant_sun_1 = upreg_sherk_significant[upreg_sherk_significant%in%only_sun]
significant_sun_2 = upreg_erk12_significant[upreg_erk12_significant%in%only_sun]
significant_sun_3 = upreg_trametinib_significant[upreg_trametinib_significant%in%only_sun]

upregulated_sun = unique(c(significant_sun_1, significant_sun_2, significant_sun_3))

significant_nosun_1 = upreg_sherk_significant[upreg_sherk_significant%in%only_nosun]
significant_nosun_2 = upreg_erk12_significant[upreg_erk12_significant%in%only_nosun]
significant_nosun_3 = upreg_trametinib_significant[upreg_trametinib_significant%in%only_nosun]

upregulated_nosun = unique(c(significant_nosun_1, significant_nosun_2, significant_nosun_3))
```

## 4. Clustering

Clustering was done to observe if there are any distinctive groups in our dataset.

```{r}
#Load libraries needed for clustering
library(factoextra)
library(clustertend)
library(fpc)
library(cluster)
library(clValid)
library(NbClust)
library(dplyr)
```

## 4.1. Clustering of sun / no sun exposure data

Sun and no sun exposure data were clustered to see if there are any obvious differences between the two gene groups.

```{r}
#Deciding on the optimal number of clusters
set.seed(123)
both = data.skin.cancer.human_skin_TRA[c(only_sun, only_nosun),]

#Elbow Method
fviz_nbclust(both, kmeans, method = "wss")+labs(subtitle = "Elbow method")

#Silhouette Method
fviz_nbclust(both, kmeans, method = "silhouette")+labs(subtitle = "Silhouette method")

#NbClust
nb.both = NbClust(both, distance = "euclidean", min.nc = 2,max.nc = 10, method = "kmeans")
fviz_nbclust(nb.both)
#The optimal number of clusters is 2.

#Is the data really clusterable?
hopkins(both, n = nrow(both)-1) #H value is lower than 0.5 and is relatively close to 1, meaning that there is a high probability that there are clusters in the sun/no sun data.

#Visualize k-means clusters
fviz_cluster(km.both, data = both, palette = c("orange", "blue"), geom = "point", ellipse.type = "convex", ggtheme = theme_bw())

#Cluster Validation: Silhouette Information
km.both = eclust(both, "kmeans", k = 2, nstart = 30, graph = FALSE)
fviz_silhouette(km.both, palette = "jco",
ggtheme = theme_classic())
#Average silhouette width of both clusters are higher than 0.5, meaning they are significantly clustered.
```


## 5. Principal component analysis

PCA was done to find the genes that explain most of the variance in the data. These genes were picked from principal components with the highest percentage of variance explained. Later on, the expression of upregulated genes will be predicted by using one or more of these genes through regression models.

```{r}
#Centering and scaling, pick data only from final time and only genes with highest variance
pca_dataframe = scale(data.skin.cancer.human_skin_TRA)
pca_dataframe = t(pca_dataframe[,c(4:5, 10:15)])

topVar = apply(pca_dataframe, 2, var)
q75 = quantile(topVar, probs = 0.75)
i.topvar = which(topVar >= q75)
pca_dataframe_topVar = pca_dataframe[,i.topvar]

#PCA
skin_pca_topVar = prcomp(pca_dataframe_topVar, scale. = F, center = F)

#Biplot of first 2 principal components
biplot(skin_pca_topVar)

#Proportion of variance of each principal component
variance = (skin_pca_topVar$sdev)^2
prop.variance = variance/sum(variance)
names(prop.variance) = 1:length(prop.variance)
barplot(prop.variance, ylab='Proportion of variance', xlab = "Principal components", main = "Proportion of variance of each principal component")

#Barplots of principal components
par(las = 2)
barplot(skin_pca_topVar$rotation[,1], main = "PC1", ylab = "Contribution", xlab = "Genes")
barplot(skin_pca_topVar$rotation[,2], main = "PC2", ylab = "Contribution", xlab = "Genes")

#Genes with most contribution to PC1 and PC2
top_genes_pca1 = names(sort(abs(skin_pca_topVar$rotation[,1]), decreasing = T)[1:5])
top_genes_pca2 = names(sort(abs(skin_pca_topVar$rotation[,2]), decreasing = T)[1:5])

#Scatterplot of data with PC1 and PC2 as new coordinates
plot(skin_pca_topVar$x[,1], skin_pca_topVar$x[,2], 
     col= c("red", "red", "green", "green", "green", "blue", "blue", "blue"), pch=19,
     xlab='PC1', ylab='PC2', main = "PC1 and PC2 as new coordinates")
legend("topright",c("Sherk", "ERK1/2 inhibitor", "Trametinib"),col=c("red", "green", "blue"), pch = 15, bg = "yellow")
```

## 6. Regression models

Finally, regression models were made to predict the expression of the upregulated TRAs by using the genes previously selected by PCA. It is hoped that the expression of these essential upregulated genes in cancer can be modulated by targeting the genes from PCA. This would then be a good opportunity in developing new drugs with new drug targets.

```{r}

```

## I

You

```{r}

```

## I

You

```{r}

```


